// @redocly/openapi-docs/lib-esm/components/common/Skeleton/LanguageListSkeleton.js
import{jsx as _jsx,jsxs as _jsxs}from"react/jsx-runtime";import{memo}from"react";import styled from"styled-components";const Container=styled.div`
  width: 100%;
  display: flex;
  justify-content: space-between;
  gap: var(--spacing-xs);
  padding: var(--spacing-xs) 0;
`,Item=styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  justify-content: center;
  align-items: center;
  gap: var(--spacing-sm);
`,SkeletonBox=styled.div`
  display: inline-block;
  height: ${({type:e})=>"icon"===e?"var(--control-height-sm)":"calc(var(--control-height-sm) * 0.5)"};
  position: relative;
  overflow: hidden;
  background-color: var(--layer-color-hover);
  border-radius: var(--border-radius);
  width: ${({type:e})=>"icon"===e?"var(--control-height-sm)":"64px"};
`;function LanguageListSkeletonComponent(){return _jsxs(Container,{children:[_jsxs(Item,{children:[_jsx(SkeletonBox,{type:"icon"}),_jsx(SkeletonBox,{type:"label"})]}),_jsxs(Item,{children:[_jsx(SkeletonBox,{type:"icon"}),_jsx(SkeletonBox,{type:"label"})]}),_jsxs(Item,{children:[_jsx(SkeletonBox,{type:"icon"}),_jsx(SkeletonBox,{type:"label"})]}),_jsxs(Item,{children:[_jsx(SkeletonBox,{type:"icon"}),_jsx(SkeletonBox,{type:"label"})]})]})}export const LanguageListSkeleton=memo(LanguageListSkeletonComponent);
