// @redocly/theme/lib/components/JsonViewer/JsonViewer.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.JsonViewerWrap = exports.JsonViewer = void 0;
const react_1 = __importStar(require("react"));
const styled_components_1 = __importDefault(require("styled-components"));
const CodeBlock_1 = require("../../components/CodeBlock/CodeBlock");
const helpers_1 = require("./helpers");
function JsonComponent({ data, expandLevel = 1, className, onCopyClick, title, }) {
    const showFoldingButtons = data && Object.values(data).some((value) => typeof value === 'object' && value !== null);
    const [expandAllSignal, setExpandAllSignal] = react_1.default.useState(undefined);
    const expandAll = () => {
        setExpandAllSignal(true);
        setTimeout(() => {
            setExpandAllSignal(undefined);
        });
    };
    const collapseAll = () => {
        setExpandAllSignal(false);
        setTimeout(() => {
            setExpandAllSignal(undefined);
        });
    };
    const source = JSON.stringify(data, null, 2);
    return (react_1.default.createElement(exports.JsonViewerWrap, { "data-testid": "json-viewer", "data-component-name": "JsonViewer/JsonViewer", className: className },
        react_1.default.createElement(CodeBlock_1.CodeBlock, { header: {
                title,
                className: 'code-block-header',
                controls: {
                    copy: { data, onClick: onCopyClick, handleOutside: true },
                    expand: showFoldingButtons ? { onClick: expandAll } : undefined,
                    collapse: showFoldingButtons ? { onClick: collapseAll } : undefined,
                },
            }, source: source },
            react_1.default.createElement(FoldingWrap, null,
                react_1.default.createElement(helpers_1.JsonValue, { value: data, level: 0, standalone: true, expandAllSignal: expandAllSignal, defaultExpandLevel: Math.max(1, expandLevel) })))));
}
exports.JsonViewer = (0, react_1.memo)(JsonComponent);
const FoldingWrap = styled_components_1.default.div `
  position: relative;
`;
exports.JsonViewerWrap = styled_components_1.default.div `
  display: flex;
  flex-direction: column;
  flex: 1;

  & > div {
    border: none;
    margin: 0;
  }

  pre {
    contain: content;
    overflow-x: auto;
    position: relative;
    border-radius: 0 0 var(--border-radius) var(--border-radius);
  }

  .code-block-header {
    border-bottom: 0;
    padding-right: var(--spacing-sm);
  }
`;
