// @redocly/openapi-docs/lib-esm/components/Tabs/Tabs.js
import{jsx as _jsx}from"react/jsx-runtime";import styled,{css}from"styled-components";import{StatusCode}from"@redocly/theme";export function Tabs({tabs:e,onChange:t,activeTab:r}){return _jsx(TabList,{children:e.map((e=>_jsx(Tab,{$active:e.key===(null==r?void 0:r.key),onClick:()=>t(e),children:_jsx(StatusCode,{status:e.key,children:e.title})},e.key)))})}const TabList=styled.ol`
  display: flex;
  flex-wrap: wrap;
  list-style: none;
  margin: 0;
  padding: 0;
`,Tab=styled.li`
  display: inline-flex;
  align-items: center;
  list-style: none;

  padding: 0 var(--spacing-sm);
  background-color: transparent;
  height: 24px;
  border-radius: var(--border-radius);
  font-family: var(--font-family-base);
  font-size: var(--font-size-base);
  font-weight: var(--font-weight-regular);
  line-height: var(--line-height-base);

  &,
  button {
    color: var(--text-color-secondary);

    ${({$active:e})=>e?css`
            color: var(--text-color-primary);
            background-color: var(--tab-bg-color-filled);
          `:css`
            &:hover {
              color: var(--text-color-primary);
            }
          `}
  }
`;
