// @redocly/theme/lib/components/Admonition/Admonition.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Admonition = Admonition;
const react_1 = __importDefault(require("react"));
const styled_components_1 = __importStar(require("styled-components"));
const WarningFilledIcon_1 = require("../../icons/WarningFilledIcon/WarningFilledIcon");
const CheckmarkFilledIcon_1 = require("../../icons/CheckmarkFilledIcon/CheckmarkFilledIcon");
const CloseFilledIcon_1 = require("../../icons/CloseFilledIcon/CloseFilledIcon");
const InformationFilledIcon_1 = require("../../icons/InformationFilledIcon/InformationFilledIcon");
const IconsMap = {
    warning: WarningFilledIcon_1.WarningFilledIcon,
    success: CheckmarkFilledIcon_1.CheckmarkFilledIcon,
    danger: CloseFilledIcon_1.CloseFilledIcon,
    info: InformationFilledIcon_1.InformationFilledIcon,
};
function Admonition({ type = 'info', name, children, className, 'data-source': dataSource, 'data-hash': dataHash, }) {
    const Icon = IconsMap[type] || IconsMap['info'];
    return (react_1.default.createElement(AdmonitionWrapper, { type: type, className: className, "data-component-name": "Admonition/Admonition", "data-source": dataSource, "data-hash": dataHash },
        react_1.default.createElement(Icon, { color: `--admonition-${type}-icon-color` }),
        react_1.default.createElement(TextContainer, null,
            name ? react_1.default.createElement(Heading, { type: type }, name) : null,
            children ? react_1.default.createElement(Content, null, children) : null)));
}
const Content = styled_components_1.default.div `
  & > p:first-child {
    margin-top: 0;
  }
`;
const AdmonitionWrapper = styled_components_1.default.div `
  display: flex;
  flex-direction: row;
  gap: var(--spacing-base);
  margin: var(--admonition-margin-vertical) var(--admonition-margin-horizontal);
  padding: var(--admonition-padding-vertical-xs) var(--admonition-padding-horizontal);
  border-radius: var(--admonition-border-radius);
  font-size: var(--admonition-font-size);
  font-weight: var(--admonition-font-weight);
  line-height: var(--admonition-line-height);

  print-color-adjust: exact;
  -webkit-print-color-adjust: exact;

  ${({ type }) => (0, styled_components_1.css) `
    background-color: var(--admonition-${type}-bg-color);
    color: var(--admonition-${type}-text-color);
    border-left: var(--admonition-${type}-border);
  `};

  &:has(${Content}) {
    padding: var(--admonition-padding-vertical-md) var(--admonition-padding-horizontal);
  }

  &:not(:has(${Content})) {
    --admonition-heading-font-size: var(--admonition-font-size);
    --admonition-heading-line-height: var(--admonition-line-height);
    --admonition-icon-size: 16px;

    align-items: center;
    gap: calc(var(--spacing-xxs) * 1.5);
  }

  svg {
    width: var(--admonition-icon-size);
    height: var(--admonition-icon-size);
    flex-shrink: 0;

    fill: ${({ type }) => `var(--admonition-${type}-icon-color)`};

    background-image: ${({ type }) => `var(--admonition-${type}-icon)`};
    background-repeat: no-repeat;
    background-position: center;
    background-size: contain;
  }
`;
const TextContainer = styled_components_1.default.div `
  display: flex;
  flex-direction: column;
  gap: var(--spacing-unit);
  width: 100%;
`;
const Heading = styled_components_1.default.div `
  letter-spacing: var(--admonition-heading-letter-spacing);
  color: ${({ type }) => `var(--admonition-${type}-heading-text-color)`};

  && {
    font-size: var(--admonition-heading-font-size);
    font-weight: var(--admonition-heading-font-weight);
    line-height: var(--admonition-heading-line-height);
    text-transform: var(--admonition-heading-transform);
  }
`;
