// @redocly/theme/lib/components/Catalog/CatalogCard.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CatalogCard = CatalogCard;
const React = __importStar(require("react"));
const styled_components_1 = __importDefault(require("styled-components"));
const utils_1 = require("../../core/utils");
const hooks_1 = require("../../core/hooks");
const ArrowRightIcon_1 = require("../../icons/ArrowRightIcon/ArrowRightIcon");
const Link_1 = require("../../components/Link/Link");
const Tag_1 = require("../../components/Tag/Tag");
const CatalogHighlight_1 = require("../../components/Catalog/CatalogHighlight");
const Button_1 = require("../../components/Button/Button");
function CatalogCard({ item }) {
    var _a, _b;
    const { useTelemetry } = (0, hooks_1.useThemeHooks)();
    const telemetry = useTelemetry();
    return (React.createElement(Link_1.Link, { key: item.docsLink || item.link, to: item.docsLink || item.link },
        React.createElement(StyledCard, { "data-component-name": "Catalog/CatalogCard", onClick: () => telemetry.send('catalog_item_clicked', {}) },
            React.createElement(CardContent, null,
                React.createElement(CardTitleWrapper, null,
                    React.createElement(CardTitle, null,
                        React.createElement(CatalogHighlight_1.CatalogHighlight, null, item.title))),
                React.createElement(CardDescription, null,
                    React.createElement(CatalogHighlight_1.CatalogHighlight, null, (_b = (_a = item.summary) !== null && _a !== void 0 ? _a : item.description) !== null && _b !== void 0 ? _b : ''))),
            React.createElement(CardFooter, null,
                React.createElement(CardTags, null,
                    (item.tags || []).map((tag, index) => (React.createElement(CardTag, { key: tag + index, color: (0, utils_1.slug)(tag) },
                        React.createElement(CatalogHighlight_1.CatalogHighlight, null, (0, utils_1.capitalize)(tag))))),
                    (item.scorecardLevel && (React.createElement(Tag_1.Tag, { borderless: true, withStatusDot: true, statusDotColor: `var(${(0, utils_1.getScorecardColorVariable)(item.scorecardLevelIdx || 0, Object.keys(item.scorecardLevels || {}).length)})` },
                        React.createElement(CatalogHighlight_1.CatalogHighlight, null, item.scorecardLevel)))) ||
                        null),
                React.createElement(SelectButton, { size: "medium", variant: "secondary", icon: React.createElement(ArrowRightIcon_1.ArrowRightIcon, null) })))));
}
const SelectButton = (0, styled_components_1.default)(Button_1.Button) `
  border-radius: 100%;
  transition: all 0.2s ease-in-out;
`;
const StyledCard = styled_components_1.default.div `
  min-height: var(--catalog-card-min-height);
  height: 100%;
  color: var(--catalog-card-text-color);
  font-size: var(--catalog-card-font-size);
  font-weight: var(--catalog-card-font-weight);
  background-color: var(--catalog-card-bg-color);
  border: var(--catalog-card-border-width) var(--catalog-card-border-style)
    var(--catalog-card-border-color);
  transition: all 0.2s ease-in-out;
  border-radius: var(--catalog-card-border-radius);
  cursor: pointer;
  display: flex;
  gap: var(--catalog-card-gap);
  padding: var(--catalog-card-padding-vertical) var(--catalog-card-padding-horizontal);
  flex-direction: column;

  &:hover ${SelectButton} {
    transform: rotate(-45deg);
  }
`;
const CardTitleWrapper = styled_components_1.default.div `
  display: flex;
  align-items: center;
  min-height: var(--catalog-card-title-min-height);
`;
const CardTitle = styled_components_1.default.div `
  display: -webkit-box;
  -webkit-line-clamp: var(--catalog-card-title-line-clamp);
  -webkit-box-orient: vertical;
  overflow: hidden;
  text-overflow: ellipsis;

  color: var(--catalog-card-title-color);
  font-size: var(--catalog-card-title-font-size);
  font-weight: var(--catalog-card-title-font-weight);
  line-height: var(--catalog-card-title-line-height);
  margin: 0;
`;
const CardDescription = styled_components_1.default.div `
  display: -webkit-box;
  -webkit-line-clamp: var(--catalog-card-description-line-clamp);
  -webkit-box-orient: vertical;
  overflow: hidden;
  text-overflow: ellipsis;
  color: var(--catalog-card-description-color);
  font-size: var(--catalog-card-description-font-size);
  font-weight: var(--catalog-card-description-font-weight);
  line-height: var(--catalog-card-description-line-height);
`;
const CardContent = styled_components_1.default.div `
  flex: 1;
  display: flex;
  gap: var(--catalog-card-content-gap);
  flex-direction: column;
`;
const CardFooter = styled_components_1.default.div `
  display: flex;
  align-items: flex-end;
  justify-content: space-between;
`;
const CardTags = styled_components_1.default.div `
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  gap: var(--spacing-xs);
  min-height: var(--spacing-xl);
`;
const CardTag = (0, styled_components_1.default)(Tag_1.Tag) `
  text-transform: inherit;
  margin: 0;
  background-color: var(--catalog-card-tag-bg);
`;
