// @redocly/openapi-docs/lib-esm/components/ViewNested/ViewNested.js
import{jsx as _jsx,jsxs as _jsxs}from"react/jsx-runtime";import styled,{css}from"styled-components";import{memo,useEffect,useState}from"react";import{isUndefined}from"@redocly/theme";import{cycleColorsByLevel}from"../PropertyDetails/cycleColorsByLevel";import{CircleIcon}from"../PropertyDetails/PlusCircleIcon";function ViewNestedComponent({expandByDefault:e,level:r,expandable:o=!1,expandText:t,hideText:s,hideDivider:i=!1,children:n,expandedAll:l}){const[d,a]=useState(null!=e&&e);useEffect((()=>{isUndefined(l)||a(l)}),[l]);const c=d?cycleColorsByLevel(r):void 0;if(!o)return n;return _jsxs(Wrapper,{$divider:!i&&!d,className:"view-nested-wrapper",children:[_jsxs(ShowProperty,{onClick:()=>{a((e=>!e))},children:[_jsx(CircleIcon,{sign:d?"-":"+",color:c}),!d&&t,d&&(s||"")]}),d&&_jsx(StyledNested,{color:c,children:n})]})}export const ViewNested=memo(ViewNestedComponent);const ShowProperty=styled.button`
  background: none;
  border: none;
  cursor: pointer;
  display: flex;
  align-items: center;
  padding: 0;
  gap: var(--spacing-xxs);
  color: var(--text-color-secondary);
  font-size: var(--font-size-base);
  font-family: var(--font-family-base);
  line-height: var(--line-height-base);
  margin-top: var(--spacing-xxs);
  width: 100%;
`,StyledNested=styled.div`
  padding-left: var(--schema-nested-offset);
  border-left: 1px solid ${({color:e})=>e||"var(--border-color-primary)"};
  margin: -10px 0 0 9px;
  ${({color:e})=>e&&css`
      .schema-name {
        color: ${e};
      }
    `}
`,Wrapper=styled.div`
  width: 100%;
  ${({$divider:e})=>e&&css`
      border-bottom: 1px solid var(--border-color-primary);
      padding-bottom: var(--schema-property-details-spacing);
    `}
`;
