// @redocly/theme/lib/layouts/ThreePanelLayout.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ThreePanelLayout = ThreePanelLayout;
const react_1 = __importDefault(require("react"));
const styled_components_1 = __importStar(require("styled-components"));
const config_1 = require("@redocly/config");
const media_css_1 = require("../core/utils/media-css");
function ThreePanelLayout(_a) {
    var { children } = _a, props = __rest(_a, ["children"]);
    return (react_1.default.createElement(Wrapper, Object.assign({ "data-component-name": "Layout/ThreePanelLayout" }, props), children));
}
const Wrapper = styled_components_1.default.div `
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 0 auto;
  width: 100%;

  ${({ layout = config_1.LayoutVariant.THREE_PANEL, collapsedSidebar }) => {
    return (0, styled_components_1.css) `
      max-width: ${layout === config_1.LayoutVariant.THREE_PANEL ? '100%' : ''};

      @media screen and (min-width: ${media_css_1.breakpoints.small}) {
        width: ${collapsedSidebar
        ? `var(--layout-${layout}-small-max-width)`
        : 'calc(100% - var(--sidebar-width))'};
        max-width: var(--layout-${layout}-small-max-width);
      }

      @media screen and (min-width: ${media_css_1.breakpoints.medium}) {
        max-width: var(--layout-${layout}-medium-max-width);
      }

      @media screen and (min-width: ${media_css_1.breakpoints.large}) {
        max-width: var(--layout-${layout}-large-max-width);
      }

      @media print {
        width: 100%;
        padding: 0;
      }
    `;
}};
`;
