// @redocly/realm/dist/server/esbuild/plugins/codegen/index.js
import{slash as e}from"../../../../shared/utils.js";function s({schemaFiles:n}){return`${n.map((o,r)=>`import * as schema${r} from '${e(o)}';`).join(`
`)}

export const schema = {
  ${n.map((o,r)=>`    ...schema${r},`).join(`
`)}
  functions: {
    ${n.map((o,r)=>`    ...schema${r}.functions,`).join(`
`)}
  },
  tags: {
    ${n.map((o,r)=>`    ...schema${r}.tags,`).join(`
`)}
  }
};`}function m({componentsFiles:n}){return`${n.map((o,r)=>`import * as components${r} from '${e(o)}';`).join(`
`)}
export const components = {
${n.map((o,r)=>`  ...components${r},`).join(`
`)}
};`}function a(n){return`
export const serverProps = {
${n.map(([t,o])=>`  "${t}": () => import('${e(o)}'),`).join(`
`)}
};`}function c(n){return`
export const requestHandlers = {
${n.map(([t,o])=>`  "${t}": () => import('${e(o)}'),`).join(`
`)}
};`}function i(n){return`
export const middleware = {
${n.map(({id:t,importPath:o})=>`  "${t}": () => import('${e(o)}'),`).join(`
`)}
};`}export{c as generateApiRequestHandlersEntry,m as generateMarkdocComponentsEntry,s as generateMarkdocSchemasEntry,i as generateMiddlewareEntry,a as generateServerPropsEntry};
