// @redocly/openapi-docs/lib-esm/components/common/Badges.js
import{Fragment as _Fragment,jsx as _jsx,jsxs as _jsxs}from"react/jsx-runtime";import{memo}from"react";import styled,{css}from"styled-components";import{Badge}from"@redocly/theme";function CustomBadgesComponent({badges:e,children:o}){if(!e)return _jsx(_Fragment,{children:o});const r=e.filter((({position:e})=>"before"===e)),t=e.filter((({position:e})=>"after"===e));return _jsxs(_Fragment,{children:[r.map((({name:e,color:o})=>_jsx(StyledBadge,{color:o,children:e},e))),o,t.map((({name:e,color:o})=>_jsx(StyledBadge,{color:o,children:e},e)))]})}export const CustomBadges=memo(CustomBadgesComponent);export const StyledBadge=styled(Badge)`
  margin-left: 0;
  background-color: ${({color:e})=>e||"var(--color-info-base)"};
  ${({deprecated:e})=>e&&css`
      color: var(--badge-deprecated-text-color);
      background-color: var(--badge-deprecated-bg-color);
      border-radius: var(--badge-deprecated-border-radius);
    `};
`;export const NavigationBadge=styled(StyledBadge)`
  margin-left: 0;
  font-size: var(--font-size-sm);
  line-height: var(--line-height-sm);
  padding: 0 var(--spacing-xxs);
  max-width: 80px;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
`;
