// @redocly/openapi-docs/lib/components/common/Fields/fields-layout.js
"use strict";var __createBinding=this&&this.__createBinding||(Object.create?function(e,r,t,o){void 0===o&&(o=t);var n=Object.getOwnPropertyDescriptor(r,t);n&&!("get"in n?!r.__esModule:n.writable||n.configurable)||(n={enumerable:!0,get:function(){return r[t]}}),Object.defineProperty(e,o,n)}:function(e,r,t,o){void 0===o&&(o=t),e[o]=r[t]}),__setModuleDefault=this&&this.__setModuleDefault||(Object.create?function(e,r){Object.defineProperty(e,"default",{enumerable:!0,value:r})}:function(e,r){e.default=r}),__importStar=this&&this.__importStar||function(e){if(e&&e.__esModule)return e;var r={};if(null!=e)for(var t in e)"default"!==t&&Object.prototype.hasOwnProperty.call(e,t)&&__createBinding(r,e,t);return __setModuleDefault(r,e),r};Object.defineProperty(exports,"__esModule",{value:!0}),exports.PropertiesTable=exports.InnerPropertiesWrap=exports.PropertyBullet=exports.PropertyDetailsCell=exports.PropertyNameCell=exports.PropertyCellWithInner=exports.PropertyCell=exports.PropertiesTableCaption=exports.getCssVariable=void 0;const styled_components_1=__importStar(require("styled-components")),theme_1=require("@redocly/theme"),utils_1=require("../../../utils"),mixins_1=require("../mixins"),getCssVariable=e=>"undefined"!=typeof document?getComputedStyle(document.documentElement).getPropertyValue(e).trim():"";exports.getCssVariable=getCssVariable,exports.PropertiesTableCaption=styled_components_1.default.caption`
  text-align: right;
  font-size: 0.9em;
  font-weight: normal;
  color: var(--text-color-secondary);
`;const repeatingGradient=styled_components_1.css`
  repeating-linear-gradient(0deg,
  var(--schemas-lines-color),
  var(--schemas-lines-color) 3px,
  transparent 3px,
  transparent 5px,
  var(--schemas-lines-color) 5px);
`;exports.PropertyCell=styled_components_1.default.td`
  box-sizing: border-box;
  position: relative;
  background-image: ${repeatingGradient};
  background-repeat: no-repeat;
  background-size: 1px 100%;
  display: block;
  overflow: hidden;
  padding: 0 20px;
  margin-bottom: 2px;

  @media screen and (min-width: ${theme_1.breakpoints.small}) {
    display: table-cell;
    overflow: initial;
    padding: 10px;
    margin-bottom: 0;
  }

  tr:first-of-type > & {
    background-image: ${repeatingGradient};
    background-position: 0 10px;
    padding-top: 0;
  }

  tr.last > & {
    background-image: ${repeatingGradient};
    padding-bottom: 0;
    background-size: 0.9px 100%;

    @media screen and (min-width: ${theme_1.breakpoints.small}) {
      background-size: 0.9px 22px;
    }
  }

  tr.last + tr > & {
    background: none;
  }

  tr.last:first-child > & {
    background-image: ${repeatingGradient};
    background-size: 0.9px 100%;
    background-position: 0 20px;

    @media screen and (min-width: ${theme_1.breakpoints.small}) {
      background-size: 0.9px 22px;
      background-position: 0 0;
    }
  }
`,exports.PropertyCellWithInner=(0,styled_components_1.default)(exports.PropertyCell)`
  padding: 0;
`,exports.PropertyNameCell=(0,styled_components_1.default)(exports.PropertyCell)`
  vertical-align: top;
  line-height: 23px;
  white-space: nowrap;
  font-size: 14px;
  font-family: var(--field-name-font-family);

  &.deprecated {
    ${mixins_1.deprecatedCss};
  }

  ${({kind:e})=>"patternProperties"===e&&styled_components_1.css`
      span.field-name {
        white-space: break-spaces;
        text-align: left;
        display: inline-table;
        line-height: 1.5;
      }
    `}

  ${()=>{const e=(0,exports.getCssVariable)("--schema-field-name-word-break");if(["break-all","break-word"].includes(e)){const r=`\n      && span:last-child {\n        white-space: normal;\n        word-break: ${e};\n        vertical-align: top;\n        max-width: calc(100% - 21px);\n        display: inline-flex;\n      }\n    `;return styled_components_1.css`
        ${(0,utils_1.typography)("schemas-property-name")}
        ${r}
      `}return(0,utils_1.typography)("schemas-property-name")}};

  .field-name {
    display: inline-flex;
    align-items: center;
  }
`,exports.PropertyDetailsCell=styled_components_1.default.td`
  box-sizing: border-box;
  vertical-align: top;

  tr.expanded & {
    border-bottom: none;
  }

  tr.last > & {
    border: none;
    background-position: top left;
    background-repeat: no-repeat;
    background-size: 1px 100%;
    padding-bottom: 0;

    /* workaround for discriminator dropdown oveflowing scrolling container,
    fixes the issue for the last item in the schema only */
    & > div > div.dropdown > .dropdown-selector-content {
      position: relative;
    }
  }

  tr:first-of-type > & {
    padding-top: 0;
  }

  border-bottom: none;
  border-left: 1px solid var(--schemas-lines-color);
  padding: 0 20px;
  width: var(--schema-default-details-width);
  min-width: 200px;

  @media screen and (min-width: ${theme_1.breakpoints.small}) {
    border-bottom: 1px solid var(--border-color-primary);
    border-left: none;
    padding: 10px 0;
  }
`,exports.PropertyBullet=styled_components_1.default.span`
  margin-right: 10px;
  color: var(--schemas-lines-color);
  font-family: var(--code-font-family);
  width: var(--schema-default-details-width);
  border-left: 1px dashed var(--schemas-lines-color);
  padding: 5px 10px;

  @media screen and (min-width: ${theme_1.breakpoints.small}) {
    border-left: none;
    padding: 10px;
  }

  &::before {
    content: '';
    display: inline-block;
    vertical-align: middle;
    width: 10px;
    height: 1px;
    background: var(--schemas-lines-color);
  }
`,exports.InnerPropertiesWrap=styled_components_1.default.div`
  padding: var(--schema-nested-offset);
`,exports.PropertiesTable=styled_components_1.default.table`
  border-collapse: separate;
  border-radius: var(--border-radius);
  width: 100%;
  font-size: var(--font-size-base);
  display: block;

  @media screen and (min-width: ${theme_1.breakpoints.small}) {
    display: table;
  }

  > tr,
  > tbody > tr {
    display: block;
    margin-bottom: 10px;
    border-spacing: 0;

    @media screen and (min-width: ${theme_1.breakpoints.small}) {
      display: table-row;
      margin-bottom: 0;
      border-spacing: 0 2px;
    }
  }

  > tr {
    vertical-align: middle;
  }

  @media screen and (max-width: ${theme_1.breakpoints.small}) and (-ms-high-contrast: none) {
    td {
      float: left;
      width: 100%;
    }
  }

  &
    ${exports.InnerPropertiesWrap},
    &
    ${exports.InnerPropertiesWrap}
    ${exports.InnerPropertiesWrap}
    ${exports.InnerPropertiesWrap},
    &
    ${exports.InnerPropertiesWrap}
    ${exports.InnerPropertiesWrap}
    ${exports.InnerPropertiesWrap}
    ${exports.InnerPropertiesWrap}
    ${exports.InnerPropertiesWrap} {
    margin-left: var(--schema-nested-offset);
    margin-bottom: var(--schema-nested-offset);
    background: var(--schema-nested-background-color);
    border-radius: var(--panel-border-radius);
  }

  &
    ${exports.InnerPropertiesWrap}
    ${exports.InnerPropertiesWrap},
    &
    ${exports.InnerPropertiesWrap}
    ${exports.InnerPropertiesWrap}
    ${exports.InnerPropertiesWrap}
    ${exports.InnerPropertiesWrap},
    &
    ${exports.InnerPropertiesWrap}
    ${exports.InnerPropertiesWrap}
    ${exports.InnerPropertiesWrap}
    ${exports.InnerPropertiesWrap}
    ${exports.InnerPropertiesWrap}
    ${exports.InnerPropertiesWrap} {
    background: var(--panel-bg-color, #fff);
  }
`;
