// @redocly/openapi-docs/lib/components/common/Fields/fields.js
"use strict";var __importDefault=this&&this.__importDefault||function(e){return e&&e.__esModule?e:{default:e}};Object.defineProperty(exports,"__esModule",{value:!0}),exports.ToggleButton=exports.ConstraintItem=exports.ExtensionValue=exports.ExampleValue=exports.DefaultValue=exports.EnumValue=exports.RecursiveLabel=exports.FieldValueLabel=exports.AccessLabel=exports.Tag=exports.AdditionalPropertiesLabel=exports.RequiredLabel=exports.PropertyLabel=exports.TypeTitle=exports.TypeName=exports.TypePrefix=exports.FieldLabel=void 0;const styled_components_1=__importDefault(require("styled-components")),theme_1=require("@redocly/theme");exports.FieldLabel=styled_components_1.default.span`
  vertical-align: middle;
  line-height: var(--schema-labels-line-height);
  font-size: var(--schema-property-labels-font-size);
  color: var(--schema-labels--text-color);
`,exports.TypePrefix=(0,styled_components_1.default)(exports.FieldLabel)`
  color: var(--schema-type-text-color);
`,exports.TypeName=(0,styled_components_1.default)(exports.FieldLabel)`
  color: var(--schema-type-text-color);
`,exports.TypeTitle=(0,styled_components_1.default)(exports.FieldLabel)`
  color: var(--schema-type-title-text-color);
  word-break: break-word;
`,exports.PropertyLabel=(0,styled_components_1.default)(exports.FieldLabel.withComponent("div"))`
  color: var(--text-color-secondary);
  font-size: var(--schema-labels-font-size);
  line-height: var(--schema-labels-line-height);
  font-weight: normal;
  display: block;
`,exports.RequiredLabel=(0,styled_components_1.default)(exports.FieldLabel)`
  color: var(--schema-property-required-label-text-color);
  font-size: var(--font-size-base);
  line-height: var(--line-height-base);
`,exports.AdditionalPropertiesLabel=(0,styled_components_1.default)(exports.PropertyLabel)`
  color: var(--schema-property-additional-label-text-color);
`,exports.Tag=styled_components_1.default.span.attrs((()=>({className:"tag-grey"})))`
  background: var(--tag-bg-color);
  padding: 0 var(--spacing-xxs);
  font-family: var(--font-family-monospaced);
  font-size: var(--font-size-sm);
  line-height: var(--line-height-sm);
  border-radius: var(--border-radius);
  color: var(--text-color-secondary);
  display: inline-flex;
  word-break: var(--code-word-break);
`,exports.AccessLabel=(0,styled_components_1.default)(exports.Tag)`
  color: var(--schema-property-access-label-text-color);
  background-color: var(--bg-color);
  border: 1px solid var(--border-color-secondary);
`,exports.FieldValueLabel=(0,styled_components_1.default)(exports.Tag)`
  background-color: var(--schema-inline-bg-color);
  border-radius: var(--border-radius);
  border: var(--schema-inline-border);
  padding: 0 var(--spacing-unit);
`,exports.RecursiveLabel=(0,styled_components_1.default)(exports.Tag)`
  background-color: var(--schema-recursive-bg-color);
  border-color: var(--schema-recursive-border-color);
  color: var(--schema-recursive-text-color);
  padding: 0 var(--spacing-xs);
`,exports.EnumValue=(0,styled_components_1.default)(exports.FieldValueLabel)`
  background-color: var(--schema-enum-bg-color);
  border-color: var(--schema-enum-border-color);
  color: var(--schema-enum-text-color);
`,exports.DefaultValue=(0,styled_components_1.default)(exports.Tag)`
  background-color: var(--schema-default-bg-color);
  border-color: var(--schema-default-border-color);
  color: var(--schema-default-text-color);
`,exports.ExampleValue=(0,styled_components_1.default)(exports.Tag)`
  background-color: var(--schema-example-bg-color);
  border-color: var(--schema-example-border-color);
  color: var(--schema-example-text-color);
  word-break: var(--code-word-break);

  & + & {
    margin-left: 0;
  }

  display: inline-flex;
  white-space: pre-wrap;
  overflow-wrap: anywhere;

  & p,
  p:only-of-type {
    font-size: var(--schema-inline-code-font-size);
    font-family: var(--schema-inline-code-font-family);
    margin: 0;
  }
`,exports.ExtensionValue=(0,styled_components_1.default)(exports.ExampleValue)``,exports.ConstraintItem=(0,styled_components_1.default)(exports.FieldValueLabel)`
  background-color: var(--schema-constraint-bg-color);
  border-color: var(--schema-constraint-border-color);
  color: var(--schema-constraint-text-color);
  word-break: var(--code-word-break);
  margin: 0 var(--spacing-unit);

  & + & {
    margin-left: 0;
  }
`,exports.ToggleButton=(0,styled_components_1.default)(theme_1.Button).attrs((()=>({variant:"link",size:"small"})))`
  padding: 0;
  font-size: var(--font-size-sm);
  line-height: var(--line-height-sm);
`;
