// @redocly/openapi-docs/lib/components/Language/styled.js
"use strict";var __createBinding=this&&this.__createBinding||(Object.create?function(e,t,n,r){void 0===r&&(r=n);var i=Object.getOwnPropertyDescriptor(t,n);i&&!("get"in i?!t.__esModule:i.writable||i.configurable)||(i={enumerable:!0,get:function(){return t[n]}}),Object.defineProperty(e,r,i)}:function(e,t,n,r){void 0===r&&(r=n),e[r]=t[n]}),__setModuleDefault=this&&this.__setModuleDefault||(Object.create?function(e,t){Object.defineProperty(e,"default",{enumerable:!0,value:t})}:function(e,t){e.default=t}),__importStar=this&&this.__importStar||function(e){if(e&&e.__esModule)return e;var t={};if(null!=e)for(var n in e)"default"!==n&&Object.prototype.hasOwnProperty.call(e,n)&&__createBinding(t,e,n);return __setModuleDefault(t,e),t};Object.defineProperty(exports,"__esModule",{value:!0}),exports.DropdownTriggerButton=exports.StyledDropdownMenuItem=exports.LanguageListItem=exports.LanguageListContainer=exports.LanguageIcon=exports.LanguageTitleContainer=exports.LanguageTitle=exports.Container=void 0;const styled_components_1=__importStar(require("styled-components")),theme_1=require("@redocly/theme"),constants_1=require("./constants");exports.Container=styled_components_1.default.span`
  display: flex;
  justify-content: space-between;
  align-items: center;
`,exports.LanguageTitle=styled_components_1.default.span`
  max-width: 150px;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  padding: 0 var(--spacing-xxs);
  font-size: var(--font-size-base);
  line-height: var(--line-height-base);
  font-weight: ${({active:e})=>e?"var(--font-weight-medium)":"var(--font-weight-regular)"};
`,exports.LanguageTitleContainer=styled_components_1.default.span`
  width: 100%;
  display: inherit;
  gap: inherit;
  flex-direction: inherit;
  justify-content: inherit;
  align-items: inherit;
`,exports.LanguageIcon=styled_components_1.default.span`
  width: 24px;
  height: 24px;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-shrink: 0;
  overflow: hidden;
`,exports.LanguageListContainer=styled_components_1.default.div`
  display: flex;
  justify-content: ${({$justifyContent:e="start"})=>e};
  align-items: center;
  width: 100%;
`,exports.LanguageListItem=styled_components_1.default.button`
  background: none;
  border: none;
  width: ${constants_1.ITEM_WIDTH}px;
  flex-shrink: 0;
  display: flex;
  align-items: center;
  cursor: pointer;
  flex-direction: column;
  gap: calc(var(--spacing-unit) * 0.5);
  padding: var(--spacing-xs) 0;
  color: ${({active:e})=>e?"var(--text-color-primary)":"var(--text-color-description)"};
  border-radius: var(--border-radius);
  text-align: center;
  position: relative;

  :hover {
    color: var(--text-color-primary);
  }

  ${({active:e})=>e&&styled_components_1.css`
      &::before {
        position: absolute;
        top: -21px;
        border-top: 3px solid var(--text-color-primary);
        width: 100%;
        content: '';
        border-bottom-left-radius: calc(var(--border-radius) / 2);
        border-bottom-right-radius: calc(var(--border-radius) / 2);
      }
    `}
`,exports.StyledDropdownMenuItem=(0,styled_components_1.default)(theme_1.DropdownMenuItem)`
  display: flex;
  justify-content: space-between;
  align-items: center;
  gap: var(--spacing-xs);
  padding: calc(var(--spacing-unit) * 1.5) var(--spacing-xs) calc(var(--spacing-unit) * 1.5)
    var(--spacing-sm);
  width: ${({$width:e=204})=>e}px;
  fill: var(--menu-content-color-active);
`,exports.DropdownTriggerButton=(0,styled_components_1.default)(theme_1.Button)`
  display: inline-flex;

  line-height: var(--line-height-base);
  font-size: var(--font-size-base);
  font-weight: var(--font-weight-medium);
  cursor: pointer;
`;
