// @redocly/theme/src/components/StatusCode/StatusCode.ts
import styled from 'styled-components';

export const StatusCode = styled.button.attrs({ 'data-component-name': 'StatusCode/StatusCode' })<{
  status: string;
}>`
  background: none;
  padding: 0;
  display: flex;
  align-items: center;
    
  &:before {
    display: inline-block;
    content: '';
    border-radius: 50%;
    margin-right: 4px;
    height: var(--status-code-height);
    width: var(--status-code-width);
    border: var(--status-code-border);
    background-color: ${({ status }) => {
      if (Math.floor(Number(status) / 100) === 2) {
        return 'var(--status-code-200-bg-color)';
      } else if (Math.floor(Number(status) / 100) === 4) {
        return 'var(--status-code-400-bg-color)';
      } else if (Math.floor(Number(status) / 100) === 5) {
        return 'var(--status-code-500-bg-color)';
      } else if (Math.floor(Number(status) / 100) === 3) {
        return 'var(--status-code-300-bg-color)';
      } else {
        return 'var(--status-code-100-bg-color)';
      }
    }}
`;
