// @redocly/openapi-docs/lib/components/ApiInfo/styled.js
"use strict";var __importDefault=this&&this.__importDefault||function(e){return e&&e.__esModule?e:{default:e}};Object.defineProperty(exports,"__esModule",{value:!0}),exports.InfoSpanBox=exports.InfoSpanBoxWrap=exports.InfoSpan=exports.ApiHeader=void 0;const styled_components_1=__importDefault(require("styled-components")),theme_1=require("@redocly/theme"),delimiterWidth=15;exports.ApiHeader=(0,styled_components_1.default)(theme_1.H1)`
  margin-top: 0;
  margin-bottom: 0;
  & + ${theme_1.Markdown} {
    margin-top: var(--h1-margin-bottom);
  }
`,exports.InfoSpan=styled_components_1.default.span`
  &::before {
    content: '|';
    display: inline-block;
    opacity: 0.5;
    width: ${15}px;
    text-align: center;
  }

  &:last-child::after {
    display: none;
  }
`,exports.InfoSpanBoxWrap=styled_components_1.default.div`
  overflow: hidden;
`,exports.InfoSpanBox=styled_components_1.default.div`
  display: flex;
  flex-wrap: wrap;
  // hide separator on new lines: idea from https://stackoverflow.com/a/31732902/1749888
  margin-left: -${15}px;
`;
