// @redocly/openapi-docs/lib/components/Responses/styled.js
"use strict";var __importDefault=this&&this.__importDefault||function(e){return e&&e.__esModule?e:{default:e}};Object.defineProperty(exports,"__esModule",{value:!0}),exports.StyledDescription=exports.StyledTitle=exports.Row=exports.StyledHeadersProperties=exports.HeadersCaption=void 0;const styled_components_1=__importDefault(require("styled-components")),OperationItemTitle_1=require("../common/OperationItemTitle"),Markdown_1=require("../Markdown");exports.HeadersCaption=styled_components_1.default.div`
  position: relative;
  display: flex;
  align-items: center;
  gap: var(--spacing-xs);
  font-size: 18px;
  line-height: var(--line-height-lg);
  font-weight: var(--font-weight-semibold);
  color: var(--text-color-primary);
  padding: var(--spacing-xxs) 0;
`,exports.StyledHeadersProperties=styled_components_1.default.div`
  margin-top: var(--spacing-md);
`,exports.Row=styled_components_1.default.div`
  display: flex;
  align-items: center;
  margin-top: var(--spacing-base);
`,exports.StyledTitle=(0,styled_components_1.default)(OperationItemTitle_1.Title)`
  margin: 0 var(--spacing-sm) 0 0;
  position: relative;
`,exports.StyledDescription=(0,styled_components_1.default)(Markdown_1.Markdown)`
  margin-top: var(--spacing-xs);
`;
