// @redocly/graphql-docs/lib/src/definition/components/styled.js
var g=Object.create;var i=Object.defineProperty;var f=Object.getOwnPropertyDescriptor;var b=Object.getOwnPropertyNames;var x=Object.getPrototypeOf,h=Object.prototype.hasOwnProperty;var y=(o,a)=>{for(var e in a)i(o,e,{get:a[e],enumerable:!0})},d=(o,a,e,l)=>{if(a&&typeof a=="object"||typeof a=="function")for(let t of b(a))!h.call(o,t)&&t!==e&&i(o,t,{get:()=>a[t],enumerable:!(l=f(a,t))||l.enumerable});return o};var u=(o,a,e)=>(e=o!=null?g(x(o)):{},d(a||!o||!o.__esModule?i(e,"default",{value:o,enumerable:!0}):e,o)),$=o=>d(i({},"__esModule",{value:!0}),o);var L={};y(L,{Annotation:()=>c,ArgsCollapsed:()=>k,Breadcrumbs:()=>S,ExpandableFieldBody:()=>N,FieldInfo:()=>H,FieldName:()=>I,FieldNameBox:()=>F,FieldWrapper:()=>v,InlineCode:()=>B,PanelAnnotation:()=>w,PanelBody:()=>E,PanelHeading:()=>q,PanelWrapper:()=>W,ReferenceField:()=>P,ScalarAnnotation:()=>z,SchemaItemWrapper:()=>A,Value:()=>C,requiredCss:()=>m});module.exports=$(L);var r=u(require("styled-components")),s=require("@redocly/theme"),p=require("../../../shared/index.js"),n=require("../../common");const c=r.default.div`
  margin-bottom: var(--spacing-base);
  color: var(--text-color-primary);
  word-break: break-all;
`,w=(0,r.default)(c)`
  margin-bottom: 0;
  color: var(--text-color-secondary);
`,z=(0,r.default)(c)`
  margin-top: var(--spacing-base);
`,k=r.default.span`
  font-size: ${({size:o})=>`var(--font-size-${o})`};
  line-height: ${({size:o})=>`var(--line-height-${o})`};
  white-space: normal;

  margin-left: var(--spacing-unit);
  color: var(--text-color-description);
  font-weight: var(--font-weight-bold);
`,m=r.css`
  &:after {
    content: '*';
    vertical-align: text-bottom;
    color: var(--schema-property-required-label-text-color);
  }
`,B=r.default.span`
  border: 1px solid var(--schema-inline-border-color);
  border-radius: var(--border-radius);
  padding: 0 4px;
  background-color: var(--schema-inline-background-color);

  color: var(--schema-inline-code-text-color);

  ${({required:o})=>o?m:null}
`,v=r.default.div`
  padding: var(--spacing-base) 0;

  ${({firstOfType:o})=>o&&r.css`
      padding-top: var(--spacing-sm);
    `};

  ${({showBorder:o})=>o?r.css`
          border-bottom: 1px solid var(--border-color-primary);
        `:r.css`
          padding-bottom: 0;
        `}

  &:first-of-type {
    padding-top: var(--spacing-sm);
  }
`,F=(0,r.default)(n.Box)`
  display: flex;
  gap: var(--spacing-unit);
  overflow: hidden;
  color: var(--schemas-property-name-text-color);
  font-size: var(--schemas-property-name-font-size);
  font-family: var(--schemas-property-name-font-family);

  ${p.ShelfIcon} {
    margin-right: 4px;
    height: var(--schema-chevron-size);
    width: var(--schema-chevron-size);
    fill: var(--schema-chevron-color);
  }
`,A=r.default.div`
  border-bottom: 1px solid var(--border-color-primary);
  padding: var(--spacing-sm) 0;
`,I=r.default.span`
  color: var(--schemas-property-name-text-color);
  font-size: var(--schemas-property-name-font-size);
  font-family: var(--schemas-property-name-font-family);
  font-weight: var(--font-weight-bold);

  ${({isDeprecated:o})=>r.css`
    text-decoration: ${o&&"line-through"};
  `};
`,P=r.default.span`
  color: var(--text-color-secondary);

  ${n.AnchorLink} {
    color: var(--text-color-secondary);
  }
`,C=r.default.div`
  display: flex;
  font-family: var(--font-family-base);
  font-size: var(--font-size-base);
  font-weight: var(--font-weight-semibold);
  line-height: var(--line-height-md);
  color: var(--text-color-secondary);
`,N=r.default.div`
  display: flex;
  flex-direction: column;
  ${({contrast:o})=>r.css`
    border-radius: var(--panel-border-radius);
    background-color: ${o?"var(--schema-nested-background-color)":""};
  `};

  ${v} {
    background-color: transparent;
  }

  h4 {
    margin-top: 0;
  }
`,S=(0,r.default)(s.Breadcrumbs)`
  padding: var(--panel-gap-vertical) var(--panel-gap-horizontal);
  padding-bottom: 0;
`,W=r.default.div.attrs({className:"panel-overview"})`
  border: 1px solid var(--border-color-secondary);
  border-radius: var(--border-radius);
  background-color: var(--panel-bg-color);

  @media screen and (min-width: ${s.breakpoints.large}) {
    margin-top: ${({$spacing:o="base"})=>`var(--spacing-${o})`};
  }
`,q=r.default.div`
  font-family: var(--font-family-base);
  font-size: var(--font-size-base);
  font-weight: var(--font-weight-semibold);
  padding: calc(var(--spacing-sm) + 2px) var(--spacing-md) calc(var(--spacing-xs) + 2px);
  border-bottom: 1px solid var(--border-color-secondary);
`,E=r.default.div`
  display: flex;
  flex-direction: column;
  padding: var(--spacing-sm) var(--spacing-md) var(--spacing-md);
  gap: var(--spacing-xs);
  max-height: 400px;
  overflow-x: auto;
`,H=(0,r.default)(n.Box)`
  border-bottom: ${({expanded:o})=>o?"none!important":"inherit"};
`;0&&(module.exports={Annotation,ArgsCollapsed,Breadcrumbs,ExpandableFieldBody,FieldInfo,FieldName,FieldNameBox,FieldWrapper,InlineCode,PanelAnnotation,PanelBody,PanelHeading,PanelWrapper,ReferenceField,ScalarAnnotation,SchemaItemWrapper,Value,requiredCss});
