// @redocly/openapi-docs/lib/components/common/OperationItemTitle.js
"use strict";var __importDefault=this&&this.__importDefault||function(e){return e&&e.__esModule?e:{default:e}};Object.defineProperty(exports,"__esModule",{value:!0}),exports.Heading=exports.Title=void 0;const styled_components_1=__importDefault(require("styled-components")),theme_1=require("@redocly/theme");exports.Title=styled_components_1.default.h4`
  font-size: var(--h4-font-size);
  font-weight: var(--h4-font-weight);
  line-height: var(--h4-line-height);
  padding: 0;
  color: var(--h4-text-color);
  margin: 0 0 var(--h4-margin-bottom) 0;
  display: flex;
  align-items: center;
  justify-content: space-between;
  position: relative;

  :hover {
    ${theme_1.LinkIcon} {
      opacity: 1;
      visibility: visible;
    }
  }
`,exports.Heading=(0,styled_components_1.default)(theme_1.H2)`
  position: relative;
  display: flex;
  align-items: center;
  gap: var(--spacing-xs);
  flex-wrap: wrap;

  :hover {
    ${theme_1.LinkIcon} {
      opacity: 1;
      visibility: visible;
    }
  }
`;
