// @redocly/openapi-docs/lib/components/common/schema.js
"use strict";var __createBinding=this&&this.__createBinding||(Object.create?function(e,t,o,r){void 0===r&&(r=o);var n=Object.getOwnPropertyDescriptor(t,o);n&&!("get"in n?!t.__esModule:n.writable||n.configurable)||(n={enumerable:!0,get:function(){return t[o]}}),Object.defineProperty(e,r,n)}:function(e,t,o,r){void 0===r&&(r=o),e[r]=t[o]}),__setModuleDefault=this&&this.__setModuleDefault||(Object.create?function(e,t){Object.defineProperty(e,"default",{enumerable:!0,value:t})}:function(e,t){e.default=t}),__importStar=this&&this.__importStar||function(e){if(e&&e.__esModule)return e;var t={};if(null!=e)for(var o in e)"default"!==o&&Object.prototype.hasOwnProperty.call(e,o)&&__createBinding(t,e,o);return __setModuleDefault(t,e),t};Object.defineProperty(exports,"__esModule",{value:!0}),exports.SelectionTitle=exports.Wrapper=exports.OneOfList=exports.OneOfButton=void 0;const styled_components_1=__importStar(require("styled-components")),mixins_1=require("./mixins");exports.OneOfButton=styled_components_1.default.button`
  border: 1px solid var(--schema-buttons-border-color);
  height: 24px;
  border-radius: var(--border-radius);
  cursor: pointer;
  outline: none;
  font-family: var(--schema-buttons-font-family);
  font-weight: var(--schema-buttons-font-weight);
  font-size: var(--schema-buttons-font-size);
  line-height: 1em;
  color: var(--schema-buttons-text-color);
  background-color: var(--schema-buttons-bg-color);
  padding: 0 10px;
  min-width: 90px;
  transition:
    background-color 0.2s ease-in,
    border-color 0.2s ease-in;

  &:hover {
    background-color: var(--schema-buttons-bg-color-hover);
  }

  &:active {
    background-color: var(--schema-buttons-bg-color-active);
  }

  ${({deprecated:e})=>e&&mixins_1.deprecatedCss};

  ${({selected:e})=>e&&styled_components_1.css`
      border-color: var(--schema-buttons-selected-border-color);
      background-color: var(--schema-buttons-selected-bg-color);
      color: var(--schema-buttons-selected-text-color);

      &:hover {
        background-color: var(--schema-buttons-selected-bg-color-hover);
      }
      &:active {
        background-color: var(--schema-buttons-active-hover-background-color);
      }
    `}
`,exports.OneOfList=styled_components_1.default.div`
  display: inline;

  flex-wrap: wrap;
  margin: 4px -3px;
  margin-left: 10px;

  ${exports.OneOfButton} {
    margin: 2px;
  }
`,exports.Wrapper=styled_components_1.default.div`
  margin-bottom: 1em;
`,exports.SelectionTitle=styled_components_1.default.div`
  font-size: var(--font-size-sm);
  line-height: var(--line-height-sm);
  color: var(--tag-basic-content-color);
  padding: 0 var(--spacing-xs);
  border: 1px solid var(--border-color-secondary);
  border-radius: var(--border-radius);
  margin: var(--spacing-xxs) 0 var(--spacing-xs);
  position: relative;
  width: fit-content;
  &:before {
    content: ' ';
    width: 1px;
    height: var(--spacing-xs);
    background: var(--border-color-secondary);
    display: block;
    position: absolute;
    bottom: calc(-1 * var(--spacing-xs));
    left: var(--spacing-xs);
  }
`;
