// @redocly/openapi-docs/lib/components/common/styled.js
"use strict";var __importDefault=this&&this.__importDefault||function(e){return e&&e.__esModule?e:{default:e}};Object.defineProperty(exports,"__esModule",{value:!0}),exports.MimeLabel=exports.HttpVerb=exports.Container=void 0;const styled_components_1=__importDefault(require("styled-components")),theme_1=require("@redocly/theme");exports.Container=styled_components_1.default.div`
  display: flex;
  flex-direction: column;
  margin: var(--spacing-unit) var(--spacing-md) var(--spacing-sm);
`,exports.HttpVerb=(0,styled_components_1.default)(theme_1.Tag)`
  text-transform: uppercase;
  background-color: unset;
  border: none;
  padding: 0;
  margin: 0;
  font-family: var(--font-family-monospaced);
  font-weight: var(--font-weight-semibold);
  font-size: var(--font-size-sm);
  line-height: var(--line-height-sm);
  align-self: flex-end;
`,exports.MimeLabel=styled_components_1.default.div`
  background-color: var(--panel-samples-dropdown-bg-color);
  color: var(--panel-samples-text-color);
  font-family: inherit;
  margin-bottom: 0;
  height: 36px;
  padding: var(--docs-dropdown-padding);
  font-size: var(--docs-dropdown-font-size);
  border: var(--panel-samples-dropdown-border);
  border-radius: var(--border-radius-md);
`;
