// @redocly/openapi-docs/lib/components/PropertyDetails/styled.js
"use strict";var __importDefault=this&&this.__importDefault||function(e){return e&&e.__esModule?e:{default:e}};Object.defineProperty(exports,"__esModule",{value:!0}),exports.FieldDescriptionWrapper=exports.ExamplesList=exports.ExampleWrap=exports.ExampleSummary=exports.ExampleDescription=exports.TypeWrapper=exports.DescriptionEnumsBlock=void 0;const styled_components_1=__importDefault(require("styled-components")),theme_1=require("@redocly/theme");exports.DescriptionEnumsBlock=(0,styled_components_1.default)(theme_1.Markdown)`
  table.md {
    margin: var(--spacing-xs) 0 0;
  }
`,exports.TypeWrapper=styled_components_1.default.div`
  line-height: 20px;
`,exports.ExampleDescription=styled_components_1.default.div``,exports.ExampleSummary=styled_components_1.default.label`
  display: block;
  font-size: inherit;
  margin-bottom: 0.2em;
  font-weight: var(--font-weight-bold);
`,exports.ExampleWrap=styled_components_1.default.div`
  margin-top: 1em;
  font-size: var(--field-name-font-size);
`,exports.ExamplesList=styled_components_1.default.div`
  padding-left: 1.5em;
  margin-bottom: 1em;
`,exports.FieldDescriptionWrapper=styled_components_1.default.div`
  p {
    overflow-wrap: anywhere;
  }
`;
