// @redocly/openapi-docs/lib-esm/components/Overlay.js
import{jsx as _jsx,jsxs as _jsxs}from"react/jsx-runtime";import{useState,useEffect}from"react";import styled from"styled-components";const INVALID_KEY="Invalid license key",INVALID_KEY_MISSING=INVALID_KEY+": missing key",EXPIRED="Invalid license key: expired",TRIAL_EXPIRED=EXPIRED,INVALID_HOST="Invalid license key: host not allowed";export const Overlay=({licenseInfo:e})=>{const[i,s]=useState(null),[t,r]=useState(-1),[l,n]=useState(!1);return useEffect((()=>{var i;e.local?n(!0):(e.valid?e.allowed?e.expired&&s(e.t?EXPIRED:TRIAL_EXPIRED):s(INVALID_HOST):s(!1===e.valid?INVALID_KEY+(e.cryptoMissing?": works only with https":""):INVALID_KEY_MISSING),e.t&&r(Math.ceil(((null!==(i=null==e?void 0:e.e)&&void 0!==i?i:0)-Date.now()/1e3)/60/60/24)))}),[e]),l?_jsxs(TrialWrap,{children:[_jsx("a",{href:"https://redoc.ly",children:"Redocly API reference docs"})," preview"]}):!i&&t>-1?_jsxs(TrialWrap,{children:[" Trial version: ",t," days remaining "]}):i?_jsx(Flash,{children:_jsxs("span",{children:[i," ",_jsx("br",{}),_jsxs("small",{children:["Please contact ",_jsx("a",{href:"mailto:team@redoc.ly",children:"team@redoc.ly"})," to remedy the problem."," "]})]})}):null};const TrialWrap=styled.div`
  padding: 10px 40px;
  position: fixed;
  text-align: right;
  width: 100%;
  box-sizing: border-box;
  color: orange;
  z-index: var(--z-index-surface);
  opacity: 0.6;

  > a,
  a:visited {
    color: orange;
  }
`,Flash=styled.div`
  font-size: 1.5em;
  position: fixed;
  z-index: var(--z-index-raised);
  background: rgba(38, 50, 56, 0.93);
  margin: 0;
  padding: 1em;
  width: 100%;
  color: white;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;

  text-align: center;
  box-sizing: border-box;

  > span {
    display: inline-block;
    line-height: 1.5;
  }

  a,
  a:hover,
  a:visited {
    color: white;
  }
`;
