// @redocly/portal-legacy-ui/lib/Burger.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Burger = void 0;
const react_1 = __importDefault(require("react"));
const styled_components_1 = __importDefault(require("styled-components"));
function BurgerComponent({ onClick, className }) {
    return (react_1.default.createElement(Button, { onClick: onClick, className: className, "data-component-name": "ui/Burger" },
        react_1.default.createElement(Line, null),
        react_1.default.createElement(Line, null),
        react_1.default.createElement(Line, null)));
}
const Button = styled_components_1.default.button `
  all: unset;
  max-width: 20px;
  min-width: 20px;
  width: 100%;
  min-width: 10px;
  display: block;
  cursor: pointer;
`;
const Line = styled_components_1.default.div `
  margin: 4px 0;
  width: 100%;
  height: 2px;
  background: var(--border-color-primary);
  border-radius: 1px;
`;
exports.Burger = (0, styled_components_1.default)(BurgerComponent) ``;
