// @redocly/theme/lib/components/Search/SearchShortcut.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchShortcut = SearchShortcut;
const React = __importStar(require("react"));
const styled_components_1 = __importDefault(require("styled-components"));
function SearchShortcut({ combination, text, className }) {
    return (React.createElement(ShortcutWrapper, { "data-component-name": "Search/SearchShortcut", className: className },
        React.createElement(ShortcutKey, null, combination),
        text));
}
const ShortcutWrapper = styled_components_1.default.div `
  display: flex;
  font-size: var(--font-size-sm);
  font-weight: var(--font-weight-regular);
  line-height: var(--line-height-sm);
  color: var(--search-item-text-color);
  gap: var(--spacing-xs);
`;
const ShortcutKey = styled_components_1.default.span `
  background: var(--bg-color-active);
  border-radius: var(--border-radius);
  padding: 0px var(--spacing-xxs);
`;
