// @redocly/theme/lib/components/Search/SearchSuggestedPages.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchSuggestedPages = SearchSuggestedPages;
const React = __importStar(require("react"));
const styled_components_1 = __importDefault(require("styled-components"));
const hooks_1 = require("../../core/hooks");
const RecentlyViewedIcon_1 = require("../../icons/RecentlyViewedIcon/RecentlyViewedIcon");
const Link_1 = require("../../components/Link/Link");
function SearchSuggestedPages({ className }) {
    const pages = (0, hooks_1.useSuggestedPages)();
    const { useTranslate } = (0, hooks_1.useThemeHooks)();
    const { translate } = useTranslate();
    if (!pages.length)
        return null;
    return (React.createElement(SearchSuggestedWrapper, { "data-component-name": "Search/SearchSuggestedPages", className: className },
        React.createElement(SearchSuggestedTitle, { "data-translation-key": "search.suggested" }, translate('search.suggested', 'Suggested pages')),
        React.createElement(SearchSuggestedItems, null, pages.map((page) => page.link ? (React.createElement(SearchSuggestedItem, Object.assign({ key: page.label, to: page.link }, page),
            React.createElement(SearchSuggestedItemWrapper, null,
                React.createElement(RecentlyViewedIcon_1.RecentlyViewedIcon, { color: "--search-item-text-color" }),
                page.label))) : null))));
}
const SearchSuggestedWrapper = styled_components_1.default.div `
  display: flex;
  flex-direction: column;
`;
const SearchSuggestedTitle = styled_components_1.default.div `
  color: var(--search-suggested-item-title-text-color);
  font-size: var(--search-suggested-item-title-font-size);
  font-weight: var(--search-suggested-item-title-font-weight);
  line-height: var(--search-suggested-item-title-line-height);
  padding: var(--search-suggested-item-title-padding);
`;
const SearchSuggestedItems = styled_components_1.default.div `
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  order: 1;
`;
const SearchSuggestedItem = (0, styled_components_1.default)(Link_1.Link) `
  width: 100%;
  text-decoration: none;
  font-size: var(--search-suggested-item-font-size);
  font-weight: var(--search-suggested-item-font-weight);
  line-height: var(--search-suggested-item-line-height);
  color: var(--search-suggested-item-text-color);
  padding: var(--search-suggested-item-padding);
  cursor: pointer;

  &:hover {
    background-color: var(--search-suggested-item-bg-color-hover);
    color: var(--search-suggested-item-text-color-hover);
  }

  &:focus-visible {
    outline: 1px solid var(--search-suggested-item-border-color-focused);
    background-color: var(--search-suggested-item-bg-color-active);
  }
`;
const SearchSuggestedItemWrapper = styled_components_1.default.div `
  display: flex;
  align-items: center;
  gap: var(--spacing-xs);
`;
