// @redocly/theme/lib/components/Search/SearchTrigger.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchTrigger = SearchTrigger;
const React = __importStar(require("react"));
const styled_components_1 = __importDefault(require("styled-components"));
const hooks_1 = require("../../core/hooks");
const Button_1 = require("../../components/Button/Button");
const SearchIcon_1 = require("../../icons/SearchIcon/SearchIcon");
const utils_1 = require("../../core/utils");
function SearchTrigger({ onClick, className }) {
    var _a, _b;
    const themeSettings = (0, hooks_1.useThemeConfig)();
    const { useTranslate } = (0, hooks_1.useThemeHooks)();
    const { translate } = useTranslate();
    const keyShortcuts = (_b = (_a = themeSettings === null || themeSettings === void 0 ? void 0 : themeSettings.search) === null || _a === void 0 ? void 0 : _a.shortcuts) !== null && _b !== void 0 ? _b : ['/'];
    let mainShortcutKey = null;
    if (keyShortcuts) {
        if (Array.isArray(keyShortcuts)) {
            mainShortcutKey = keyShortcuts[0];
        }
        else {
            mainShortcutKey = keyShortcuts;
        }
    }
    mainShortcutKey = mainShortcutKey === null || mainShortcutKey === void 0 ? void 0 : mainShortcutKey.toUpperCase();
    return (React.createElement(SearchTriggerWrapper, { onClick: onClick, className: className, "data-component-name": "Search/SearchTrigger" },
        React.createElement(SearchTriggerButton, { "data-testid": "search-trigger-button", variant: "text", size: "medium", icon: React.createElement(SearchIcon_1.SearchIcon, null) }),
        React.createElement(SearchTriggerInput, { "data-testid": "search-trigger-input", "data-translation-key": "search.navbar.label" },
            React.createElement(SearchIcon_1.SearchIcon, null),
            translate('search.navbar.label', 'Search'),
            mainShortcutKey && React.createElement("span", null, mainShortcutKey))));
}
const SearchTriggerWrapper = styled_components_1.default.div `
  color: var(--search-trigger-color);
  line-height: var(--search-trigger-line-height);

  svg {
    width: var(--search-trigger-icon-size);
    height: var(--search-trigger-icon-size);
  }
`;
const SearchTriggerButton = (0, styled_components_1.default)(Button_1.Button) `
  display: inline-flex;

  @media screen and (min-width: ${utils_1.breakpoints.medium}) {
    display: none;
  }
`;
const SearchTriggerInput = styled_components_1.default.div `
  display: none;
  align-items: center;
  justify-content: center;
  gap: var(--search-trigger-gap);
  border: var(--search-trigger-border-width) var(--search-trigger-border-style)
    var(--search-trigger-border-color);
  border-radius: var(--search-trigger-border-radius);
  background: var(--search-trigger-bg-color);
  padding: var(--search-trigger-padding);

  &:hover {
    border-color: var(--search-trigger-border-color-hover);
  }

  @media screen and (min-width: ${utils_1.breakpoints.medium}) {
    display: inline-flex;
  }
`;
