// @redocly/openapi-docs/lib-esm/components/TagItem/OperationNavigationItems.js
import{jsx as _jsx,jsxs as _jsxs,Fragment as _Fragment}from"react/jsx-runtime";import{memo}from"react";import styled from"styled-components";import{joinWithSeparator}from"../../services";import{encodeBackSlashes}from"../../utils";import{HttpVerb,NavigationBadge}from"../common";import{StyledButton}from"./styled";function OperationNavigationItemsComponent({title:e,items:t,onClick:a,routingBasePath:r,translate:o}){return _jsxs(_Fragment,{children:[_jsx(Heading,{children:e}),t.map((({id:e,href:t,httpVerb:i,path:n,deprecated:s,badges:l})=>_jsx(Item,{variant:"outlined",size:"large",onClick:()=>a(joinWithSeparator(r,encodeBackSlashes(t))),children:_jsxs("span",{children:[_jsx(HttpVerb,{color:s?"http-deprecated":i,children:i}),_jsx(Path,{children:n}),s&&_jsx(NavigationBadge,{deprecated:!0,children:o("openapi.badges.deprecated","Deprecated")}),null==l?void 0:l.map((({name:e,color:t})=>_jsx(NavigationBadge,{color:t,children:e},e)))]})},e)))]})}export const OperationNavigationItems=memo(OperationNavigationItemsComponent);const Heading=styled.span`
  font-size: var(--font-size-base);
  line-height: var(--line-height-base);
  font-weight: var(--font-weight-bold);
  color: var(--text-color-primary);
  margin: var(--spacing-sm) 0 var(--spacing-xxs) 0;
`,Item=styled(StyledButton)`
  border-radius: var(--border-radius);
  background: var(--bg-color);
  justify-content: space-between;

  & > span {
    display: flex;
    align-items: baseline;
    gap: var(--spacing-unit);
    overflow-x: hidden;
  }

  &:hover {
    background: var(--bg-color);
  }

  &:hover::after {
    content: '→';
    line-height: var(--line-height-base);
  }
`,Path=styled.span`
  font-family: var(--font-family-base);
  font-size: var(--font-size-base);
  line-height: var(--line-height-base);
  color: var(--text-color-primary);
  text-decoration: none;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow-x: hidden;
`;
