// @redocly/theme/lib/components/Filter/Filter.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Filter = Filter;
const react_1 = __importDefault(require("react"));
const styled_components_1 = __importDefault(require("styled-components"));
const FilterSelect_1 = require("../../components/Filter/FilterSelect");
const FilterCheckboxes_1 = require("../../components/Filter/FilterCheckboxes");
const FilterDateRange_1 = require("../../components/Filter/FilterDateRange");
const filterComponents = {
    select: FilterSelect_1.FilterSelect,
    'date-range': FilterDateRange_1.FilterDateRange,
    checkboxes: FilterCheckboxes_1.FilterCheckboxes,
};
function Filter({ filter, filterValuesCasing }) {
    if (!filter.parentUsed)
        return null;
    const FilterComponent = filterComponents[(filter.type || 'checkboxes')];
    return (react_1.default.createElement(FilterGroup, { "data-component-name": "Filter/Filter", key: filter.property + filter.title },
        react_1.default.createElement(FilterComponent, { filter: filter, filterValuesCasing: filterValuesCasing })));
}
const FilterGroup = styled_components_1.default.div `
  padding: var(--filter-group-padding);
  display: flex;
  flex-direction: column;
  gap: var(--filter-group-gap);
`;
