// @redocly/theme/lib/components/Filter/FilterPopover.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FilterPopover = FilterPopover;
const react_1 = __importDefault(require("react"));
const styled_components_1 = __importDefault(require("styled-components"));
const utils_1 = require("../../core/utils");
const hooks_1 = require("../../core/hooks");
const FilterContent_1 = require("../../components/Filter/FilterContent");
function FilterPopover({ onClose, filters, filterValuesCasing, filterTerm, setFilterTerm, }) {
    const { useTranslate } = (0, hooks_1.useThemeHooks)();
    const { translate } = useTranslate();
    return (react_1.default.createElement(FilterPopoverWrapper, { "data-component-name": "Filter/FilterPopover" },
        react_1.default.createElement(FilterPopoverHeader, null,
            react_1.default.createElement(FilterPopoverHeaderLabel, { "data-translation-key": "catalog.filters.add" }, translate('catalog.filters.add', 'Add Filter')),
            react_1.default.createElement(FilterPopoverHeaderButton, { onClick: onClose }, translate('catalog.filters.done', 'Done'))),
        react_1.default.createElement(FilterContent_1.FilterContent, { setFilterTerm: setFilterTerm, filters: filters, filterTerm: filterTerm, filterValuesCasing: filterValuesCasing })));
}
const FilterPopoverWrapper = styled_components_1.default.aside `
  position: absolute;
  top: 0;
  z-index: var(--z-index-popover);
  height: 100vh;
  overflow: auto;
  transition: height 0.2s ease-in-out;
  width: 100%;
  background-color: var(--filter-popover-bg-color);
  display: block;

  @media screen and (min-width: ${utils_1.breakpoints.medium}) {
    display: none;
  }
`;
const FilterPopoverHeader = styled_components_1.default.div `
  height: var(--navbar-height);
  padding: var(--filter-popover-header-padding);
  display: grid;
  align-items: center;
  grid-template-columns: 1fr 1fr 1fr;
  border-bottom: 1px solid var(--filter-popover-header-border-color);
  background-color: var(--filter-popover-header-bg-color);
`;
const FilterPopoverHeaderLabel = styled_components_1.default.div `
  color: var(--filter-popover-header-label-color);
  justify-self: center;
  grid-column-start: 2;
  font-size: var(--filter-popover-header-label-font-size);
  font-weight: var(--filter-popover-header-label-font-weight);
`;
const FilterPopoverHeaderButton = styled_components_1.default.div.attrs({
    'data-testid': 'FilterPopover/DoneButton',
}) `
  color: var(--filter-popover-header-button-color);
  justify-self: end;
  font-size: var(--filter-popover-header-button-font-size);
  font-weight: var(--filter-popover-header-button-font-weight);
  height: var(--filter-popover-header-button-height);
  line-height: var(--filter-popover-header-button-height);
`;
