// @redocly/openapi-docs/lib/components/common/Select/styled.js
"use strict";var __createBinding=this&&this.__createBinding||(Object.create?function(e,o,t,r){void 0===r&&(r=t);var n=Object.getOwnPropertyDescriptor(o,t);n&&!("get"in n?!o.__esModule:n.writable||n.configurable)||(n={enumerable:!0,get:function(){return o[t]}}),Object.defineProperty(e,r,n)}:function(e,o,t,r){void 0===r&&(r=t),e[r]=o[t]}),__setModuleDefault=this&&this.__setModuleDefault||(Object.create?function(e,o){Object.defineProperty(e,"default",{enumerable:!0,value:o})}:function(e,o){e.default=o}),__importStar=this&&this.__importStar||function(e){if(e&&e.__esModule)return e;var o={};if(null!=e)for(var t in e)"default"!==t&&Object.prototype.hasOwnProperty.call(e,t)&&__createBinding(o,e,t);return __setModuleDefault(o,e),o};Object.defineProperty(exports,"__esModule",{value:!0}),exports.Arrow=exports.CollapsingSelect=exports.SimpleSelect=exports.Select=void 0;const styled_components_1=__importStar(require("styled-components")),theme_1=require("@redocly/theme"),Select_1=require("./Select"),darkSelectStyle=styled_components_1.css`
  background-color: var(--panel-samples-dropdown-bg-color);
  border: var(--panel-samples-dropdown-border);
  color: var(--panel-samples-dropdown-color);

  &,
  &:hover,
  &:focus-within {
    box-shadow: none;
  }
`;exports.Select=(0,styled_components_1.default)(Select_1.Select)`
  box-sizing: border-box;
  outline: none;
  display: inline-block;
  vertical-align: bottom;
  position: relative;
  width: ${({fullWidth:e})=>e?"100%":"auto"};
  cursor: pointer;
  text-transform: none;

  label {
    box-sizing: border-box;
    min-width: 90px;
    height: 36px;
    outline: none;
    display: inline-block;
    color: var(--docs-dropdown-text-color);
    border-radius: var(--border-radius-md);
    padding: var(--docs-dropdown-padding);
    vertical-align: bottom;
    width: ${({fullWidth:e})=>e?"100%":"auto"};
    text-transform: none;
    line-height: inherit;
    font-size: var(--docs-dropdown-font-size);
    font-family: inherit;
    text-overflow: ellipsis;
    overflow: hidden;
    white-space: nowrap;

    &,
    &:hover,
    &:focus-within {
      box-shadow: none;
    }

    ${({variant:e})=>"dark"===e?darkSelectStyle:""};
  }

  .dropdown-select {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    opacity: 0;
    border: none;
    appearance: none;
    cursor: pointer;

    color: var(--text-color-secondary);
    line-height: inherit;
    font-size: var(--font-size-base);
    font-family: inherit;
    padding: var(--docs-dropdown-padding);
    ${({variant:e})=>"dark"===e?darkSelectStyle:""};
  }
`,exports.SimpleSelect=(0,styled_components_1.default)(exports.Select)`
  svg {
    top: 60%;
    transform: translateY(-60%);
  }
  label {
    padding: 0 26px 0 4px;
    background: transparent;
    font-size: var(--font-size-base);
    font-weight: var(--font-weight-regular);
    appearance: none;
    height: auto;

    &:hover,
    &:focus-within {
      box-shadow: none;
    }
  }
`,exports.CollapsingSelect=(0,styled_components_1.default)(exports.Select)`
  svg {
    top: 60%;
    transform: translateY(-60%);
  }
  label {
    ${darkSelectStyle};

    border-radius: var(--border-radius-lg);
    margin-left: 5px;
    transition: none;
    min-width: auto;
    background-color: ${({active:e})=>e?"var(--panel-samples-tabs-bg-color-active)":"var(--bg-color-raised)"};
    & {
      &[open],
      &:focus-within,
      &:hover {
        background-color: var(--bg-color-tonal);
      }
    }
  }
`,exports.Arrow=(0,styled_components_1.default)(theme_1.ChevronDownIcon)`
  position: absolute;
  right: 10px;
  top: 10px;
`;
