// @redocly/openapi-docs/lib/components/PanelItem/styled.js
"use strict";var __importDefault=this&&this.__importDefault||function(e){return e&&e.__esModule?e:{default:e}};Object.defineProperty(exports,"__esModule",{value:!0}),exports.StyledCheckmarkIcon=exports.PanelItemDescription=exports.PanelItemsList=exports.ActionsWrap=exports.Title=exports.Header=exports.Item=exports.PanelItemWrap=void 0;const styled_components_1=__importDefault(require("styled-components")),theme_1=require("@redocly/theme");exports.PanelItemWrap=styled_components_1.default.span`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0;
  position: relative;
  width: 100%;
`,exports.Item=styled_components_1.default.span`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  overflow: hidden;
`,exports.Header=styled_components_1.default.span`
  text-align: left;
  color: var(--menu-content-title-color);
  font-size: var(--h6-font-size);
  line-height: var(--line-height-xsm);
`,exports.Title=styled_components_1.default.span`
  margin: 0;
  color: var(--text-color-secondary);
  font-size: var(--font-size-base);
  line-height: var(--line-height-base);
  overflow: hidden;
  text-overflow: ellipsis;
  font-weight: ${({active:e})=>e&&"var(--font-weight-medium)"};
`,exports.ActionsWrap=styled_components_1.default.div`
  display: flex;
  gap: var(--spacing-xs);
`,exports.PanelItemsList=styled_components_1.default.div`
  & > span {
    border-bottom: 1px solid var(--border-color-secondary);
    margin-bottom: var(--spacing-xs);
    padding-bottom: var(--spacing-xs);
  }

  & > span:last-child {
    border-bottom: none;
    margin-bottom: 0;
    padding-bottom: 0;
  }
`,exports.PanelItemDescription=styled_components_1.default.span`
  color: var(--text-color-secondary);
  font-size: var(--font-size-base);
  line-height: var(--line-height-base);
  font-weight: var(--font-weight-medium);
`,exports.StyledCheckmarkIcon=(0,styled_components_1.default)(theme_1.CheckmarkIcon)`
  width: 14px;
  height: 14px;
  position: absolute;
  left: -18px;
  bottom: 3px;
`;
