// @redocly/theme/lib/components/Switch/Switch.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Switch = Switch;
const react_1 = __importDefault(require("react"));
const styled_components_1 = __importStar(require("styled-components"));
function Switch({ value = false, disabled = false, onChange }) {
    const handleClick = () => {
        if (disabled)
            return;
        onChange(!value);
    };
    return (react_1.default.createElement(SwitchWrapper, { onClick: handleClick, role: "switch", selected: value, disabled: disabled },
        react_1.default.createElement(Knob, { selected: value })));
}
const SwitchWrapper = styled_components_1.default.div `
  width: var(--switch-width);
  height: var(--switch-height);
  border-radius: var(--switch-border-radius);
  background-color: ${({ selected, disabled }) => disabled
    ? 'var(--switch-bg-color-disabled)'
    : selected
        ? 'var(--switch-bg-color-selected)'
        : 'var(--switch-bg-color)'};
  display: flex;
  align-items: center;
  padding: var(--switch-padding);
  cursor: ${({ disabled }) => (disabled ? 'not-allowed' : 'pointer')};
  transition: var(--switch-bg-transition);

  ${({ disabled }) => !disabled &&
    (0, styled_components_1.css) `
      &:hover {
        background-color: var(--switch-bg-color-hover);
      }
    `}

  &:active {
    background-color: var(--switch-bg-color-pressed);
  }
`;
const Knob = styled_components_1.default.div `
  width: var(--switch-knob-width);
  height: var(--switch-knob-height);
  border-radius: var(--switch-knob-border-radius);
  background-color: ${({ selected }) => selected ? 'var(--switch-knob-bg-color-selected)' : 'var(--switch-knob-bg-color)'};
  transform: ${({ selected }) => selected ? 'translateX(var(--switch-knob-width))' : 'translateX(0)'};
  transition: var(--switch-knob-transition);
`;
