// @redocly/openapi-docs/lib/components/common/Skeleton/LanguageListSkeleton.js
"use strict";var __importDefault=this&&this.__importDefault||function(e){return e&&e.__esModule?e:{default:e}};Object.defineProperty(exports,"__esModule",{value:!0}),exports.LanguageListSkeleton=void 0;const jsx_runtime_1=require("react/jsx-runtime"),react_1=require("react"),styled_components_1=__importDefault(require("styled-components")),Container=styled_components_1.default.div`
  width: 100%;
  display: flex;
  justify-content: space-between;
  gap: var(--spacing-xs);
  padding: var(--spacing-xs) 0;
`,Item=styled_components_1.default.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  justify-content: center;
  align-items: center;
  gap: var(--spacing-sm);
`,SkeletonBox=styled_components_1.default.div`
  display: inline-block;
  height: ${({type:e})=>"icon"===e?"var(--control-height-sm)":"calc(var(--control-height-sm) * 0.5)"};
  position: relative;
  overflow: hidden;
  background-color: var(--layer-color-hover);
  border-radius: var(--border-radius);
  width: ${({type:e})=>"icon"===e?"var(--control-height-sm)":"64px"};
`;function LanguageListSkeletonComponent(){return(0,jsx_runtime_1.jsxs)(Container,{children:[(0,jsx_runtime_1.jsxs)(Item,{children:[(0,jsx_runtime_1.jsx)(SkeletonBox,{type:"icon"}),(0,jsx_runtime_1.jsx)(SkeletonBox,{type:"label"})]}),(0,jsx_runtime_1.jsxs)(Item,{children:[(0,jsx_runtime_1.jsx)(SkeletonBox,{type:"icon"}),(0,jsx_runtime_1.jsx)(SkeletonBox,{type:"label"})]}),(0,jsx_runtime_1.jsxs)(Item,{children:[(0,jsx_runtime_1.jsx)(SkeletonBox,{type:"icon"}),(0,jsx_runtime_1.jsx)(SkeletonBox,{type:"label"})]}),(0,jsx_runtime_1.jsxs)(Item,{children:[(0,jsx_runtime_1.jsx)(SkeletonBox,{type:"icon"}),(0,jsx_runtime_1.jsx)(SkeletonBox,{type:"label"})]})]})}exports.LanguageListSkeleton=(0,react_1.memo)(LanguageListSkeletonComponent);
