// @redocly/theme/src/markdoc/tags/code-snippet.ts
import markdoc from '@markdoc/markdoc';

import type { MarkdocSchemaWrapper } from '@redocly/theme/markdoc/tags/types';

import { CodeSnippetFile } from '@redocly/theme/markdoc/attributes/code-snippet-file';

export const codeSnippet: MarkdocSchemaWrapper = {
  schema: {
    attributes: {
      file: { type: CodeSnippetFile, render: false, required: true },
      language: { type: String, render: true },
      title: { type: String, render: true },
      to: { type: [String, Number], render: false },
      from: { type: [String, Number], render: false },
      after: { type: [String, Number], render: false },
      before: { type: [String, Number], render: false },
      prefix: { type: String, render: false },
      wrap: { type: Boolean, render: false },
      rawContent: { type: String, render: false }, // internal
    },
    render: 'CodeBlock',
    selfClosing: true,
    transform: (node, config) => {
      const attributes = node.transformAttributes(config);
      const rawContent = node.attributes.rawContent;
      const header = {
        title: node.attributes.title,
        controls: { copy: {} },
      };
      if (typeof rawContent === 'string') {
        return new markdoc.Tag(
          'CodeBlock',
          {
            ...attributes,
            header,
            lang: attributes.language,
            source: rawContent,
            wrapContents: node.attributes.wrap,
          },
          [],
        );
      } else {
        return new markdoc.Tag(node.tag, attributes, []);
      }
    },
    validate(node) {
      const errors = [];
      if (node.attributes.from && node.attributes.after) {
        errors.push({
          id: 'from-and-after-attributes-are-mutually-exclusive',
          level: 'error' as const,
          message: 'attributes "from" and "after" are mutually exclusive',
        });
      }
      if (node.attributes.to && node.attributes.before) {
        errors.push({
          id: 'to-and-before-attributes-are-mutually-exclusive',
          level: 'error' as const,
          message: 'attributes "to" and "before" are mutually exclusive',
        });
      }
      return errors;
    },
  },
  tagName: 'code-snippet',
};
