// @redocly/theme/src/markdoc/tags/partial.ts
import markdoc from '@markdoc/markdoc';

import type { RedoclyConfig } from '@redocly/theme/config';
import type { Config } from '@markdoc/markdoc';
import type { MarkdocSchemaWrapper } from '@redocly/theme/markdoc/tags/types';

import { addTrailingSlash } from '@redocly/theme/core/utils';
import { RelativePath } from '@redocly/theme/markdoc/attributes/relative-path';

export const partial: MarkdocSchemaWrapper = {
  schema: {
    ...markdoc.tags.partial,
    attributes: {
      ...markdoc.tags.partial.attributes,
      file: { type: RelativePath },
    },
    validate(node, config: Config & { themeConfig: RedoclyConfig['markdown'] }) {
      const folders = (config.themeConfig?.partialsFolders || []).map(addTrailingSlash);
      if (!folders.length) {
        return [
          {
            id: '',
            message: `Could not resolve partial ${node.attributes.file}. markdown.partialsFolders is empty.`,
            level: 'error',
          },
        ];
      }

      if (!config.partials?.[node.attributes.file]) {
        return [
          {
            id: '',
            message: `Could not resolve partial ${
              node.attributes.file
            }. Starting with 0.73.0 version all partials should be placed in partials folders: \n\n  - ${folders.join(
              '\n  - ',
            )}`,
            level: 'error',
          },
        ];
      }

      return [];
    },
  },
  tagName: 'partial',
};
