// @redocly/realm/dist/server/esbuild/cache/server/chunks/chunk-AB7LF64Z.js
import { createRequire as topLevelCreateRequire } from 'module';
 const require = topLevelCreateRequire(import.meta.url);
import {
  Button,
  Link,
  init_Button,
  init_Link
} from "./chunk-SX3NFZJN.js";
import {
  DEFAULT_FEEDBACK_TYPE,
  Ne,
  breakpoints,
  getCssColorVariable,
  getDisplayedHeadings,
  getDisplayedHeadingsIds,
  getLeastDepth,
  init_constants,
  init_hooks,
  init_styled_components_esm,
  init_theme_helpers,
  init_utils,
  styled_components_esm_default,
  typography,
  useActiveHeading,
  useFullHeight,
  useOutsideClick,
  useThemeConfig,
  useThemeHooks
} from "./chunk-WNBJGIID.js";
import {
  __esm
} from "./chunk-35AYGKFE.js";

// node_modules/@redocly/theme/src/components/Markdown/styles/baseTable.ts
var markdownBaseTableCss;
var init_baseTable = __esm({
  "node_modules/@redocly/theme/src/components/Markdown/styles/baseTable.ts"() {
    init_styled_components_esm();
    markdownBaseTableCss = Ne([".md-table-wrapper{overflow-x:auto;}table.md{width:100%;overflow:hidden;word-break:keep-all;border-spacing:0;margin:var(--md-table-margin-vertical) 0;border-radius:var(--md-table-border-radius);border:1px solid var(--md-table-border-color);th,td{padding:var(--md-table-cell-padding);color:var(--md-table-cell-text-color);font-size:var(--md-table-font-size);line-height:var(--md-table-line-height);vertical-align:top;}thead td,thead th,tbody td{border-bottom:1px solid var(--md-table-border-color);}tbody tr:last-child td{border-bottom:none;}tfoot td{border-top:1px solid var(--md-table-border-color);padding:var(--md-table-footer-cell-padding);}thead,tfoot{background-color:var(--md-table-header-bg-color);}tr{background-color:var(--md-table-stripe-bg-color);}th{text-align:left;font-weight:var(--md-table-head-font-weight);color:var(--md-table-head-text-color);}th[align='center']{text-align:center;}th[align='right']{text-align:right;}}"]);
  }
});

// node_modules/@redocly/theme/src/components/Markdown/styles/links.ts
var markdownLinksCss;
var init_links = __esm({
  "node_modules/@redocly/theme/src/components/Markdown/styles/links.ts"() {
    init_styled_components_esm();
    markdownLinksCss = Ne(["a:not([role='button']){text-decoration:var(--link-decoration);color:var(--link-color-primary);font-weight:var(--link-font-weight);&:visited{color:var(--link-color-visited);text-decoration:var(--link-visited-decoration);}&:hover,&:focus{text-decoration:var(--link-decoration-hover);color:var(--link-color-primary-hover);}}"]);
  }
});

// node_modules/@redocly/theme/src/components/Markdown/styles/headingAnchor.ts
function headingAnchorCss(className = "anchor") {
  return Ne([".", ".before{position:absolute;left:0;transform:translateX(-100%);padding-right:var(--heading-anchor-offset-right);}.", ".after{display:inline-block;padding-left:4px;}svg,.", " svg{fill:var(--heading-anchor-color);visibility:hidden;background-image:var(--heading-anchor-icon);background-size:contain;background-repeat:no-repeat;background-position:center;}:hover .", " svg,.", ":focus svg{visibility:visible;}"], className, className, className, className, className);
}
var init_headingAnchor = __esm({
  "node_modules/@redocly/theme/src/components/Markdown/styles/headingAnchor.ts"() {
    init_styled_components_esm();
  }
});

// node_modules/@redocly/theme/src/components/Markdown/Markdown.tsx
var Markdown;
var init_Markdown = __esm({
  "node_modules/@redocly/theme/src/components/Markdown/Markdown.tsx"() {
    init_styled_components_esm();
    init_theme_helpers();
    init_baseTable();
    init_links();
    init_headingAnchor();
    Markdown = styled_components_esm_default.main.attrs(({
      className
    }) => ({
      "data-component-name": "Markdown/Markdown",
      className
    })).withConfig({
      displayName: "Markdown",
      componentId: "sc-zvbgeq-0"
    })(["font-size:var(--md-content-font-size);font-family:var(--md-content-font-family);font-weight:var(--md-content-font-weight);line-height:var(--md-content-line-height);color:var(--md-content-text-color);padding:0;", " ", " ", " ", " div[id],a[id],a[name]{scroll-margin-top:var(--navbar-height);}img,iframe{max-width:100%;}h1.md,h2.md,h3.md,h4.md,h5.md,h6.md{font-weight:var(--heading-font-weight);font-family:var(--heading-font-family);position:relative;scroll-margin-top:var(--navbar-height);}strong{font-weight:var(--font-weight-bold);}hr{border-top:1px solid var(--md-hr-border-color);background-color:var(--md-hr-bg-color);height:var(--md-hr-height);margin:var(--md-hr-margin-vertical) 0;}details{margin:1.125em 0;summary{color:var(--text-color-secondary);font-weight:var(--font-weight-regular);outline:0;padding:0;cursor:pointer;}> :not(summary):first-of-type{margin-top:1.25em;}> *:last-child{margin-bottom:1.25em;}> :not(summary){margin-left:2em;}}", ";p{margin:var(--md-paragraph-margin);}h1.md{", ";margin:var(--h1-margin-top) 0 var(--h1-margin-bottom) 0;", ";}h2.md{", ";margin:var(--h2-margin-top) 0 var(--h2-margin-bottom) 0;", ";}h3.md{", ";margin:var(--h3-margin-top) 0 var(--h3-margin-bottom) 0;", ";}h4.md{", ";margin:var(--h4-margin-top) 0 var(--h4-margin-bottom) 0;", ";}h5.md{", ";margin:var(--h5-margin-top) 0 var(--h5-margin-bottom) 0;", ";}h6.md{", ";margin:var(--h6-margin-top) 0 var(--h6-margin-bottom) 0;", ";}h1.md code{font-size:var(--h1-font-size);line-height:var(--h1-code-line-height);}h2.md code{font-size:var(--h2-font-size);line-height:var(--h2-code-line-height);}h3.md code{font-size:var(--h3-font-size);line-height:var(--h3-code-line-height);}h4.md code{font-size:var(--h4-font-size);line-height:var(--h4-code-line-height);}h5.md code{font-size:var(--h5-font-size);line-height:var(--h5-code-line-height);}h6.md code{font-size:var(--h6-font-size);line-height:var(--h6-code-line-height);}code{color:var(--inline-code-text-color);background-color:var(--inline-code-bg-color);border-radius:var(--inline-code-border-radius);padding:0.19rem var(--spacing-xxs);font-weight:var(--code-font-weight);word-break:break-word;font-size:var(--font-size-sm);font-family:var(--font-family-monospaced);line-height:var(--line-height-sm);}&& pre{white-space:var(--code-wrap,pre);background-color:var(--code-block-bg-color);font-family:var(--code-block-font-family);color:var(--code-block-text-color);padding:var(--code-block-padding);overflow-x:auto;line-height:var(--line-height-base);margin:var(--md-pre-margin);code{background-color:transparent;color:inherit;font-size:inherit;font-family:inherit;border:none;}}blockquote{margin:var(--md-blockquote-margin-vertical) var(--md-blockquote-margin-horizontal);padding:var(--md-blockquote-padding-vertical) var(--md-blockquote-padding-horizontal);border-left:var(--md-blockquote-border-left);background-color:var(--md-blockquote-bg-color);color:var(--md-blockquote-text-color);box-shadow:var(--md-blockquote-box-shadow);& > p:first-child{margin-top:0;}}img{max-width:100%;box-sizing:content-box;margin:0 auto;&:only-child{display:block;}}p,li{& > img:only-child{display:inline-block;}}ul,ol{padding-left:var(--md-list-left-padding);margin:var(--md-list-margin);ul,ol{margin-bottom:0;margin-top:0;}}li{list-style:var(--md-list-item-style);&:not(:last-child){margin-bottom:calc(var(--spacing-xxs) / 2);}&::marker{padding-left:0.25rem;}}ul > li:has(input[type='checkbox']){list-style:var(--md-checkbox-list-item-style);}ol > li{list-style:var(--md-numbered-list-item-style);}ol > li:before{content:counter(list-item);display:var(--md-numbered-list-number-display);color:var(--md-numbered-list-number-text-color);font-size:var(--md-numbered-list-number-font-size);font-family:var(--md-numbered-list-number-font-family);font-weight:var(--md-numbered-list-number-font-weight);background-color:var(--md-numbered-list-number-bg-color);border-radius:var(--md-numbered-list-number-border-radius);border:var(--md-numbered-list-number-border);margin-right:var(--md-numbered-list-number-margin-right);padding:var(--md-numbered-list-number-padding);}"], ({
      compact
    }) => compact && Ne(["p:first-child{margin-top:0;}p:last-child{margin-bottom:0;}"]), ({
      base
    }) => base && Ne(["p{font-size:var(--md-content-font-size);}"]), ({
      inline
    }) => inline && Ne(["p{display:inline-block;}"]), markdownLinksCss, markdownBaseTableCss, typography("h1"), headingAnchorCss(), typography("h2"), headingAnchorCss(), typography("h3"), headingAnchorCss(), typography("h4"), headingAnchorCss(), typography("h5"), headingAnchorCss(), typography("h6"), headingAnchorCss());
  }
});

// node_modules/@redocly/theme/src/components/TableOfContent/TableOfContent.tsx
import * as React from "react";
function TableOfContent(props) {
  const {
    headings,
    contentWrapper,
    className
  } = props;
  const {
    useTranslate,
    useTelemetry
  } = useThemeHooks();
  const {
    translate
  } = useTranslate();
  const telemetry = useTelemetry();
  const sidebar = React.useRef(null);
  useFullHeight(sidebar);
  const {
    markdown: {
      toc = {}
    } = {}
  } = useThemeConfig();
  const displayedHeadings = getDisplayedHeadings(headings, toc.depth || 3);
  const leastDepth = getLeastDepth(displayedHeadings);
  const activeHeadingId = useActiveHeading(contentWrapper, getDisplayedHeadingsIds(displayedHeadings));
  if (toc?.hide) {
    return null;
  }
  return /* @__PURE__ */ React.createElement(React.Fragment, null, /* @__PURE__ */ React.createElement(TableOfContentMenu, { "data-component-name": "TableOfContent/TableOfContent", className }, /* @__PURE__ */ React.createElement(TableOfContentItems, { ref: sidebar }, displayedHeadings.length ? /* @__PURE__ */ React.createElement(TableOfContentHeader, { "data-translation-key": "toc.header" }, translate("toc.header", toc.header || "On this page")) : null, displayedHeadings.map((heading, idx) => {
    if (!heading) {
      return null;
    }
    const href = "#" + heading.id;
    return /* @__PURE__ */ React.createElement(TableOfContentMenuItem, { key: href + idx, href, depth: heading.depth - leastDepth + 1 || 0, className: activeHeadingId === heading.id ? "active" : "", dangerouslySetInnerHTML: {
      __html: heading.value || ""
    }, "data-testid": `toc-${heading.value}`, onClick: () => telemetry.send("toc_item_clicked", {}) });
  }))));
}
var TableOfContentMenu, TableOfContentHeader, TableOfContentItems, TableOfContentMenuItem;
var init_TableOfContent = __esm({
  "node_modules/@redocly/theme/src/components/TableOfContent/TableOfContent.tsx"() {
    init_styled_components_esm();
    init_hooks();
    init_utils();
    TableOfContentMenu = styled_components_esm_default.aside.withConfig({
      displayName: "TableOfContent__TableOfContentMenu",
      componentId: "sc-kstbuq-0"
    })(["background-color:var(--toc-bg-color);flex-shrink:0;display:none !important;@media screen and (min-width:", "){width:var(--toc-width);display:block !important;}"], breakpoints.medium);
    TableOfContentHeader = styled_components_esm_default.div.withConfig({
      displayName: "TableOfContent__TableOfContentHeader",
      componentId: "sc-kstbuq-1"
    })(["font-family:var(--toc-heading-font-family);font-size:var(--toc-heading-font-size);font-weight:var(--toc-heading-font-weight);padding:var(--toc-item-padding-vertical) var(--toc-item-padding-horizontal);color:var(--toc-heading-text-color);text-transform:var(--toc-heading-text-transform);line-height:var(--toc-heading-line-height);"]);
    TableOfContentItems = styled_components_esm_default.div.withConfig({
      displayName: "TableOfContent__TableOfContentItems",
      componentId: "sc-kstbuq-2"
    })(["position:sticky;top:calc(var(--navbar-height) + var(--toc-offset-top));max-height:calc(100vh - var(--navbar-height) - var(--toc-offset-top));overflow-y:auto;width:var(--toc-width);border-left:solid 1px var(--toc-border-color);"]);
    TableOfContentMenuItem = styled_components_esm_default.a.withConfig({
      displayName: "TableOfContent__TableOfContentMenuItem",
      componentId: "sc-kstbuq-3"
    })(["display:block;color:var(--toc-item-text-color);font-weight:var(--toc-item-font-weight);background-color:var(--toc-item-bg-color);cursor:pointer;font-size:var(--toc-item-font-size);padding:var(--toc-item-padding-vertical) var(--toc-item-padding-horizontal);padding-left:calc(var(--toc-item-nested-offset) * ", ");transition:background-color 0.3s,color 0.3s;text-decoration:none;word-break:break-word;font-family:var(--heading-font-family,var(--toc-item-font-family));line-height:var(--toc-item-line-height);border-left:solid 2px transparent;:hover{color:var(--toc-item-text-color-active);}&.active{color:var(--toc-item-text-color-active);border-left:solid 2px var(--toc-item-border-color-active);}:empty{padding:0;}"], ({
      depth
    }) => depth);
  }
});

// node_modules/@redocly/theme/src/icons/EditIcon/EditIcon.tsx
import React2 from "react";
var Icon, EditIcon;
var init_EditIcon = __esm({
  "node_modules/@redocly/theme/src/icons/EditIcon/EditIcon.tsx"() {
    init_styled_components_esm();
    init_utils();
    Icon = (props) => /* @__PURE__ */ React2.createElement("svg", { viewBox: "0 0 16 16", fill: "none", xmlns: "http://www.w3.org/2000/svg", ...props }, /* @__PURE__ */ React2.createElement("path", { d: "M15 13H1V14H15V13Z" }), /* @__PURE__ */ React2.createElement("path", { d: "M12.7 4.5C13.1 4.1 13.1 3.5 12.7 3.1L10.9 1.3C10.5 0.9 9.9 0.9 9.5 1.3L2 8.8V12H5.2L12.7 4.5ZM10.2 2L12 3.8L10.5 5.3L8.7 3.5L10.2 2ZM3 11V9.2L8 4.2L9.8 6L4.8 11H3Z" }));
    EditIcon = styled_components_esm_default(Icon).attrs(() => ({
      "data-component-name": "icons/EditIcon/EditIcon"
    })).withConfig({
      displayName: "EditIcon",
      componentId: "sc-1v0u04j-0"
    })(["path{fill:", ";}height:", ";width:", ";"], ({
      color
    }) => getCssColorVariable(color), ({
      size
    }) => size || "16px", ({
      size
    }) => size || "16px");
  }
});

// node_modules/@redocly/theme/src/icons/RadioCheckButtonIcon/RadioCheckButtonIcon.tsx
import React3 from "react";
var Icon2, RadioCheckButtonIcon;
var init_RadioCheckButtonIcon = __esm({
  "node_modules/@redocly/theme/src/icons/RadioCheckButtonIcon/RadioCheckButtonIcon.tsx"() {
    init_styled_components_esm();
    Icon2 = () => /* @__PURE__ */ React3.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: "16", height: "16", viewBox: "0 0 16 16", fill: "none" }, /* @__PURE__ */ React3.createElement("rect", { width: "16", height: "16", fill: "none" }), /* @__PURE__ */ React3.createElement("path", { d: "M8 1C6.61553 1 5.26216 1.41054 4.11101 2.17971C2.95987 2.94888 2.06266 4.04213 1.53285 5.32122C1.00303 6.6003 0.86441 8.00776 1.13451 9.36563C1.4046 10.7235 2.07129 11.9708 3.05026 12.9497C4.02922 13.9287 5.2765 14.5954 6.63437 14.8655C7.99224 15.1356 9.3997 14.997 10.6788 14.4672C11.9579 13.9373 13.0511 13.0401 13.8203 11.889C14.5895 10.7378 15 9.38447 15 8C15 6.14348 14.2625 4.36301 12.9497 3.05025C11.637 1.7375 9.85652 1 8 1ZM8 14C6.81332 14 5.65328 13.6481 4.66658 12.9888C3.67989 12.3295 2.91085 11.3925 2.45673 10.2961C2.0026 9.19974 1.88378 7.99334 2.11529 6.82946C2.3468 5.66557 2.91825 4.59647 3.75736 3.75736C4.59648 2.91824 5.66558 2.3468 6.82946 2.11529C7.99335 1.88378 9.19975 2.0026 10.2961 2.45672C11.3925 2.91085 12.3295 3.67988 12.9888 4.66658C13.6481 5.65327 14 6.81331 14 8C14 9.5913 13.3679 11.1174 12.2426 12.2426C11.1174 13.3679 9.5913 14 8 14Z", fill: "currentColor" }), /* @__PURE__ */ React3.createElement("path", { d: "M7 10.625L4.5 8.145L5.295 7.375L7 9.05L10.705 5.375L11.5 6.165L7 10.625Z", fill: "currentColor" }));
    RadioCheckButtonIcon = styled_components_esm_default(Icon2).attrs(() => ({
      "data-component-name": "icons/RadioCheckButtonIcon/RadioCheckButtonIcon"
    })).withConfig({
      displayName: "RadioCheckButtonIcon",
      componentId: "sc-qjq9kc-0"
    })([""]);
  }
});

// node_modules/@redocly/theme/src/components/Feedback/Comment.tsx
import React4, { useRef as useRef2 } from "react";
function Comment({
  settings,
  onSubmit,
  onCancel,
  className,
  standAlone = true,
  isDialog = false
}) {
  const {
    useTranslate
  } = useThemeHooks();
  const {
    translate
  } = useTranslate();
  const {
    label,
    submitText
  } = settings || {};
  const [text, setText] = React4.useState("");
  const [submitValue, setSubmitValue] = React4.useState("");
  const modalRef = useRef2(null);
  useOutsideClick(modalRef, onCancel);
  const send = () => {
    if (!text)
      return;
    setSubmitValue(text);
    onSubmit({
      comment: text
    });
  };
  const handleTextAreaChange = (e) => {
    const commentValue = e.target.value;
    setText(commentValue);
    if (!standAlone) {
      onSubmit({
        comment: commentValue
      });
    }
  };
  if (submitValue) {
    return /* @__PURE__ */ React4.createElement(StateWrapper, { "data-component-name": "Feedback/Comment", className }, /* @__PURE__ */ React4.createElement(StyledFormMandatoryFields, null, /* @__PURE__ */ React4.createElement(Label, { "data-translation-key": "feedback.settings.comment.submitText", standAlone }, submitText || translate("feedback.settings.comment.submitText", "Thank you for helping improve our documentation!")), /* @__PURE__ */ React4.createElement(RadioCheckButtonIcon, null)));
  }
  return /* @__PURE__ */ React4.createElement(CommentWrapper, { ref: modalRef, "data-component-name": "Feedback/Comment", className, style: standAlone ? {
    width: "var(--feedback-report-dialog-width)"
  } : {
    width: "auto"
  } }, /* @__PURE__ */ React4.createElement(Label, { "data-translation-key": "feedback.settings.comment.label", standAlone }, label || translate("feedback.settings.comment.label", "Please share your feedback with us.")), /* @__PURE__ */ React4.createElement(TextArea, { rows: 3, onChange: handleTextAreaChange }), standAlone && /* @__PURE__ */ React4.createElement(ButtonsContainer, null, onCancel && /* @__PURE__ */ React4.createElement(Button, { "data-translation-key": "feedback.settings.comment.cancel", onClick: onCancel, variant: "text", size: "small" }, translate("feedback.settings.comment.cancel", "Cancel")), /* @__PURE__ */ React4.createElement(Button, { "data-translation-key": "feedback.settings.comment.send", onClick: send, variant: isDialog ? "primary" : "secondary", size: "small" }, translate("feedback.settings.comment.send", "Send"))));
}
var StyledFormMandatoryFields, CommentWrapper, StateWrapper, Label, TextArea, ButtonsContainer;
var init_Comment = __esm({
  "node_modules/@redocly/theme/src/components/Feedback/Comment.tsx"() {
    init_styled_components_esm();
    init_hooks();
    init_RadioCheckButtonIcon();
    init_Button();
    StyledFormMandatoryFields = styled_components_esm_default.div.withConfig({
      displayName: "Comment__StyledFormMandatoryFields",
      componentId: "sc-lbdj57-0"
    })(["display:flex;justify-content:space-between;align-items:center;gap:var(--spacing-xs);> svg{margin:var(--button-icon-padding) 0;}"]);
    CommentWrapper = styled_components_esm_default.div.withConfig({
      displayName: "Comment__CommentWrapper",
      componentId: "sc-lbdj57-1"
    })(["font-family:var(--font-family-base);display:flex;flex-direction:column;justify-content:space-between;border-radius:var(--border-radius-lg);background:var(--feedback-bg-color);gap:var(--spacing-xxs);max-width:var(--feedback-report-dialog-max-width);"]);
    StateWrapper = styled_components_esm_default(CommentWrapper).withConfig({
      displayName: "Comment__StateWrapper",
      componentId: "sc-lbdj57-2"
    })(["flex-direction:row;align-items:center;gap:var(--spacing-xxs);"]);
    Label = styled_components_esm_default.h4.withConfig({
      displayName: "Comment__Label",
      componentId: "sc-lbdj57-3"
    })(["font-weight:var(--font-weight-regular);font-size:var(--feedback-font-size);line-height:var(--feedback-line-height);", " margin:0;"], ({
      standAlone
    }) => standAlone && Ne(["font-size:var(--feedback-header-font-size);line-height:var(--feedback-header-line-height);"]));
    TextArea = styled_components_esm_default.textarea.withConfig({
      displayName: "Comment__TextArea",
      componentId: "sc-lbdj57-4"
    })(["background-color:var(--bg-color);border-radius:var(--border-radius-lg);border:var(--input-border);outline:none;color:var(--feedback-text-color);font-family:var(--feedback-font-family);padding:10px;min-height:calc(var(--line-height-base) * 3 + 20px);min-width:var(--feedback-comment-min-width);max-width:var(--feedback-comment-max-width);resize:vertical;"]);
    ButtonsContainer = styled_components_esm_default.div.withConfig({
      displayName: "Comment__ButtonsContainer",
      componentId: "sc-lbdj57-5"
    })(["display:flex;justify-content:end;margin-bottom:var(--spacing-xxs);margin-top:var(--spacing-xs);gap:var(--spacing-xs);"]);
  }
});

// node_modules/@redocly/theme/src/components/Feedback/Reasons.tsx
import * as React5 from "react";
function Reasons({
  settings,
  onChange,
  className
}) {
  const {
    label,
    component,
    items = []
  } = settings;
  const [checkedState, setCheckedState] = React5.useState(new Array(items.length).fill(false));
  const {
    useTranslate
  } = useThemeHooks();
  const {
    translate
  } = useTranslate();
  if (!items.length) {
    return null;
  }
  const input_type = component || "checkbox";
  const handleOptionChange = (position) => {
    const updatedCheckedState = component === "checkbox" ? checkedState.map((item, index) => index === position ? !item : item) : items.map((_, idx) => position === idx);
    setCheckedState(updatedCheckedState);
    onChange(items.filter((_, index) => !!updatedCheckedState[index]));
  };
  return /* @__PURE__ */ React5.createElement(ReasonsWrapper, { "data-component-name": "Feedback/Reasons", "data-translation-key": "feedback.settings.reasons.label", className }, /* @__PURE__ */ React5.createElement(Label2, null, label || translate("feedback.settings.reasons.label", "Which statement describes your thoughts about this page?")), items.map((reason, idx) => /* @__PURE__ */ React5.createElement(OptionWrapper, { key: reason }, /* @__PURE__ */ React5.createElement("input", { type: input_type, value: reason, checked: checkedState[idx], name: "reasons", onChange: () => handleOptionChange(idx) }), /* @__PURE__ */ React5.createElement("label", { "data-translation-key": `feedback.settings.reasons.items.${idx + 1}`, id: reason, onClick: () => handleOptionChange(idx) }, translate(`feedback.settings.reasons.items.${idx + 1}`, reason)))));
}
var ReasonsWrapper, Label2, OptionWrapper;
var init_Reasons = __esm({
  "node_modules/@redocly/theme/src/components/Feedback/Reasons.tsx"() {
    init_styled_components_esm();
    init_hooks();
    ReasonsWrapper = styled_components_esm_default.div.withConfig({
      displayName: "Reasons__ReasonsWrapper",
      componentId: "sc-clbfy6-0"
    })(["font-family:var(--feedback-font-family);display:flex;flex-direction:column;"]);
    Label2 = styled_components_esm_default.h4.withConfig({
      displayName: "Reasons__Label",
      componentId: "sc-clbfy6-1"
    })(["font-family:var(--feedback-font-family);font-weight:var(--font-weight-regular);font-size:var(--feedback-font-size);line-height:var(--feedback-header-line-height);color:var(--feedback-header-text-color);margin:0;"]);
    OptionWrapper = styled_components_esm_default.div.withConfig({
      displayName: "Reasons__OptionWrapper",
      componentId: "sc-clbfy6-2"
    })(["margin:5px 0;display:flex;input{margin-right:10px;cursor:pointer;}label{font-size:var(--feedback-font-size);cursor:pointer;}"]);
  }
});

// node_modules/@redocly/theme/src/icons/StarIcon/StarIcon.tsx
import React6 from "react";
var Icon3, StarIcon;
var init_StarIcon = __esm({
  "node_modules/@redocly/theme/src/icons/StarIcon/StarIcon.tsx"() {
    init_styled_components_esm();
    init_utils();
    Icon3 = (props) => /* @__PURE__ */ React6.createElement("svg", { viewBox: "0 0 16 16", fill: "none", xmlns: "http://www.w3.org/2000/svg", ...props }, /* @__PURE__ */ React6.createElement("path", { d: "M8.00001 3.26L9.38001 6.05L9.61001 6.55L10.11 6.625L13.19 7.07L11 9.22L10.625 9.585L10.715 10.085L11.24 13.15L8.48501 11.705L8.00001 11.5L7.53501 11.745L4.78001 13.17L5.28001 10.105L5.37001 9.605L5.00001 9.22L2.79001 7.045L5.87001 6.6L6.37001 6.525L6.60001 6.025L8.00001 3.26ZM8.00001 1L5.72501 5.61L0.640015 6.345L4.32001 9.935L3.45001 15L8.00001 12.61L12.55 15L11.68 9.935L15.36 6.35L10.275 5.61L8.00001 1Z" }));
    StarIcon = styled_components_esm_default(Icon3).attrs(() => ({
      "data-component-name": "icons/StarIcon/StarIcon"
    })).withConfig({
      displayName: "StarIcon",
      componentId: "sc-1ddrqba-0"
    })(["path{fill:", ";}height:", ";width:", ";"], ({
      color
    }) => getCssColorVariable(color), ({
      size
    }) => size || "16px", ({
      size
    }) => size || "16px");
  }
});

// node_modules/@redocly/theme/src/icons/StarFilledIcon/StarFilledIcon.tsx
import React7 from "react";
var Icon4, StarFilledIcon;
var init_StarFilledIcon = __esm({
  "node_modules/@redocly/theme/src/icons/StarFilledIcon/StarFilledIcon.tsx"() {
    init_styled_components_esm();
    init_utils();
    Icon4 = (props) => /* @__PURE__ */ React7.createElement("svg", { viewBox: "0 0 16 16", fill: "none", xmlns: "http://www.w3.org/2000/svg", ...props }, /* @__PURE__ */ React7.createElement("path", { d: "M8.00001 1L5.72501 5.61L0.640015 6.345L4.32001 9.935L3.45001 15L8.00001 12.61L12.55 15L11.68 9.935L15.36 6.35L10.275 5.61L8.00001 1Z" }));
    StarFilledIcon = styled_components_esm_default(Icon4).attrs(() => ({
      "data-component-name": "icons/StarFilledIcon/StarFilledIcon"
    })).withConfig({
      displayName: "StarFilledIcon",
      componentId: "sc-1n1f60x-0"
    })(["path{fill:", ";}height:", ";width:", ";"], ({
      color
    }) => getCssColorVariable(color), ({
      size
    }) => size || "16px", ({
      size
    }) => size || "16px");
  }
});

// node_modules/@redocly/theme/src/components/Feedback/Stars.tsx
import * as React8 from "react";
function Stars({
  max = 5,
  onChange,
  value
}) {
  const [hovered, setHovered] = React8.useState(value || 0);
  const stars = [];
  for (let index = 1; index <= max; index++) {
    stars.push(/* @__PURE__ */ React8.createElement(Star, { "data-testid": `star-${index}`, key: index, onClick: () => onChange(index), onMouseOver: () => setHovered(index), onMouseLeave: () => !value && setHovered(0) }, hovered < index ? /* @__PURE__ */ React8.createElement(StarIcon, { color: "var(--feedback-star-color)" }) : /* @__PURE__ */ React8.createElement(StarFilledIcon, { color: "var(--feedback-star-color)" })));
  }
  return /* @__PURE__ */ React8.createElement(StarsWrapper, { "data-component-name": "Feedback/Stars" }, stars);
}
var StarsWrapper, Star;
var init_Stars = __esm({
  "node_modules/@redocly/theme/src/components/Feedback/Stars.tsx"() {
    init_styled_components_esm();
    init_StarIcon();
    init_StarFilledIcon();
    StarsWrapper = styled_components_esm_default.div.withConfig({
      displayName: "Stars__StarsWrapper",
      componentId: "sc-7enpoj-0"
    })(["display:flex;gap:var(--spacing-xs);flex-direction:row;align-items:center;"]);
    Star = styled_components_esm_default.button.attrs(() => ({
      type: "button"
    })).withConfig({
      displayName: "Stars__Star",
      componentId: "sc-7enpoj-1"
    })(["cursor:pointer;padding:0;border:none;background-color:transparent;height:16px;gap:var(--spacing-xs);"]);
  }
});

// node_modules/@redocly/theme/src/components/Feedback/Rating.tsx
import * as React9 from "react";
import { useEffect } from "react";
function Rating({
  settings,
  onSubmit,
  className
}) {
  const {
    label,
    submitText,
    comment: commentSettings,
    reasons: reasonsSettings
  } = settings || {};
  const [isSubmitted, setIsSubmitted] = React9.useState(false);
  const [score, setScore] = React9.useState(0);
  const [reasons, setReasons] = React9.useState([]);
  const [comment, setComment] = React9.useState("");
  const {
    useTranslate
  } = useThemeHooks();
  const {
    translate
  } = useTranslate();
  const onSubmitRatingForm = () => {
    onSubmit({
      score,
      comment,
      reasons,
      max: FEEDBACK_MAX_RATING
    });
    setIsSubmitted(true);
  };
  const onCancelRatingForm = () => {
    setScore(0);
  };
  const displayReasons = !!(score && reasonsSettings && !reasonsSettings.hide);
  const displayComment = !!(score && !commentSettings?.hide);
  const displaySubmitBnt = !!(score && (displayReasons || displayComment));
  useEffect(() => {
    if (score && !displayComment && !displayReasons) {
      onSubmitRatingForm();
    }
  }, [score, displayComment, displayReasons]);
  if (isSubmitted) {
    return /* @__PURE__ */ React9.createElement(RatingWrapper, { "data-component-name": "Feedback/Rating" }, /* @__PURE__ */ React9.createElement(StyledFormMandatoryFields2, null, /* @__PURE__ */ React9.createElement(Label3, { "data-translation-key": "feedback.settings.submitText" }, submitText || translate("feedback.settings.submitText", "Thank you for helping improve our documentation!")), /* @__PURE__ */ React9.createElement(RadioCheckButtonIcon, null)));
  }
  return /* @__PURE__ */ React9.createElement(RatingWrapper, { "data-component-name": "Feedback/Rating", className }, /* @__PURE__ */ React9.createElement(StyledForm, null, /* @__PURE__ */ React9.createElement(StyledFormMandatoryFields2, null, /* @__PURE__ */ React9.createElement(Label3, { "data-translation-key": "feedback.settings.label" }, label || translate("feedback.settings.label", "Was this helpful?")), /* @__PURE__ */ React9.createElement(StyledMandatoryFieldContainer, null, /* @__PURE__ */ React9.createElement(Stars, { max: FEEDBACK_MAX_RATING, onChange: setScore, value: score }))), (displayReasons || displayComment) && /* @__PURE__ */ React9.createElement(StyledFormOptionalFields, null, displayReasons && /* @__PURE__ */ React9.createElement(Reasons, { settings: {
    label: reasonsSettings?.label,
    items: reasonsSettings?.items || [],
    component: reasonsSettings?.component
  }, onChange: setReasons }), displayComment && /* @__PURE__ */ React9.createElement(Comment, { standAlone: false, onSubmit: ({
    comment: comment2
  }) => setComment(comment2), settings: {
    label: commentSettings?.label || translate("feedback.settings.comment.label", "Please share your feedback with us.")
  } })), displaySubmitBnt && /* @__PURE__ */ React9.createElement(ButtonsContainer2, null, /* @__PURE__ */ React9.createElement(Button, { onClick: onCancelRatingForm, variant: "text", size: "small" }, "Cancel"), /* @__PURE__ */ React9.createElement(Button, { onClick: onSubmitRatingForm, variant: "secondary", size: "small" }, "Submit"))));
}
var FEEDBACK_MAX_RATING, StyledForm, StyledFormOptionalFields, StyledFormMandatoryFields2, StyledMandatoryFieldContainer, RatingWrapper, Label3, ButtonsContainer2;
var init_Rating = __esm({
  "node_modules/@redocly/theme/src/components/Feedback/Rating.tsx"() {
    init_styled_components_esm();
    init_hooks();
    init_RadioCheckButtonIcon();
    init_Comment();
    init_Reasons();
    init_Stars();
    init_Button();
    FEEDBACK_MAX_RATING = 5;
    StyledForm = styled_components_esm_default.form.withConfig({
      displayName: "Rating__StyledForm",
      componentId: "sc-l0bg02-0"
    })(["width:100%;gap:var(--spacing-sm);display:flex;flex-direction:column;"]);
    StyledFormOptionalFields = styled_components_esm_default.div.withConfig({
      displayName: "Rating__StyledFormOptionalFields",
      componentId: "sc-l0bg02-1"
    })(["display:flex;flex-flow:column;"]);
    StyledFormMandatoryFields2 = styled_components_esm_default.div.withConfig({
      displayName: "Rating__StyledFormMandatoryFields",
      componentId: "sc-l0bg02-2"
    })(["display:flex;justify-content:space-between;align-items:center;gap:var(--spacing-xs);"]);
    StyledMandatoryFieldContainer = styled_components_esm_default.div.withConfig({
      displayName: "Rating__StyledMandatoryFieldContainer",
      componentId: "sc-l0bg02-3"
    })(["display:flex;"]);
    RatingWrapper = styled_components_esm_default.div.withConfig({
      displayName: "Rating__RatingWrapper",
      componentId: "sc-l0bg02-4"
    })(["font-family:var(--font-family-base);display:flex;justify-content:space-between;align-items:center;"]);
    Label3 = styled_components_esm_default.h4.withConfig({
      displayName: "Rating__Label",
      componentId: "sc-l0bg02-5"
    })(["font-family:var(--feedback-font-family);font-weight:var(--font-weight-regular);font-size:var(--feedback-header-font-size);line-height:var(--feedback-header-line-height);color:var(--feedback-header-text-color);margin:0;"]);
    ButtonsContainer2 = styled_components_esm_default.div.withConfig({
      displayName: "Rating__ButtonsContainer",
      componentId: "sc-l0bg02-6"
    })(["display:flex;justify-content:end;margin-bottom:var(--spacing-xxs);gap:var(--spacing-xxs);"]);
  }
});

// node_modules/@redocly/theme/src/icons/ThumbDownIcon/ThumbDownIcon.tsx
import React10 from "react";
var Icon5, ThumbDownIcon;
var init_ThumbDownIcon = __esm({
  "node_modules/@redocly/theme/src/icons/ThumbDownIcon/ThumbDownIcon.tsx"() {
    init_styled_components_esm();
    Icon5 = () => /* @__PURE__ */ React10.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: "14", height: "14", viewBox: "0 0 14 14", fill: "none" }, /* @__PURE__ */ React10.createElement("path", { d: "M12.8422 6.66036C12.8984 6.47286 12.9266 6.27911 12.9266 6.08224C12.9266 5.64005 12.7812 5.21505 12.5188 4.86818C12.575 4.68068 12.6031 4.48693 12.6031 4.29005C12.6031 3.84786 12.4578 3.42286 12.1953 3.07599C12.2516 2.88849 12.2797 2.69474 12.2797 2.49786C12.2797 1.69161 11.8 0.965052 11.0562 0.647865C10.9255 0.591503 10.7845 0.562777 10.6422 0.56349H1.25C0.973437 0.56349 0.75 0.786927 0.75 1.06349V6.75099C0.75 7.02755 0.973437 7.25099 1.25 7.25099H3.27031L4.61094 12.1072C4.82656 12.8901 5.54531 13.4369 6.35781 13.4369C6.82188 13.4369 7.25469 13.2526 7.575 12.9151C7.89531 12.5791 8.05937 12.1385 8.03594 11.6744L7.94219 9.75411H11.6906C11.8797 9.75411 12.0641 9.70411 12.2266 9.6088C12.8578 9.24161 13.25 8.57599 13.25 7.87443C13.25 7.43224 13.1047 7.00724 12.8422 6.66036ZM1.875 6.12443V1.68693H3.14062V6.12443H1.875ZM11.675 8.63068H6.7625L6.9125 11.7307C6.92188 11.9166 6.83906 12.0916 6.68437 12.2072C6.58906 12.2776 6.47188 12.3135 6.35469 12.3119C6.20497 12.3105 6.05977 12.2605 5.94087 12.1695C5.82198 12.0785 5.73581 11.9514 5.69531 11.8072L4.14062 6.17443V1.68693H10.6313C10.7874 1.75692 10.9201 1.87054 11.0132 2.01413C11.1063 2.15772 11.156 2.32515 11.1562 2.4963C11.1562 2.64786 11.1203 2.79161 11.0484 2.92286L10.8313 3.31974L11.1734 3.61661C11.2698 3.70003 11.347 3.80322 11.3998 3.91917C11.4527 4.03511 11.4799 4.16107 11.4797 4.28849C11.4797 4.44005 11.4437 4.5838 11.3719 4.71505L11.1547 5.11193L11.4969 5.4088C11.5932 5.49222 11.6704 5.59541 11.7232 5.71135C11.7761 5.82729 11.8033 5.95326 11.8031 6.08068C11.8031 6.23224 11.7672 6.37599 11.6953 6.50724L11.4766 6.90568L11.8188 7.20255C11.9151 7.28597 11.9923 7.38916 12.0451 7.5051C12.098 7.62104 12.1252 7.74701 12.125 7.87443C12.125 8.17286 11.9531 8.46036 11.675 8.63068Z", fill: "currentColor" }));
    ThumbDownIcon = styled_components_esm_default(Icon5).attrs(() => ({
      "data-component-name": "icons/ThumbDownIcon/ThumbDownIcon"
    })).withConfig({
      displayName: "ThumbDownIcon",
      componentId: "sc-4zvi9h-0"
    })([""]);
  }
});

// node_modules/@redocly/theme/src/icons/ThumbUpIcon/ThumbUpIcon.tsx
import React11 from "react";
var Icon6, ThumbUpIcon;
var init_ThumbUpIcon = __esm({
  "node_modules/@redocly/theme/src/icons/ThumbUpIcon/ThumbUpIcon.tsx"() {
    init_styled_components_esm();
    Icon6 = () => /* @__PURE__ */ React11.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: "14", height: "14", viewBox: "0 0 14 14", fill: "none" }, /* @__PURE__ */ React11.createElement("path", { d: "M12.8422 7.33966C13.1047 6.99279 13.25 6.56779 13.25 6.1256C13.25 5.42404 12.8578 4.75997 12.2266 4.38966C12.0641 4.29434 11.879 4.24417 11.6906 4.24435H7.94375L8.0375 2.32404C8.05938 1.85997 7.89531 1.41935 7.57656 1.08341C7.42013 0.917829 7.23141 0.786087 7.02206 0.696324C6.8127 0.606562 6.58716 0.560687 6.35938 0.561535C5.54688 0.561535 4.82812 1.10841 4.6125 1.89122L3.27031 6.7506H1.25C0.973437 6.7506 0.75 6.97404 0.75 7.2506V12.9381C0.75 13.2147 0.973437 13.4381 1.25 13.4381H10.6453C10.7891 13.4381 10.9297 13.41 11.0594 13.3537C11.8031 13.0365 12.2828 12.31 12.2828 11.5037C12.2828 11.3068 12.2547 11.1131 12.1984 10.9256C12.4609 10.5787 12.6062 10.1537 12.6062 9.71154C12.6062 9.51466 12.5781 9.32091 12.5219 9.13341C12.7844 8.78654 12.9297 8.36154 12.9297 7.91935C12.9266 7.72247 12.8984 7.52716 12.8422 7.33966ZM1.875 12.3131V7.8756H3.14062V12.3131H1.875ZM11.8188 6.79747L11.4766 7.09435L11.6938 7.49122C11.7653 7.62196 11.8024 7.76875 11.8016 7.91779C11.8016 8.1756 11.6891 8.42091 11.4953 8.58966L11.1531 8.88654L11.3703 9.28341C11.4419 9.41414 11.479 9.56094 11.4781 9.70997C11.4781 9.96779 11.3656 10.2131 11.1719 10.3818L10.8297 10.6787L11.0469 11.0756C11.1184 11.2063 11.1555 11.3531 11.1547 11.5022C11.1547 11.8522 10.9484 12.1678 10.6297 12.3115H4.14062V7.8256L5.69531 2.19279C5.7354 2.04841 5.82146 1.92104 5.94044 1.82997C6.05943 1.7389 6.20485 1.68909 6.35469 1.6881C6.47344 1.6881 6.59062 1.72247 6.68437 1.79279C6.83906 1.90841 6.92188 2.08341 6.9125 2.26935L6.7625 5.36935H11.675C11.9531 5.53966 12.125 5.82716 12.125 6.1256C12.125 6.38341 12.0125 6.62716 11.8188 6.79747Z", fill: "currentColor" }));
    ThumbUpIcon = styled_components_esm_default(Icon6).attrs(() => ({
      "data-component-name": "icons/ThumbUpIcon/ThumbUpIcon"
    })).withConfig({
      displayName: "ThumbUpIcon",
      componentId: "sc-1mr22fg-0"
    })([""]);
  }
});

// node_modules/@redocly/theme/src/components/Feedback/Sentiment.tsx
import * as React12 from "react";
import { useEffect as useEffect2 } from "react";
function Sentiment({
  settings,
  onSubmit,
  className
}) {
  const {
    label,
    submitText,
    comment: commentSettings,
    reasons: reasonsSettings
  } = settings || {};
  const [isSubmitted, setIsSubmitted] = React12.useState(false);
  const [score, setScore] = React12.useState(0);
  const [comment, setComment] = React12.useState("");
  const [reasons, setReasons] = React12.useState([]);
  const {
    useTranslate
  } = useThemeHooks();
  const {
    translate
  } = useTranslate();
  const getScoreSpecificReasonSettings = (score2) => {
    switch (score2) {
      case 1:
        return reasonsSettings?.like;
      case -1:
        return reasonsSettings?.dislike;
      default:
        return reasonsSettings;
    }
  };
  const checkIfShouldDisplayReasons = (score2) => {
    if (!score2 || !reasonsSettings) {
      return false;
    }
    const scoreSpecificReasonsSettings = getScoreSpecificReasonSettings(score2);
    if (typeof scoreSpecificReasonsSettings?.hide === "undefined") {
      return !reasonsSettings.hide;
    }
    return !scoreSpecificReasonsSettings?.hide;
  };
  const displayReasons = checkIfShouldDisplayReasons(score);
  const displayComment = !!(score && !commentSettings?.hide);
  const displaySubmitBnt = !!(score && (displayReasons || displayComment));
  const commentLabel = score === 1 ? commentSettings && commentSettings.likeLabel || translate("feedback.settings.comment.likeLabel", "What was most helpful?") : commentSettings && commentSettings.dislikeLabel || translate("feedback.settings.comment.dislikeLabel", "What can we improve?");
  const renderReasonsLabel = (score2) => {
    const scoreSpecificReasonSettings = getScoreSpecificReasonSettings(score2);
    return scoreSpecificReasonSettings?.label;
  };
  const resolveReasonsItems = (score2) => {
    const defaultReasonsItems = reasonsSettings?.items || [];
    const scoreSpecificReasonSettings = getScoreSpecificReasonSettings(score2);
    return defaultReasonsItems.concat(scoreSpecificReasonSettings?.items ?? []);
  };
  const resolveReasonsComponent = (score2) => {
    const scoreSpecificReasonSettings = getScoreSpecificReasonSettings(score2);
    return scoreSpecificReasonSettings?.component;
  };
  const onSubmitSentimentForm = () => {
    onSubmit({
      score,
      comment,
      reasons
    });
    setIsSubmitted(true);
  };
  const onCancelSentimentForm = () => {
    setScore(0);
  };
  useEffect2(() => {
    if (score && !displayComment && !displayReasons) {
      onSubmitSentimentForm();
    }
  }, [score, displayComment, displayReasons]);
  if (isSubmitted) {
    return /* @__PURE__ */ React12.createElement(SentimentWrapper, { "data-component-name": "Feedback/Sentiment" }, /* @__PURE__ */ React12.createElement(StyledFormMandatoryFields3, null, /* @__PURE__ */ React12.createElement(Label4, { "data-translation-key": "feedback.settings.submitText" }, submitText || translate("feedback.settings.submitText", "Thank you for helping improve our documentation!")), /* @__PURE__ */ React12.createElement(RadioCheckButtonIcon, null)));
  }
  return /* @__PURE__ */ React12.createElement(SentimentWrapper, { "data-component-name": "Feedback/Sentiment", className }, /* @__PURE__ */ React12.createElement(StyledForm2, null, /* @__PURE__ */ React12.createElement(StyledFormMandatoryFields3, null, /* @__PURE__ */ React12.createElement(Label4, { "data-translation-key": "feedback.settings.label" }, label || translate("feedback.settings.label", "Was this helpful?")), /* @__PURE__ */ React12.createElement(StyledMandatoryFieldContainer2, null, /* @__PURE__ */ React12.createElement(Button, { type: "button", size: "medium", variant: score === 1 ? "primary" : "secondary", onClick: () => {
    setScore(1);
  }, "aria-label": "Like", icon: /* @__PURE__ */ React12.createElement(ThumbUpIcon, null) }), /* @__PURE__ */ React12.createElement(Button, { size: "medium", type: "button", variant: score === -1 ? "primary" : "secondary", tone: score === -1 ? "danger" : "default", onClick: () => {
    setScore(-1);
  }, "aria-label": "Dislike", icon: /* @__PURE__ */ React12.createElement(ThumbDownIcon, null) }))), (displayReasons || displayComment) && /* @__PURE__ */ React12.createElement(StyledFormOptionalFields2, null, displayReasons && /* @__PURE__ */ React12.createElement(Reasons, { settings: {
    label: renderReasonsLabel(score),
    items: resolveReasonsItems(score),
    component: resolveReasonsComponent(score)
  }, onChange: setReasons }), displayComment && /* @__PURE__ */ React12.createElement(Comment, { standAlone: false, onSubmit: ({
    comment: comment2
  }) => setComment(comment2), settings: {
    label: commentLabel
  } })), displaySubmitBnt && /* @__PURE__ */ React12.createElement(ButtonsContainer3, null, /* @__PURE__ */ React12.createElement(Button, { onClick: onCancelSentimentForm, variant: "text", size: "small" }, "Cancel"), /* @__PURE__ */ React12.createElement(Button, { onClick: onSubmitSentimentForm, variant: "secondary", size: "small" }, "Submit"))));
}
var SentimentWrapper, Label4, StyledForm2, StyledFormOptionalFields2, StyledFormMandatoryFields3, StyledMandatoryFieldContainer2, ButtonsContainer3;
var init_Sentiment = __esm({
  "node_modules/@redocly/theme/src/components/Feedback/Sentiment.tsx"() {
    init_styled_components_esm();
    init_hooks();
    init_RadioCheckButtonIcon();
    init_Comment();
    init_Reasons();
    init_Button();
    init_ThumbDownIcon();
    init_ThumbUpIcon();
    SentimentWrapper = styled_components_esm_default.div.withConfig({
      displayName: "Sentiment__SentimentWrapper",
      componentId: "sc-1kmnv8-0"
    })(["font-family:var(--font-family-base);display:flex;justify-content:space-between;align-items:center;"]);
    Label4 = styled_components_esm_default.h4.withConfig({
      displayName: "Sentiment__Label",
      componentId: "sc-1kmnv8-1"
    })(["font-family:var(--feedback-font-family);font-weight:var(--font-weight-regular);font-size:var(--feedback-header-font-size);line-height:var(--feedback-header-line-height);color:var(--feedback-header-text-color);margin:0;"]);
    StyledForm2 = styled_components_esm_default.form.withConfig({
      displayName: "Sentiment__StyledForm",
      componentId: "sc-1kmnv8-2"
    })(["width:100%;gap:var(--spacing-sm);display:flex;flex-direction:column;"]);
    StyledFormOptionalFields2 = styled_components_esm_default.div.withConfig({
      displayName: "Sentiment__StyledFormOptionalFields",
      componentId: "sc-1kmnv8-3"
    })(["display:flex;flex-flow:column;"]);
    StyledFormMandatoryFields3 = styled_components_esm_default.div.withConfig({
      displayName: "Sentiment__StyledFormMandatoryFields",
      componentId: "sc-1kmnv8-4"
    })(["display:flex;justify-content:space-between;align-items:center;gap:var(--spacing-xs);> svg{margin:var(--button-icon-padding) 0;}"]);
    StyledMandatoryFieldContainer2 = styled_components_esm_default.div.withConfig({
      displayName: "Sentiment__StyledMandatoryFieldContainer",
      componentId: "sc-1kmnv8-5"
    })(["display:flex;align-items:center;gap:var(--spacing-xxs);", "{margin:0;}"], Button);
    ButtonsContainer3 = styled_components_esm_default.div.withConfig({
      displayName: "Sentiment__ButtonsContainer",
      componentId: "sc-1kmnv8-6"
    })(["display:flex;justify-content:end;margin-bottom:var(--spacing-xxs);gap:var(--spacing-xxs);"]);
  }
});

// node_modules/@redocly/theme/src/icons/FaceDissatisfiedIcon/FaceDissatisfiedIcon.tsx
import React13 from "react";
var Icon7, FaceDissatisfiedIcon;
var init_FaceDissatisfiedIcon = __esm({
  "node_modules/@redocly/theme/src/icons/FaceDissatisfiedIcon/FaceDissatisfiedIcon.tsx"() {
    init_styled_components_esm();
    init_utils();
    Icon7 = (props) => /* @__PURE__ */ React13.createElement("svg", { viewBox: "1 1 14 14", fill: "none", xmlns: "http://www.w3.org/2000/svg", ...props }, /* @__PURE__ */ React13.createElement("path", { d: "M8 1C6.61553 1 5.26216 1.41054 4.11101 2.17971C2.95987 2.94888 2.06266 4.04213 1.53285 5.32122C1.00303 6.6003 0.86441 8.00776 1.13451 9.36563C1.4046 10.7235 2.07129 11.9708 3.05026 12.9497C4.02922 13.9287 5.2765 14.5954 6.63437 14.8655C7.99224 15.1356 9.3997 14.997 10.6788 14.4672C11.9579 13.9373 13.0511 13.0401 13.8203 11.889C14.5895 10.7378 15 9.38447 15 8C15 6.14348 14.2625 4.36301 12.9497 3.05025C11.637 1.7375 9.85652 1 8 1ZM8 14C6.81332 14 5.65328 13.6481 4.66658 12.9888C3.67989 12.3295 2.91085 11.3925 2.45673 10.2961C2.0026 9.19974 1.88378 7.99334 2.11529 6.82946C2.3468 5.66557 2.91825 4.59647 3.75736 3.75736C4.59648 2.91824 5.66558 2.3468 6.82946 2.11529C7.99335 1.88378 9.19975 2.0026 10.2961 2.45672C11.3925 2.91085 12.3295 3.67988 12.9888 4.66658C13.6481 5.65327 14 6.81331 14 8C14 9.5913 13.3679 11.1174 12.2426 12.2426C11.1174 13.3679 9.5913 14 8 14Z" }), /* @__PURE__ */ React13.createElement("path", { d: "M5.75 5.5C5.50278 5.5 5.2611 5.57331 5.05554 5.71066C4.84998 5.84801 4.68976 6.04324 4.59515 6.27165C4.50054 6.50005 4.47579 6.75139 4.52402 6.99386C4.57225 7.23634 4.6913 7.45907 4.86612 7.63388C5.04094 7.8087 5.26366 7.92775 5.50614 7.97598C5.74862 8.02421 5.99995 7.99946 6.22836 7.90485C6.45677 7.81024 6.65199 7.65002 6.78934 7.44446C6.92669 7.2389 7 6.99723 7 6.75C7 6.41848 6.86831 6.10054 6.63389 5.86612C6.39947 5.6317 6.08152 5.5 5.75 5.5Z" }), /* @__PURE__ */ React13.createElement("path", { d: "M10.25 5.5C10.0028 5.5 9.7611 5.57331 9.55554 5.71066C9.34998 5.84801 9.18976 6.04324 9.09515 6.27165C9.00054 6.50005 8.97579 6.75139 9.02402 6.99386C9.07225 7.23634 9.1913 7.45907 9.36612 7.63388C9.54093 7.8087 9.76366 7.92775 10.0061 7.97598C10.2486 8.02421 10.4999 7.99946 10.7284 7.90485C10.9568 7.81024 11.152 7.65002 11.2893 7.44446C11.4267 7.2389 11.5 6.99723 11.5 6.75C11.5 6.41848 11.3683 6.10054 11.1339 5.86612C10.8995 5.6317 10.5815 5.5 10.25 5.5Z" }), /* @__PURE__ */ React13.createElement("path", { d: "M8 9.5C7.30982 9.50117 6.63168 9.6809 6.03151 10.0217C5.43135 10.3625 4.92958 10.8529 4.575 11.445L5.43 11.945C5.69683 11.5019 6.07369 11.1353 6.524 10.8808C6.97431 10.6263 7.48276 10.4926 8 10.4926C8.51724 10.4926 9.0257 10.6263 9.47601 10.8808C9.92631 11.1353 10.3032 11.5019 10.57 11.945L11.425 11.445C11.0704 10.8529 10.5687 10.3625 9.96849 10.0217C9.36833 9.6809 8.69019 9.50117 8 9.5Z" }));
    FaceDissatisfiedIcon = styled_components_esm_default(Icon7).attrs(() => ({
      "data-component-name": "icons/FaceDissatisfiedIcon/FaceDissatisfiedIcon"
    })).withConfig({
      displayName: "FaceDissatisfiedIcon",
      componentId: "sc-1mq0tcx-0"
    })(["path{fill:", ";}height:", ";width:", ";"], ({
      color
    }) => getCssColorVariable(color), ({
      size
    }) => size || "16px", ({
      size
    }) => size || "16px");
  }
});

// node_modules/@redocly/theme/src/icons/FaceSatisfiedIcon/FaceSatisfiedIcon.tsx
import React14 from "react";
var Icon8, FaceSatisfiedIcon;
var init_FaceSatisfiedIcon = __esm({
  "node_modules/@redocly/theme/src/icons/FaceSatisfiedIcon/FaceSatisfiedIcon.tsx"() {
    init_styled_components_esm();
    init_utils();
    Icon8 = (props) => /* @__PURE__ */ React14.createElement("svg", { viewBox: "1 1 14 14", fill: "none", xmlns: "http://www.w3.org/2000/svg", ...props }, /* @__PURE__ */ React14.createElement("path", { d: "M8 1C6.61553 1 5.26216 1.41054 4.11101 2.17971C2.95987 2.94888 2.06266 4.04213 1.53285 5.32122C1.00303 6.6003 0.86441 8.00776 1.13451 9.36563C1.4046 10.7235 2.07129 11.9708 3.05026 12.9497C4.02922 13.9287 5.2765 14.5954 6.63437 14.8655C7.99224 15.1356 9.3997 14.997 10.6788 14.4672C11.9579 13.9373 13.0511 13.0401 13.8203 11.889C14.5895 10.7378 15 9.38447 15 8C15 6.14348 14.2625 4.36301 12.9497 3.05025C11.637 1.7375 9.85652 1 8 1ZM8 14C6.81332 14 5.65328 13.6481 4.66658 12.9888C3.67989 12.3295 2.91085 11.3925 2.45673 10.2961C2.0026 9.19974 1.88378 7.99334 2.11529 6.82946C2.3468 5.66557 2.91825 4.59647 3.75736 3.75736C4.59648 2.91824 5.66558 2.3468 6.82946 2.11529C7.99335 1.88378 9.19975 2.0026 10.2961 2.45672C11.3925 2.91085 12.3295 3.67988 12.9888 4.66658C13.6481 5.65327 14 6.81331 14 8C14 9.5913 13.3679 11.1174 12.2426 12.2426C11.1174 13.3679 9.5913 14 8 14Z" }), /* @__PURE__ */ React14.createElement("path", { d: "M5.75 5.5C5.50278 5.5 5.2611 5.57331 5.05554 5.71066C4.84998 5.84801 4.68976 6.04324 4.59515 6.27165C4.50054 6.50005 4.47579 6.75139 4.52402 6.99386C4.57225 7.23634 4.6913 7.45907 4.86612 7.63388C5.04094 7.8087 5.26366 7.92775 5.50614 7.97598C5.74862 8.02421 5.99995 7.99946 6.22836 7.90485C6.45677 7.81024 6.65199 7.65002 6.78934 7.44446C6.92669 7.2389 7 6.99723 7 6.75C7.00134 6.58548 6.96991 6.42233 6.90757 6.27007C6.84522 6.11782 6.7532 5.97949 6.63686 5.86315C6.52052 5.74681 6.38219 5.65478 6.22993 5.59244C6.07767 5.53009 5.91453 5.49867 5.75 5.5Z" }), /* @__PURE__ */ React14.createElement("path", { d: "M10.25 5.5C10.0028 5.5 9.7611 5.57331 9.55554 5.71066C9.34998 5.84801 9.18976 6.04324 9.09515 6.27165C9.00054 6.50005 8.97579 6.75139 9.02402 6.99386C9.07225 7.23634 9.1913 7.45907 9.36612 7.63388C9.54093 7.8087 9.76366 7.92775 10.0061 7.97598C10.2486 8.02421 10.4999 7.99946 10.7284 7.90485C10.9568 7.81024 11.152 7.65002 11.2893 7.44446C11.4267 7.2389 11.5 6.99723 11.5 6.75C11.5013 6.58548 11.4699 6.42233 11.4076 6.27007C11.3452 6.11782 11.2532 5.97949 11.1369 5.86315C11.0205 5.74681 10.8822 5.65478 10.7299 5.59244C10.5777 5.53009 10.4145 5.49867 10.25 5.5Z" }), /* @__PURE__ */ React14.createElement("path", { d: "M8 12C8.69019 11.9988 9.36833 11.8191 9.96849 11.4783C10.5687 11.1375 11.0704 10.6471 11.425 10.055L10.57 9.555C10.3032 9.9981 9.92631 10.3647 9.47601 10.6192C9.0257 10.8737 8.51724 11.0074 8 11.0074C7.48276 11.0074 6.97431 10.8737 6.524 10.6192C6.07369 10.3647 5.69683 9.9981 5.43 9.555L4.575 10.055C4.92958 10.6471 5.43135 11.1375 6.03151 11.4783C6.63168 11.8191 7.30982 11.9988 8 12Z" }));
    FaceSatisfiedIcon = styled_components_esm_default(Icon8).attrs(() => ({
      "data-component-name": "icons/FaceSatisfiedIcon/FaceSatisfiedIcon"
    })).withConfig({
      displayName: "FaceSatisfiedIcon",
      componentId: "sc-1qghnnx-0"
    })(["path{fill:", ";}height:", ";width:", ";"], ({
      color
    }) => getCssColorVariable(color), ({
      size
    }) => size || "16px", ({
      size
    }) => size || "16px");
  }
});

// node_modules/@redocly/theme/src/icons/FaceNeutralIcon/FaceNeutralIcon.tsx
import React15 from "react";
var Icon9, FaceNeutralIcon;
var init_FaceNeutralIcon = __esm({
  "node_modules/@redocly/theme/src/icons/FaceNeutralIcon/FaceNeutralIcon.tsx"() {
    init_styled_components_esm();
    init_utils();
    Icon9 = (props) => /* @__PURE__ */ React15.createElement("svg", { viewBox: "1 1 14 14", fill: "none", xmlns: "http://www.w3.org/2000/svg", ...props }, /* @__PURE__ */ React15.createElement("path", { d: "M8 1C6.61553 1 5.26216 1.41054 4.11101 2.17971C2.95987 2.94888 2.06266 4.04213 1.53285 5.32122C1.00303 6.6003 0.86441 8.00776 1.13451 9.36563C1.4046 10.7235 2.07129 11.9708 3.05026 12.9497C4.02922 13.9287 5.2765 14.5954 6.63437 14.8655C7.99224 15.1356 9.3997 14.997 10.6788 14.4672C11.9579 13.9373 13.0511 13.0401 13.8203 11.889C14.5895 10.7378 15 9.38447 15 8C15 6.14348 14.2625 4.36301 12.9497 3.05025C11.637 1.7375 9.85652 1 8 1ZM8 14C6.81332 14 5.65328 13.6481 4.66658 12.9888C3.67989 12.3295 2.91085 11.3925 2.45673 10.2961C2.0026 9.19974 1.88378 7.99334 2.11529 6.82946C2.3468 5.66557 2.91825 4.59647 3.75736 3.75736C4.59648 2.91824 5.66558 2.3468 6.82946 2.11529C7.99335 1.88378 9.19975 2.0026 10.2961 2.45672C11.3925 2.91085 12.3295 3.67988 12.9888 4.66658C13.6481 5.65327 14 6.81331 14 8C14 9.5913 13.3679 11.1174 12.2426 12.2426C11.1174 13.3679 9.5913 14 8 14Z" }), /* @__PURE__ */ React15.createElement("path", { d: "M5.75 5.5C5.50278 5.5 5.2611 5.57331 5.05554 5.71066C4.84998 5.84801 4.68976 6.04324 4.59515 6.27165C4.50054 6.50005 4.47579 6.75139 4.52402 6.99386C4.57225 7.23634 4.6913 7.45907 4.86612 7.63388C5.04094 7.8087 5.26366 7.92775 5.50614 7.97598C5.74862 8.02421 5.99995 7.99946 6.22836 7.90485C6.45677 7.81024 6.65199 7.65002 6.78934 7.44446C6.92669 7.2389 7 6.99723 7 6.75C7 6.41848 6.86831 6.10054 6.63389 5.86612C6.39947 5.6317 6.08152 5.5 5.75 5.5Z" }), /* @__PURE__ */ React15.createElement("path", { d: "M10.25 5.5C10.0028 5.5 9.7611 5.57331 9.55554 5.71066C9.34998 5.84801 9.18976 6.04324 9.09515 6.27165C9.00054 6.50005 8.97579 6.75139 9.02402 6.99386C9.07225 7.23634 9.1913 7.45907 9.36612 7.63388C9.54093 7.8087 9.76366 7.92775 10.0061 7.97598C10.2486 8.02421 10.4999 7.99946 10.7284 7.90485C10.9568 7.81024 11.152 7.65002 11.2893 7.44446C11.4267 7.2389 11.5 6.99723 11.5 6.75C11.5 6.41848 11.3683 6.10054 11.1339 5.86612C10.8995 5.6317 10.5815 5.5 10.25 5.5Z" }), /* @__PURE__ */ React15.createElement("path", { d: "M11 10H5V11H11V10Z" }));
    FaceNeutralIcon = styled_components_esm_default(Icon9).attrs(() => ({
      "data-component-name": "icons/FaceNeutralIcon/FaceNeutralIcon"
    })).withConfig({
      displayName: "FaceNeutralIcon",
      componentId: "sc-kylii8-0"
    })(["path{fill:", ";}height:", ";width:", ";"], ({
      color
    }) => getCssColorVariable(color), ({
      size
    }) => size || "16px", ({
      size
    }) => size || "16px");
  }
});

// node_modules/@redocly/theme/src/components/Feedback/Mood.tsx
import * as React16 from "react";
import { useEffect as useEffect3 } from "react";
function Mood({
  settings,
  onSubmit,
  className
}) {
  const {
    label,
    submitText,
    comment: commentSettings,
    reasons: reasonsSettings
  } = settings || {};
  const [score, setScore] = React16.useState("");
  const [isSubmitted, setIsSubmitted] = React16.useState(false);
  const [comment, setComment] = React16.useState("");
  const [reasons, setReasons] = React16.useState([]);
  const {
    useTranslate
  } = useThemeHooks();
  const {
    translate
  } = useTranslate();
  const checkIfShouldDisplayReasons = (score2) => {
    if (!score2 || !reasonsSettings) {
      return false;
    }
    const scoreSpecificReasonsSettings = reasonsSettings[score2];
    if (typeof scoreSpecificReasonsSettings?.hide === "undefined") {
      return !reasonsSettings.hide;
    }
    return !scoreSpecificReasonsSettings?.hide;
  };
  const renderCommentLabel = (score2) => {
    if (!commentSettings || commentSettings?.hide === true)
      return "";
    switch (score2) {
      case "satisfied" /* SATISFIED */:
        return commentSettings.satisfiedLabel || translate("feedback.settings.comment.satisfiedLabel", "What was most helpful?");
      case "neutral" /* NEUTRAL */:
        return commentSettings.neutralLabel || translate("feedback.settings.comment.neutralLabel", "What can we improve?");
      case "dissatisfied" /* DISSATISFIED */:
        return commentSettings.dissatisfiedLabel || translate("feedback.settings.comment.dissatisfiedLabel", "What can we improve?");
      default:
        return translate("feedback.settings.comment.satisfiedLabel", "What can we improve?");
    }
  };
  const renderReasonsLabel = (score2) => {
    if (!reasonsSettings || reasonsSettings?.hide === true)
      return "";
    return reasonsSettings[score2]?.label;
  };
  const resolveReasonsItems = (score2) => {
    if (!reasonsSettings) {
      return [];
    }
    const defaultReasonsItems = reasonsSettings?.items || [];
    return defaultReasonsItems.concat(reasonsSettings[score2]?.items ?? []);
  };
  const resolveReasonsComponent = (score2) => {
    if (!reasonsSettings || reasonsSettings?.hide === true)
      return "checkbox";
    return reasonsSettings[score2]?.component;
  };
  const displayReasons = checkIfShouldDisplayReasons(score);
  const displayComment = !!(score && !commentSettings?.hide);
  const displaySubmitBnt = !!(score && (displayReasons || displayComment));
  const onSubmitMoodForm = () => {
    onSubmit({
      score: remapScore(score),
      comment,
      reasons
    });
    setIsSubmitted(true);
  };
  const onCancelMoodForm = () => {
    setScore("");
  };
  useEffect3(() => {
    if (score && !displayComment && !displayReasons) {
      onSubmitMoodForm();
    }
  }, [score, displayComment, displayReasons]);
  if (isSubmitted) {
    return /* @__PURE__ */ React16.createElement(MoodWrapper, { "data-component-name": "Feedback/Mood" }, /* @__PURE__ */ React16.createElement(StyledFormMandatoryFields4, null, /* @__PURE__ */ React16.createElement(Label5, { "data-translation-key": "feedback.settings.submitText" }, submitText || translate("feedback.settings.submitText", "Thank you for helping improve our documentation!")), /* @__PURE__ */ React16.createElement(RadioCheckButtonIcon, null)));
  }
  return /* @__PURE__ */ React16.createElement(MoodWrapper, { "data-component-name": "Feedback/Mood", className }, /* @__PURE__ */ React16.createElement(StyledForm3, null, /* @__PURE__ */ React16.createElement(StyledFormMandatoryFields4, null, /* @__PURE__ */ React16.createElement(Label5, { "data-translation-key": "feedback.settings.label" }, label || translate("feedback.settings.label", "Was this helpful?")), /* @__PURE__ */ React16.createElement(StyledMandatoryFieldContainer3, null, /* @__PURE__ */ React16.createElement(Button, { "aria-label": "dissatisfied" /* DISSATISFIED */, type: "button", size: "medium", variant: score === "dissatisfied" /* DISSATISFIED */ ? "primary" : "secondary", tone: score === "dissatisfied" /* DISSATISFIED */ ? "danger" : "default", onClick: () => {
    setScore("dissatisfied" /* DISSATISFIED */);
  }, icon: /* @__PURE__ */ React16.createElement(FaceDissatisfiedIcon, null) }), /* @__PURE__ */ React16.createElement(Button, { "aria-label": "neutral" /* NEUTRAL */, type: "button", variant: score === "neutral" /* NEUTRAL */ ? "primary" : "secondary", size: "medium", onClick: () => {
    setScore("neutral" /* NEUTRAL */);
  }, icon: /* @__PURE__ */ React16.createElement(FaceNeutralIcon, null) }), /* @__PURE__ */ React16.createElement(Button, { "aria-label": "satisfied" /* SATISFIED */, type: "button", variant: score === "satisfied" /* SATISFIED */ ? "primary" : "secondary", size: "medium", onClick: () => {
    setScore("satisfied" /* SATISFIED */);
  }, icon: /* @__PURE__ */ React16.createElement(FaceSatisfiedIcon, null) }))), (displayReasons || displayComment) && /* @__PURE__ */ React16.createElement(StyledFormOptionalFields3, null, displayReasons && /* @__PURE__ */ React16.createElement(Reasons, { settings: {
    label: renderReasonsLabel(score),
    items: resolveReasonsItems(score),
    component: resolveReasonsComponent(score)
  }, onChange: setReasons }), displayComment && /* @__PURE__ */ React16.createElement(Comment, { standAlone: false, onSubmit: ({
    comment: comment2
  }) => setComment(comment2), settings: {
    label: renderCommentLabel(score)
  } })), displaySubmitBnt && /* @__PURE__ */ React16.createElement(ButtonsContainer4, null, /* @__PURE__ */ React16.createElement(Button, { onClick: onCancelMoodForm, variant: "text", size: "small" }, "Cancel"), /* @__PURE__ */ React16.createElement(Button, { onClick: onSubmitMoodForm, variant: "secondary", size: "small" }, "Submit"))));
}
var MOOD_STATES, remapScore, MoodWrapper, Label5, ButtonsContainer4, StyledForm3, StyledFormOptionalFields3, StyledFormMandatoryFields4, StyledMandatoryFieldContainer3;
var init_Mood = __esm({
  "node_modules/@redocly/theme/src/components/Feedback/Mood.tsx"() {
    init_styled_components_esm();
    init_hooks();
    init_RadioCheckButtonIcon();
    init_Comment();
    init_Reasons();
    init_Button();
    init_FaceDissatisfiedIcon();
    init_FaceSatisfiedIcon();
    init_FaceNeutralIcon();
    MOOD_STATES = /* @__PURE__ */ ((MOOD_STATES2) => {
      MOOD_STATES2["SATISFIED"] = "satisfied";
      MOOD_STATES2["NEUTRAL"] = "neutral";
      MOOD_STATES2["DISSATISFIED"] = "dissatisfied";
      return MOOD_STATES2;
    })(MOOD_STATES || {});
    remapScore = (score) => {
      switch (score) {
        case "satisfied" /* SATISFIED */:
          return 1;
        case "neutral" /* NEUTRAL */:
          return 0;
        case "dissatisfied" /* DISSATISFIED */:
          return -1;
        default:
          return 0;
      }
    };
    MoodWrapper = styled_components_esm_default.div.withConfig({
      displayName: "Mood__MoodWrapper",
      componentId: "sc-1ett8m-0"
    })(["font-family:var(--feedback-font-family);display:flex;justify-content:space-between;align-items:center;"]);
    Label5 = styled_components_esm_default.h4.withConfig({
      displayName: "Mood__Label",
      componentId: "sc-1ett8m-1"
    })(["font-family:var(--feedback-font-family);font-weight:var(--font-weight-regular);font-size:var(--feedback-header-font-size);line-height:var(--feedback-header-line-height);color:var(--feedback-header-text-color);margin:0;"]);
    ButtonsContainer4 = styled_components_esm_default.div.withConfig({
      displayName: "Mood__ButtonsContainer",
      componentId: "sc-1ett8m-2"
    })(["display:flex;justify-content:end;gap:var(--spacing-xxs);"]);
    StyledForm3 = styled_components_esm_default.form.withConfig({
      displayName: "Mood__StyledForm",
      componentId: "sc-1ett8m-3"
    })(["width:100%;gap:var(--spacing-sm);display:flex;flex-direction:column;"]);
    StyledFormOptionalFields3 = styled_components_esm_default.div.withConfig({
      displayName: "Mood__StyledFormOptionalFields",
      componentId: "sc-1ett8m-4"
    })(["display:flex;flex-flow:column;"]);
    StyledFormMandatoryFields4 = styled_components_esm_default.div.withConfig({
      displayName: "Mood__StyledFormMandatoryFields",
      componentId: "sc-1ett8m-5"
    })(["display:flex;justify-content:space-between;align-items:center;gap:var(--spacing-xs);> svg{margin:var(--button-icon-padding) 0;}"]);
    StyledMandatoryFieldContainer3 = styled_components_esm_default.div.withConfig({
      displayName: "Mood__StyledMandatoryFieldContainer",
      componentId: "sc-1ett8m-6"
    })(["display:flex;align-items:center;gap:var(--spacing-xxs);"]);
  }
});

// node_modules/@redocly/theme/src/components/Feedback/Scale.tsx
import * as React17 from "react";
import { useEffect as useEffect4 } from "react";
function Scale({
  settings,
  onSubmit,
  className
}) {
  const {
    label,
    submitText,
    leftScaleLabel,
    rightScaleLabel,
    comment: commentSettings,
    reasons: reasonsSettings
  } = settings || {};
  const [score, setScore] = React17.useState(0);
  const [isSubmitted, setIsSubmitted] = React17.useState(false);
  const [comment, setComment] = React17.useState("");
  const [reasons, setReasons] = React17.useState([]);
  const {
    useTranslate
  } = useThemeHooks();
  const {
    translate
  } = useTranslate();
  const scaleOptions = [];
  for (let i = 1; i <= MAX_SCALE; i++) {
    scaleOptions.push(/* @__PURE__ */ React17.createElement(Button, { id: `scale-option-${i}`, key: `scale-option-${i}`, type: "button", onClick: () => setScore(i), className: `${score === i ? "active" : ""}` }, i));
  }
  const displayReasons = !!score && reasonsSettings && !reasonsSettings.hide;
  const displayComment = !!(score && !commentSettings?.hide);
  const displaySubmitBnt = !!score && (displayReasons || displayComment);
  const handleCancel = () => {
    setScore(0);
    setComment("");
    setReasons([]);
  };
  const onSubmitScaleForm = () => {
    onSubmit({
      score,
      comment,
      reasons,
      max: MAX_SCALE
    });
    setIsSubmitted(true);
  };
  useEffect4(() => {
    if (score && !displayComment && !displayReasons) {
      onSubmitScaleForm();
    }
  }, [score, displayComment, displayReasons]);
  if (isSubmitted) {
    return /* @__PURE__ */ React17.createElement(ScaleWrapper, { "data-component-name": "Feedback/Scale" }, /* @__PURE__ */ React17.createElement(Label6, { "data-translation-key": "feedback.settings.submitText" }, submitText || translate("feedback.settings.submitText", "Thank you for helping improve our documentation!")), /* @__PURE__ */ React17.createElement(RadioCheckButtonIcon, null));
  }
  return /* @__PURE__ */ React17.createElement(ScaleWrapper, { "data-component-name": "Feedback/Scale", className }, /* @__PURE__ */ React17.createElement(StyledForm4, null, /* @__PURE__ */ React17.createElement(StyledFormMandatoryFields5, null, /* @__PURE__ */ React17.createElement(Label6, { "data-translation-key": "feedback.settings.label" }, label || translate("feedback.settings.label", "Was this helpful?")), /* @__PURE__ */ React17.createElement(ScaleOptionsWrapper, null, scaleOptions), /* @__PURE__ */ React17.createElement(SubLabelContainer, null, /* @__PURE__ */ React17.createElement(SubLabel, { "data-translation-key": "feedback.settings.leftScaleLabel" }, leftScaleLabel || translate("feedback.settings.leftScaleLabel", "Not helpful at all")), /* @__PURE__ */ React17.createElement(SubLabel, { "data-translation-key": "feedback.settings.rightScaleLabel" }, rightScaleLabel || translate("feedback.settings.rightScaleLabel", "Extremely helpful")))), /* @__PURE__ */ React17.createElement(StyledFormOptionalFields4, null, displayReasons && /* @__PURE__ */ React17.createElement(Reasons, { settings: {
    label: reasonsSettings?.label,
    items: reasonsSettings?.items || [],
    component: reasonsSettings?.component
  }, onChange: setReasons }), displayComment && /* @__PURE__ */ React17.createElement(Comment, { standAlone: false, onSubmit: ({
    comment: comment2
  }) => setComment(comment2), settings: {
    label: commentSettings?.label || translate("feedback.settings.comment.label", "Please share your feedback with us.")
  } })), displaySubmitBnt && /* @__PURE__ */ React17.createElement(ButtonsContainer5, null, /* @__PURE__ */ React17.createElement(Button, { "data-translation-key": "feedback.settings.comment.cancel", onClick: handleCancel, variant: "text", size: "small" }, translate("feedback.settings.comment.cancel", "Cancel")), /* @__PURE__ */ React17.createElement(Button, { "data-translation-key": "feedback.settings.scale.send", onClick: onSubmitScaleForm, variant: "secondary", size: "small" }, translate("feedback.settings.scale.send", "Submit")))));
}
var MAX_SCALE, ScaleWrapper, Label6, SubLabelContainer, SubLabel, StyledForm4, ButtonsContainer5, StyledFormOptionalFields4, StyledFormMandatoryFields5, ScaleOptionsWrapper;
var init_Scale = __esm({
  "node_modules/@redocly/theme/src/components/Feedback/Scale.tsx"() {
    init_styled_components_esm();
    init_utils();
    init_hooks();
    init_RadioCheckButtonIcon();
    init_Comment();
    init_Reasons();
    init_Button();
    MAX_SCALE = 10;
    ScaleWrapper = styled_components_esm_default.div.withConfig({
      displayName: "Scale__ScaleWrapper",
      componentId: "sc-1i5fldo-0"
    })(["font-family:var(--feedback-font-family);display:flex;justify-content:space-between;align-items:center;"]);
    Label6 = styled_components_esm_default.h4.withConfig({
      displayName: "Scale__Label",
      componentId: "sc-1i5fldo-1"
    })(["font-family:var(--feedback-font-family);font-weight:var(--font-weight-regular);font-size:var(--feedback-header-font-size);line-height:var(--feedback-header-line-height);color:var(--feedback-header-text-color);margin:0;width:100%;"]);
    SubLabelContainer = styled_components_esm_default.div.withConfig({
      displayName: "Scale__SubLabelContainer",
      componentId: "sc-1i5fldo-2"
    })(["display:flex;justify-content:space-between;width:100%;flex-direction:row;"]);
    SubLabel = styled_components_esm_default(Label6).withConfig({
      displayName: "Scale__SubLabel",
      componentId: "sc-1i5fldo-3"
    })(["width:fit-content;color:var(--text-color-description);font-size:var(--feedback-font-size);margin-bottom:var(--spacing-xxs);"]);
    StyledForm4 = styled_components_esm_default.form.withConfig({
      displayName: "Scale__StyledForm",
      componentId: "sc-1i5fldo-4"
    })(["width:100%;gap:var(--spacing-sm);display:flex;flex-direction:column;"]);
    ButtonsContainer5 = styled_components_esm_default.div.withConfig({
      displayName: "Scale__ButtonsContainer",
      componentId: "sc-1i5fldo-5"
    })(["display:flex;justify-content:end;margin-bottom:var(--spacing-xxs);gap:var(--spacing-xxs);"]);
    StyledFormOptionalFields4 = styled_components_esm_default.div.withConfig({
      displayName: "Scale__StyledFormOptionalFields",
      componentId: "sc-1i5fldo-6"
    })(["display:flex;flex-flow:column;&:empty{display:none;}"]);
    StyledFormMandatoryFields5 = styled_components_esm_default.div.withConfig({
      displayName: "Scale__StyledFormMandatoryFields",
      componentId: "sc-1i5fldo-7"
    })(["display:flex;flex-direction:column;align-items:center;gap:var(--spacing-sm);align-items:center;"]);
    ScaleOptionsWrapper = styled_components_esm_default.div.withConfig({
      displayName: "Scale__ScaleOptionsWrapper",
      componentId: "sc-1i5fldo-8"
    })(["display:flex;justify-content:space-between;flex-direction:row;gap:var(--spacing-xxs);width:100%;", "{margin-left:0;}@media screen and (max-width:", "){gap:2px;}"], Button, breakpoints.small);
  }
});

// node_modules/@redocly/theme/src/components/Feedback/Feedback.tsx
import * as React18 from "react";
import { useLocation } from "react-router-dom";
function Feedback(props) {
  const {
    useSubmitFeedback,
    useTelemetry
  } = useThemeHooks();
  const {
    submitFeedback
  } = useSubmitFeedback();
  const telemetry = useTelemetry();
  const {
    pathname
  } = useLocation();
  const {
    feedback: themeFeedbackConf
  } = useThemeConfig();
  const feedbackConf = {
    ...themeFeedbackConf,
    path: props?.path,
    type: props?.type || themeFeedbackConf?.type || DEFAULT_FEEDBACK_TYPE,
    settings: props?.settings || themeFeedbackConf?.settings || {},
    hide: props.hide ?? themeFeedbackConf?.hide
  };
  if (feedbackConf.hide) {
    return null;
  }
  const renderFeedbackComponent = () => {
    const {
      type,
      settings,
      path
    } = feedbackConf;
    const FeedbackComponent = feedbackComponents[type];
    if (!FeedbackComponent) {
      return null;
    }
    return /* @__PURE__ */ React18.createElement(FeedbackWrapper, { "data-component-name": "Feedback/Feedback" }, /* @__PURE__ */ React18.createElement(FeedbackComponent, { settings, onSubmit: (values) => {
      submitFeedback({
        type,
        values,
        path
      });
      telemetry.send("feedback_sent", {
        type
      });
    } }));
  };
  return /* @__PURE__ */ React18.createElement(React18.Fragment, { key: pathname }, renderFeedbackComponent());
}
var feedbackComponents, FeedbackWrapper;
var init_Feedback = __esm({
  "node_modules/@redocly/theme/src/components/Feedback/Feedback.tsx"() {
    init_styled_components_esm();
    init_hooks();
    init_Rating();
    init_Sentiment();
    init_Comment();
    init_Mood();
    init_Scale();
    init_constants();
    feedbackComponents = {
      rating: Rating,
      sentiment: Sentiment,
      mood: Mood,
      scale: Scale,
      comment: Comment
    };
    FeedbackWrapper = styled_components_esm_default.div.withConfig({
      displayName: "Feedback__FeedbackWrapper",
      componentId: "sc-hcc569-0"
    })(["padding:var(--feedback-padding);background-color:var(--feedback-bg-color);color:var(--feedback-font-color);font-size:var(--feedback-font-size);width:var(--feedback-width);max-width:100%;border-radius:var(--feedback-border-radius);@media print{display:none;}"]);
  }
});

// node_modules/@redocly/theme/src/components/Buttons/EditPageButton.tsx
import React19 from "react";
function EditPageButton({
  to
}) {
  const {
    useTranslate,
    useTelemetry
  } = useThemeHooks();
  const {
    translate
  } = useTranslate();
  const telemetry = useTelemetry();
  return /* @__PURE__ */ React19.createElement(EditPageButtonWrapper, { "data-component-name": "Buttons/EditPageButton", target: "_blank", to, onClick: () => telemetry.send("edit_page_link_clicked", {}) }, /* @__PURE__ */ React19.createElement(ButtonIcon, null), /* @__PURE__ */ React19.createElement(ButtonText, { "data-translation-key": "markdown.editPage.text" }, translate("markdown.editPage.text", "Edit")));
}
var EditPageButtonWrapper, ButtonIcon, ButtonText;
var init_EditPageButton = __esm({
  "node_modules/@redocly/theme/src/components/Buttons/EditPageButton.tsx"() {
    init_styled_components_esm();
    init_hooks();
    init_EditIcon();
    init_Link();
    EditPageButtonWrapper = styled_components_esm_default(Link).withConfig({
      displayName: "EditPageButton__EditPageButtonWrapper",
      componentId: "sc-15wuywf-0"
    })(["height:fit-content;margin-left:auto;display:inline-flex;color:var(--text-color-secondary);font-weight:var(--font-weight-bold);font-size:var(--font-size-base);font-family:var(--font-family-base);text-decoration:none;&:hover{color:var(--text-color-primary);}@media print{display:none;}"]);
    ButtonIcon = styled_components_esm_default(EditIcon).withConfig({
      displayName: "EditPageButton__ButtonIcon",
      componentId: "sc-15wuywf-1"
    })(["margin-right:3px;"]);
    ButtonText = styled_components_esm_default.span.withConfig({
      displayName: "EditPageButton__ButtonText",
      componentId: "sc-15wuywf-2"
    })(["line-height:14px;"]);
  }
});

// node_modules/@redocly/theme/src/components/Breadcrumbs/Breadcrumb.tsx
import React20 from "react";
function Breadcrumb({
  label,
  link,
  isActive,
  onClick
}) {
  return /* @__PURE__ */ React20.createElement(BreadcrumbWrapper, { "data-component-name": "Breadcrumbs/Breadcrumb", isLink: link != null, isActive, onClick }, link ? /* @__PURE__ */ React20.createElement(BreadcrumbLink, { to: link }, label) : /* @__PURE__ */ React20.createElement(BreadcrumbText, null, label));
}
var BreadcrumbText, BreadcrumbWrapper, BreadcrumbLink;
var init_Breadcrumb = __esm({
  "node_modules/@redocly/theme/src/components/Breadcrumbs/Breadcrumb.tsx"() {
    init_styled_components_esm();
    init_Link();
    BreadcrumbText = styled_components_esm_default.div.withConfig({
      displayName: "Breadcrumb__BreadcrumbText",
      componentId: "sc-3atq9w-0"
    })(["cursor:default;"]);
    BreadcrumbWrapper = styled_components_esm_default.div.withConfig({
      displayName: "Breadcrumb__BreadcrumbWrapper",
      componentId: "sc-3atq9w-1"
    })(["border-radius:var(--border-radius);", " ", ""], (props) => props.isLink && `
    &:hover {
      text-decoration: var(--breadcrumbs-text-decoration-hover);
    }
  `, (props) => props.isActive && `
    color: var(--breadcrumbs-text-color-active);
  `);
    BreadcrumbLink = styled_components_esm_default(Link).withConfig({
      displayName: "Breadcrumb__BreadcrumbLink",
      componentId: "sc-3atq9w-2"
    })(["text-decoration:none;color:var(--breadcrumbs-text-color);&:visited{color:var(--breadcrumbs-text-color);}"]);
  }
});

// node_modules/@redocly/theme/src/components/Breadcrumbs/Breadcrumbs.tsx
import React21 from "react";
function Breadcrumbs(props) {
  const {
    useBreadcrumbs,
    useTelemetry
  } = useThemeHooks();
  const telemetry = useTelemetry();
  const breadcrumbs = useBreadcrumbs();
  if (breadcrumbs.length === 0) {
    return null;
  }
  return /* @__PURE__ */ React21.createElement(BreadcrumbsWrapper, { "data-component-name": "Breadcrumbs/Breadcrumbs", className: props.className }, breadcrumbs.map((breadcrumb, idx) => {
    const isLast = idx === breadcrumbs.length - 1;
    return /* @__PURE__ */ React21.createElement(React21.Fragment, { key: idx }, /* @__PURE__ */ React21.createElement(Breadcrumb, { link: breadcrumb.link, label: breadcrumb.label, isActive: isLast, onClick: () => {
      telemetry.send("breadcrumb_clicked", {
        link: breadcrumb.link,
        position: idx + 1,
        total_breadcrumbs: breadcrumbs.length
      });
    } }), isLast ? null : /* @__PURE__ */ React21.createElement("span", null, "/"));
  }));
}
var BreadcrumbsWrapper;
var init_Breadcrumbs = __esm({
  "node_modules/@redocly/theme/src/components/Breadcrumbs/Breadcrumbs.tsx"() {
    init_styled_components_esm();
    init_hooks();
    init_Breadcrumb();
    BreadcrumbsWrapper = styled_components_esm_default.div.withConfig({
      displayName: "Breadcrumbs__BreadcrumbsWrapper",
      componentId: "sc-1l1tnyj-0"
    })(["display:flex;flex-direction:row;align-items:center;color:var(--breadcrumbs-text-color);font-size:var(--breadcrumbs-font-size);flex-wrap:wrap;> div{padding:var(--breadcrumbs-padding);}> span{padding:var(--breadcrumbs-gap);}@media print{display:none;}"]);
  }
});

export {
  markdownBaseTableCss,
  init_baseTable,
  markdownLinksCss,
  init_links,
  headingAnchorCss,
  init_headingAnchor,
  Markdown,
  init_Markdown,
  Comment,
  init_Comment,
  TableOfContent,
  init_TableOfContent,
  EditIcon,
  init_EditIcon,
  EditPageButton,
  init_EditPageButton,
  Reasons,
  init_Reasons,
  StarIcon,
  init_StarIcon,
  StarFilledIcon,
  init_StarFilledIcon,
  Stars,
  init_Stars,
  FEEDBACK_MAX_RATING,
  Rating,
  init_Rating,
  Sentiment,
  init_Sentiment,
  FaceDissatisfiedIcon,
  init_FaceDissatisfiedIcon,
  FaceSatisfiedIcon,
  init_FaceSatisfiedIcon,
  FaceNeutralIcon,
  init_FaceNeutralIcon,
  MOOD_STATES,
  Mood,
  init_Mood,
  MAX_SCALE,
  Scale,
  init_Scale,
  Feedback,
  init_Feedback,
  Breadcrumb,
  init_Breadcrumb,
  Breadcrumbs,
  init_Breadcrumbs
};
