// @redocly/realm/dist/server/esbuild/cache/server/chunks/chunk-YLDZHCE2.js
import { createRequire as topLevelCreateRequire } from 'module';
 const require = topLevelCreateRequire(import.meta.url);
import {
  ArrowLeftIcon,
  init_ArrowLeftIcon
} from "./chunk-WLDWULA7.js";
import {
  ArrowRightIcon,
  init_ArrowRightIcon
} from "./chunk-XQXU7C5Q.js";
import {
  Button,
  init_Button
} from "./chunk-SX3NFZJN.js";
import {
  init_hooks,
  init_styled_components_esm,
  styled_components_esm_default,
  useThemeConfig,
  useThemeHooks
} from "./chunk-WNBJGIID.js";

// node_modules/@redocly/theme/src/components/PageNavigation/PageNavigation.tsx
init_styled_components_esm();
init_hooks();
import React3 from "react";

// node_modules/@redocly/theme/src/components/PageNavigation/PreviousButton.tsx
init_styled_components_esm();
init_hooks();
init_Button();
init_ArrowLeftIcon();
import React from "react";
function PreviousButton({
  prevPage,
  className
}) {
  const {
    navigation
  } = useThemeConfig();
  const {
    useTranslate
  } = useThemeHooks();
  const {
    translate
  } = useTranslate();
  if (!prevPage || navigation?.previousButton?.hide) {
    return /* @__PURE__ */ React.createElement("div", null, "\xA0");
  }
  const prevPageText = prevPage.label || prevPage.routeSlug || "";
  const defaultLabel = navigation?.previousButton?.text || "Previous page";
  const translationKey = "page.previousButton";
  const label = translate(translationKey, {
    defaultValue: defaultLabel
  });
  return /* @__PURE__ */ React.createElement(PreviousButtonWrapper, { "data-component-name": "PageNavigation/PreviousButton", "data-translation-key": translationKey }, /* @__PURE__ */ React.createElement(PreviousPageLabel, null, label), /* @__PURE__ */ React.createElement(PreviousPageButton, { size: "large", to: prevPage.link, extraClass: className, variant: "link", icon: /* @__PURE__ */ React.createElement(ArrowLeftIcon, null), iconPosition: "left" }, prevPageText));
}
var PreviousButtonWrapper = styled_components_esm_default.div.withConfig({
  displayName: "PreviousButton__PreviousButtonWrapper",
  componentId: "sc-1ligouq-0"
})(["display:flex;flex-direction:column;text-align:left;"]);
var PreviousPageButton = styled_components_esm_default(Button).withConfig({
  displayName: "PreviousButton__PreviousPageButton",
  componentId: "sc-1ligouq-1"
})(["text-wrap:wrap;padding-left:0;padding-right:0;"]);
var PreviousPageLabel = styled_components_esm_default.span.withConfig({
  displayName: "PreviousButton__PreviousPageLabel",
  componentId: "sc-1ligouq-2"
})(["font-size:var(--font-size-sm);line-height:var(--line-height-sm);font-weight:var(--font-weight-regular);"]);

// node_modules/@redocly/theme/src/components/PageNavigation/NextButton.tsx
init_styled_components_esm();
init_hooks();
init_Button();
init_ArrowRightIcon();
import * as React2 from "react";
function NextButton({
  nextPage,
  className
}) {
  const {
    navigation
  } = useThemeConfig();
  const {
    useTranslate
  } = useThemeHooks();
  const {
    translate
  } = useTranslate();
  if (!nextPage || navigation?.nextButton?.hide) {
    return /* @__PURE__ */ React2.createElement("div", null, "\xA0");
  }
  const nextPageText = nextPage.label || nextPage.routeSlug || "";
  const defaultLabel = navigation?.nextButton?.text || "Next page";
  const translationKey = "page.nextButton";
  const label = translate(translationKey, {
    defaultValue: defaultLabel
  });
  return /* @__PURE__ */ React2.createElement(NextButtonWrapper, { "data-component-name": "PageNavigation/NextButton", "data-translation-key": translationKey }, /* @__PURE__ */ React2.createElement(NextPageLabel, null, label), /* @__PURE__ */ React2.createElement(NextPageButton, { size: "large", to: nextPage.link, extraClass: className, variant: "link", icon: /* @__PURE__ */ React2.createElement(ArrowRightIcon, null), iconPosition: "right" }, nextPageText));
}
var NextButtonWrapper = styled_components_esm_default.div.withConfig({
  displayName: "NextButton__NextButtonWrapper",
  componentId: "sc-11l11l9-0"
})(["display:flex;flex-direction:column;text-align:right;"]);
var NextPageButton = styled_components_esm_default(Button).withConfig({
  displayName: "NextButton__NextPageButton",
  componentId: "sc-11l11l9-1"
})(["text-wrap:wrap;padding-left:0;padding-right:0;"]);
var NextPageLabel = styled_components_esm_default.span.withConfig({
  displayName: "NextButton__NextPageLabel",
  componentId: "sc-11l11l9-2"
})(["font-size:var(--font-size-sm);line-height:var(--line-height-sm);font-weight:var(--font-weight-regular);"]);

// node_modules/@redocly/theme/src/components/PageNavigation/PageNavigation.tsx
function PageNavigation({
  nextPage,
  prevPage,
  className
}) {
  const {
    navigation
  } = useThemeConfig();
  if (navigation?.previousButton?.hide && navigation?.nextButton?.hide) {
    return null;
  }
  return /* @__PURE__ */ React3.createElement(PageNavigationWrapper, { "data-component-name": "PageNavigation/PageNavigation", className }, /* @__PURE__ */ React3.createElement(PreviousButton, { prevPage }), /* @__PURE__ */ React3.createElement(NextButton, { nextPage }));
}
var PageNavigationWrapper = styled_components_esm_default.div.withConfig({
  displayName: "PageNavigation__PageNavigationWrapper",
  componentId: "sc-1814chp-0"
})(["display:flex;justify-content:space-between;margin:25px 0;width:100%;@media print{display:none;}"]);

export {
  PageNavigation
};
