// @redocly/realm/dist/server/esbuild/cache/server/chunks/chunk-TL5RDG3G.js
import { createRequire as topLevelCreateRequire } from 'module';
 const require = topLevelCreateRequire(import.meta.url);
import {
  __export
} from "./chunk-35AYGKFE.js";

// markdoc-schema-ns:@redocly-markdoc/components
var components = {};

// node_modules/@redocly/realm/dist/client/ScriptLoader.js
import { useEffect as u } from "react";
import { useLocation as d } from "react-router-dom";
var r = [];
var i = [];
var c = /* @__PURE__ */ new Map();
var s = null;
function h(t) {
  t && r.push(t);
}
function P(t) {
  t && i.push(t);
}
function w() {
  return s || Promise.resolve();
}
function x() {
  const t = d();
  return u(() => {
    s = new Promise((e) => {
      m(r).then(() => {
        const n2 = f(i);
        return i = [], n2;
      }).then(() => {
        e();
      }), r = [];
    });
  }, [t.pathname]), null;
}
function a(t) {
  const e = c.get(t);
  if (e)
    return e;
  const n2 = new Promise((p, l) => {
    const o2 = document.createElement("script");
    o2.src = t, o2.addEventListener("load", () => p()), o2.addEventListener("error", () => l("Error loading script")), document.body.appendChild(o2);
  });
  return c.set(t, n2), n2;
}
async function m(t) {
  for (const e of t)
    await a(e);
}
async function f(t) {
  return Promise.allSettled(t.map(a));
}

// node_modules/@redocly/realm/dist/server/plugins/markdown/markdoc/custom-components/index.js
var custom_components_exports = {};
__export(custom_components_exports, {
  HtmlScript: () => o,
  openapi: () => n
});

// node_modules/@redocly/realm/dist/server/plugins/markdown/markdoc/custom-components/html-script.js
import * as i2 from "react";
function o(e) {
  let t = false;
  const { children: r2, ...c2 } = e;
  if (typeof document < "u") {
    const n2 = document.querySelector(`script[src="${c2.src}"]`);
    (n2 || !n2 && r2) && (t = true);
  }
  return i2.useEffect(() => {
    t || (e.async ? P(e.src) : h(e.src));
  }, [e.src, e.async, t]), i2.createElement("script", { ...c2, dangerouslySetInnerHTML: r2 ? { __html: r2?.toString() } : void 0 });
}

// node_modules/@redocly/realm/dist/server/plugins/markdown/markdoc/custom-components/index.js
async function n() {
  return await import("./openapi-A5B7PWRT.js");
}

export {
  components,
  w,
  x,
  custom_components_exports
};
