// @redocly/openapi-docs/lib-esm/components/Security/SecurityButton.js
import{jsx as _jsx,jsxs as _jsxs}from"react/jsx-runtime";import styled from"styled-components";import{SecurityIcon}from"@redocly/theme";export function SecurityButton({securities:e,onClick:t}){return _jsxs(StyledButton,{onClick:t,children:[_jsx(SecurityIcon,{}),_jsx(Title,{children:"Security:"}),_jsx(SecurityList,{children:e.map((t=>{const i=e.length>1&&t.schemes.length>1;return`\n          ${i?"(":""}\n          ${t.schemes.map((({id:e})=>e)).join(" and ")}\n          ${i?")":""}\n          `})).join(" or ")})]})}const StyledButton=styled.button`
  display: flex;
  align-items: center;
  align-self: flex-start;
  gap: var(--spacing-xxs);
  min-height: 32px;
  background: var(--color-info-bg);
  padding: 0 var(--spacing-xs);
  margin-top: var(--spacing-base);
  color: var(--link-color-primary);
  border: none;
  font-size: var(--font-size-base);
  font-family: var(--font-family-base);
  line-height: var(--line-height-base);
  border-radius: var(--border-radius-md);
  max-width: 100%;
`,Title=styled.span`
  font-family: var(--font-family-monospaced);
  font-weight: var(--font-weight-semibold);
  letter-spacing: -0.5px;
`,SecurityList=styled.span`
  flex: 1;
  cursor: pointer;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
`;
