// @redocly/theme/src/core/types/catalog.ts
import type { CatalogFilterConfig } from '@redocly/theme/config';

export type FilteredCatalog = {
  groups: { title: string; items: CatalogItem[] }[];
  filters: ResolvedFilter[];
  filterTerm: string;
  setFilterTerm: (term: string) => void;
};

export type ResolvedFilter = Omit<CatalogFilterConfig, 'options'> & {
  options: {
    value: string;
    count: number;
  }[];
  filteredOptions: {
    value: string;
    count: number;
  }[];
  toggleOption: (option: string) => void;
  selectOption: (option: string) => void;
  parentUsed: boolean;
  selectedOptions: Set<string> | { from?: string; to?: string };
};

export type CatalogItem = {
  title: string;
  link: string;
  summary?: string;
  description?: string;
  image?: string;
  docsLink?: string;

  scorecardLevel?: string;
  scorecardLevelIdx?: number;
  scorecardLevels?: Record<string, { uniqueErrors: number; uniqueWarnings: number }>;
  scoreCardSlug?: string;
  tags?: unknown[];
  versions?: CatalogItem[];
  version?: string;
  versionFolderId?: string;
  [k: string]: unknown;
};
