// @redocly/theme/lib/core/constants/common.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MobileMenuType = exports.MenuItemType = exports.DEFAULT_FEEDBACK_TYPE = exports.FEEDBACK_TYPES = exports.DEFAULT_LOCALE_PLACEHOLDER = exports.REDOCLY_TEAMS_RBAC = void 0;
var config_1 = require("@redocly/config");
Object.defineProperty(exports, "REDOCLY_TEAMS_RBAC", { enumerable: true, get: function () { return config_1.REDOCLY_TEAMS_RBAC; } });
exports.DEFAULT_LOCALE_PLACEHOLDER = 'default_locale';
var FEEDBACK_TYPES;
(function (FEEDBACK_TYPES) {
    FEEDBACK_TYPES["RATING"] = "rating";
    FEEDBACK_TYPES["SENTIMENT"] = "sentiment";
    FEEDBACK_TYPES["COMMENT"] = "comment";
    FEEDBACK_TYPES["MOOD"] = "mood";
    FEEDBACK_TYPES["PROBLEM"] = "problem";
    FEEDBACK_TYPES["SCALE"] = "scale";
})(FEEDBACK_TYPES || (exports.FEEDBACK_TYPES = FEEDBACK_TYPES = {}));
exports.DEFAULT_FEEDBACK_TYPE = FEEDBACK_TYPES.SENTIMENT;
var MenuItemType;
(function (MenuItemType) {
    MenuItemType["Separator"] = "separator";
    MenuItemType["DrillDown"] = "drilldown";
    MenuItemType["Group"] = "group";
    MenuItemType["Operation"] = "operation";
    MenuItemType["Default"] = "default";
})(MenuItemType || (exports.MenuItemType = MenuItemType = {}));
var MobileMenuType;
(function (MobileMenuType) {
    MobileMenuType["MAIN_MENU"] = "MAIN_MENU";
    MobileMenuType["PRODUCT"] = "PRODUCT";
    MobileMenuType["PAGE"] = "PAGE";
})(MobileMenuType || (exports.MobileMenuType = MobileMenuType = {}));
