// @redocly/openapi-docs/lib-esm/components/common/Dropdown/Dropdown.js
import{jsx as _jsx}from"react/jsx-runtime";import styled from"styled-components";import{Dropdown as DropdownTheme,typedMemo,CheckmarkIcon,DropdownMenu,DropdownMenuItem,Button}from"@redocly/theme";function DropdownComponent({options:e,value:n,onChange:o,className:t,triggerVariant:r="outlined",triggerSize:i="small"}){var a;const l=null===(a=e.find((e=>e.value===n)))||void 0===a?void 0:a.label;return 1===e.length?_jsx(Title,{children:l}):_jsx(DropdownTheme,{className:t,trigger:_jsx(Button,{variant:r,size:i,type:"button",children:l}),withArrow:!0,children:_jsx(DropdownMenu,{children:e.map((e=>_jsx(StyledDropdownMenuItem,{active:e.value===n,onAction:()=>o(e),suffix:e.value===n&&_jsx(StyledCheckmarkIcon,{}),children:e.label},e.label)))})})}export const Dropdown=typedMemo(DropdownComponent);const StyledCheckmarkIcon=styled(CheckmarkIcon)`
  width: 16px;
  height: 16px;
  margin-left: auto;
`,StyledDropdownMenuItem=styled(DropdownMenuItem)`
  --dropdown-menu-item-justify-content: space-between;

  width: 100%;
  display: flex;
  align-items: center;
  gap: var(--spacing-md);
  padding: var(--spacing-xxs) var(--spacing-sm);
  font-size: var(--font-size-base);
  line-height: var(--line-height-base);
`,Title=styled.span`
  margin-right: var(--spacing-unit);
  cursor: default;
  background: none;
  border: none;
  padding: 0;
  color: var(--text-color-primary);
  font-size: var(--font-size-lg);
  line-height: var(--line-height-lg);
  font-family: var(--font-family-base);
  font-weight: var(--font-weight-regular);
`;
