// @redocly/theme/lib/components/Panel/PanelBody.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PanelBody = void 0;
const styled_components_1 = __importStar(require("styled-components"));
const showPanel = (0, styled_components_1.keyframes) `
  0% {
    transform: translateY(-10px);
  }
  100% {
    transform: translateY(0);
  }
`;
const showPanelAnimation = (0, styled_components_1.css) `
  animation: ${showPanel} 0.2s;
`;
exports.PanelBody = styled_components_1.default.div.attrs(({ className }) => ({
    'data-component-name': 'Panel/PanelBody',
    className,
})) `
  ${({ animate }) => animate && showPanelAnimation};
  ${({ hidden }) => hidden && 'visibility: hidden'};

  background-color: var(--panel-body-bg-color);

  font-family: var(--panel-body-font-family-local);
  font-size: var(--panel-body-font-size-local);
  font-weight: var(--panel-body-font-weight-local);
  border: var(--panel-body-border-local);
  padding: var(--panel-body-padding-local);
  color: var(--panel-body-text-color);

  border-top-left-radius: var(--panel-border-radius);
  border-top-right-radius: var(--panel-border-radius);
  border-bottom-right-radius: var(--panel-border-radius);
  border-bottom-left-radius: var(--panel-border-radius);
`;
