// @redocly/theme/lib/components/SidebarActions/styled.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ControlsWrapChangeLayoutButtons = exports.ControlsWrap = exports.StyledChangeViewButtonWrap = exports.StyledChangeViewButton = void 0;
const styled_components_1 = __importStar(require("styled-components"));
const utils_1 = require("../../core/utils");
exports.StyledChangeViewButton = styled_components_1.default.span `
  display: flex;
  position: relative;
  flex-direction: ${({ collapsedSidebar }) => (collapsedSidebar ? 'column' : 'row')};
  justify-content: center;
  align-items: center;
  cursor: pointer;
  border-radius: var(--border-radius);
  background: var(--segmented-buttons-bg-color-main);
  z-index: 1;
  padding: calc(var(--spacing-unit) / 2);
`;
exports.StyledChangeViewButtonWrap = styled_components_1.default.span `
  width: 20px;
  height: 20px;
  border-radius: var(--border-radius);
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: ${({ active }) => active && 'var(--segmented-buttons-bg-color-inner)'};
`;
exports.ControlsWrap = styled_components_1.default.span `
  display: none;
  justify-content: flex-end;
  bottom: 0;
  flex-direction: row;
  gap: var(--spacing-xs);
  align-items: center;
  margin-left: var(--sidebar-controls-margin-left);
  z-index: 2;
  background: var(--sidebar-bg-color);

  ${({ isCollapsed }) => isCollapsed &&
    (0, styled_components_1.css) `
      flex-direction: column;
      margin-left: 0;
      gap: var(--sidebar-controls-collapsed-gap);
      padding: var(--sidebar-controls-collapsed-padding-vertical)
        var(--sidebar-controls-collapsed-padding-horizontal);
    `}

  @media screen and (min-width: ${utils_1.breakpoints.small}) {
    display: flex;
  }

  @media print {
    display: none;
  }
`;
exports.ControlsWrapChangeLayoutButtons = styled_components_1.default.span `
  display: flex;
  bottom: 0;
  flex-direction: ${({ isCollapsed }) => (isCollapsed ? 'column-reverse' : 'row')};
  gap: ${({ isCollapsed }) => (isCollapsed ? '10px' : 'var(--spacing-xs)')};
  align-items: center;
`;
