// @redocly/theme/lib/components/Feedback/Mood.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MOOD_STATES = void 0;
exports.Mood = Mood;
const React = __importStar(require("react"));
const react_1 = require("react");
const styled_components_1 = __importDefault(require("styled-components"));
const hooks_1 = require("../../core/hooks");
const RadioCheckButtonIcon_1 = require("../../icons/RadioCheckButtonIcon/RadioCheckButtonIcon");
const Comment_1 = require("../../components/Feedback/Comment");
const Reasons_1 = require("../../components/Feedback/Reasons");
const Button_1 = require("../../components/Button/Button");
const FaceDissatisfiedIcon_1 = require("../../icons/FaceDissatisfiedIcon/FaceDissatisfiedIcon");
const FaceSatisfiedIcon_1 = require("../../icons/FaceSatisfiedIcon/FaceSatisfiedIcon");
const FaceNeutralIcon_1 = require("../../icons/FaceNeutralIcon/FaceNeutralIcon");
var MOOD_STATES;
(function (MOOD_STATES) {
    MOOD_STATES["SATISFIED"] = "satisfied";
    MOOD_STATES["NEUTRAL"] = "neutral";
    MOOD_STATES["DISSATISFIED"] = "dissatisfied";
})(MOOD_STATES || (exports.MOOD_STATES = MOOD_STATES = {}));
function Mood({ settings, onSubmit, className }) {
    const { label, submitText, comment: commentSettings, reasons: reasonsSettings } = settings || {};
    const [score, setScore] = React.useState('');
    const [isSubmitted, setIsSubmitted] = React.useState(false);
    const [comment, setComment] = React.useState('');
    const [reasons, setReasons] = React.useState([]);
    const { useTranslate } = (0, hooks_1.useThemeHooks)();
    const { translate } = useTranslate();
    const checkIfShouldDisplayReasons = (score) => {
        if (!score || !reasonsSettings) {
            return false;
        }
        const scoreSpecificReasonsSettings = reasonsSettings[score];
        if (typeof (scoreSpecificReasonsSettings === null || scoreSpecificReasonsSettings === void 0 ? void 0 : scoreSpecificReasonsSettings.hide) === 'undefined') {
            return !reasonsSettings.hide;
        }
        return !(scoreSpecificReasonsSettings === null || scoreSpecificReasonsSettings === void 0 ? void 0 : scoreSpecificReasonsSettings.hide);
    };
    const renderCommentLabel = (score) => {
        if (!commentSettings || (commentSettings === null || commentSettings === void 0 ? void 0 : commentSettings.hide) === true)
            return '';
        switch (score) {
            case MOOD_STATES.SATISFIED:
                return (commentSettings.satisfiedLabel ||
                    translate('feedback.settings.comment.satisfiedLabel', 'What was most helpful?'));
            case MOOD_STATES.NEUTRAL:
                return (commentSettings.neutralLabel ||
                    translate('feedback.settings.comment.neutralLabel', 'What can we improve?'));
            case MOOD_STATES.DISSATISFIED:
                return (commentSettings.dissatisfiedLabel ||
                    translate('feedback.settings.comment.dissatisfiedLabel', 'What can we improve?'));
            default:
                return translate('feedback.settings.comment.satisfiedLabel', 'What can we improve?');
        }
    };
    const renderReasonsLabel = (score) => {
        var _a;
        if (!reasonsSettings || (reasonsSettings === null || reasonsSettings === void 0 ? void 0 : reasonsSettings.hide) === true)
            return '';
        return (_a = reasonsSettings[score]) === null || _a === void 0 ? void 0 : _a.label;
    };
    const resolveReasonsItems = (score) => {
        var _a, _b;
        if (!reasonsSettings) {
            return [];
        }
        const defaultReasonsItems = (reasonsSettings === null || reasonsSettings === void 0 ? void 0 : reasonsSettings.items) || [];
        return defaultReasonsItems.concat((_b = (_a = reasonsSettings[score]) === null || _a === void 0 ? void 0 : _a.items) !== null && _b !== void 0 ? _b : []);
    };
    const resolveReasonsComponent = (score) => {
        var _a;
        if (!reasonsSettings || (reasonsSettings === null || reasonsSettings === void 0 ? void 0 : reasonsSettings.hide) === true)
            return 'checkbox';
        return (_a = reasonsSettings[score]) === null || _a === void 0 ? void 0 : _a.component;
    };
    const displayReasons = checkIfShouldDisplayReasons(score);
    const displayComment = !!(score && !(commentSettings === null || commentSettings === void 0 ? void 0 : commentSettings.hide));
    const displaySubmitBnt = !!(score && (displayReasons || displayComment));
    const onSubmitMoodForm = () => {
        onSubmit({
            score: remapScore(score),
            comment,
            reasons,
        });
        setIsSubmitted(true);
    };
    const onCancelMoodForm = () => {
        setScore('');
    };
    (0, react_1.useEffect)(() => {
        if (score && !displayComment && !displayReasons) {
            onSubmitMoodForm();
        }
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [score, displayComment, displayReasons]);
    if (isSubmitted) {
        return (React.createElement(MoodWrapper, { "data-component-name": "Feedback/Mood" },
            React.createElement(StyledFormMandatoryFields, null,
                React.createElement(Label, { "data-translation-key": "feedback.settings.submitText" }, submitText ||
                    translate('feedback.settings.submitText', 'Thank you for helping improve our documentation!')),
                React.createElement(RadioCheckButtonIcon_1.RadioCheckButtonIcon, null))));
    }
    return (React.createElement(MoodWrapper, { "data-component-name": "Feedback/Mood", className: className },
        React.createElement(StyledForm, null,
            React.createElement(StyledFormMandatoryFields, null,
                React.createElement(Label, { "data-translation-key": "feedback.settings.label" }, label || translate('feedback.settings.label', 'Was this helpful?')),
                React.createElement(StyledMandatoryFieldContainer, null,
                    React.createElement(Button_1.Button, { "aria-label": MOOD_STATES.DISSATISFIED, type: "button", size: "medium", variant: score === MOOD_STATES.DISSATISFIED ? 'primary' : 'secondary', tone: score === MOOD_STATES.DISSATISFIED ? 'danger' : 'default', onClick: () => {
                            setScore(MOOD_STATES.DISSATISFIED);
                        }, icon: React.createElement(FaceDissatisfiedIcon_1.FaceDissatisfiedIcon, null) }),
                    React.createElement(Button_1.Button, { "aria-label": MOOD_STATES.NEUTRAL, type: "button", variant: score === MOOD_STATES.NEUTRAL ? 'primary' : 'secondary', size: "medium", onClick: () => {
                            setScore(MOOD_STATES.NEUTRAL);
                        }, icon: React.createElement(FaceNeutralIcon_1.FaceNeutralIcon, null) }),
                    React.createElement(Button_1.Button, { "aria-label": MOOD_STATES.SATISFIED, type: "button", variant: score === MOOD_STATES.SATISFIED ? 'primary' : 'secondary', size: "medium", onClick: () => {
                            setScore(MOOD_STATES.SATISFIED);
                        }, icon: React.createElement(FaceSatisfiedIcon_1.FaceSatisfiedIcon, null) }))),
            (displayReasons || displayComment) && (React.createElement(StyledFormOptionalFields, null,
                displayReasons && (React.createElement(Reasons_1.Reasons, { settings: {
                        label: renderReasonsLabel(score),
                        items: resolveReasonsItems(score),
                        component: resolveReasonsComponent(score),
                    }, onChange: setReasons })),
                displayComment && (React.createElement(Comment_1.Comment, { standAlone: false, onSubmit: ({ comment }) => setComment(comment), settings: { label: renderCommentLabel(score) } })))),
            displaySubmitBnt && (React.createElement(ButtonsContainer, null,
                React.createElement(Button_1.Button, { onClick: onCancelMoodForm, variant: "text", size: "small" }, "Cancel"),
                React.createElement(Button_1.Button, { onClick: onSubmitMoodForm, variant: "secondary", size: "small" }, "Submit"))))));
}
const remapScore = (score) => {
    switch (score) {
        case MOOD_STATES.SATISFIED:
            return 1;
        case MOOD_STATES.NEUTRAL:
            return 0;
        case MOOD_STATES.DISSATISFIED:
            return -1;
        default:
            return 0;
    }
};
const MoodWrapper = styled_components_1.default.div `
  font-family: var(--feedback-font-family);
  display: flex;
  justify-content: space-between;
  align-items: center;
`;
const Label = styled_components_1.default.h4 `
  font-family: var(--feedback-font-family);
  font-weight: var(--font-weight-regular);
  font-size: var(--feedback-header-font-size);
  line-height: var(--feedback-header-line-height);
  color: var(--feedback-header-text-color);
  margin: 0;
`;
const ButtonsContainer = styled_components_1.default.div `
  display: flex;
  justify-content: end;
  gap: var(--spacing-xxs);
`;
const StyledForm = styled_components_1.default.form `
  width: 100%;
  gap: var(--spacing-sm);
  display: flex;
  flex-direction: column;
`;
const StyledFormOptionalFields = styled_components_1.default.div `
  display: flex;
  flex-flow: column;
`;
const StyledFormMandatoryFields = styled_components_1.default.div `
  display: flex;
  justify-content: space-between;
  align-items: center;
  gap: var(--spacing-xs);

  > svg {
    margin: var(--button-icon-padding) 0; /* prevent hight of the parent jumping */
  }
`;
const StyledMandatoryFieldContainer = styled_components_1.default.div `
  display: flex;
  align-items: center;
  gap: var(--spacing-xxs);
`;
