// @redocly/theme/lib/components/Feedback/Stars.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Stars = Stars;
const React = __importStar(require("react"));
const styled_components_1 = __importDefault(require("styled-components"));
const StarIcon_1 = require("../../icons/StarIcon/StarIcon");
const StarFilledIcon_1 = require("../../icons/StarFilledIcon/StarFilledIcon");
function Stars({ max = 5, onChange, value }) {
    const [hovered, setHovered] = React.useState(value || 0);
    const stars = [];
    for (let index = 1; index <= max; index++) {
        stars.push(React.createElement(Star, { "data-testid": `star-${index}`, key: index, onClick: () => onChange(index), onMouseOver: () => setHovered(index), onMouseLeave: () => !value && setHovered(0) }, hovered < index ? (React.createElement(StarIcon_1.StarIcon, { color: "var(--feedback-star-color)" })) : (React.createElement(StarFilledIcon_1.StarFilledIcon, { color: "var(--feedback-star-color)" }))));
    }
    return React.createElement(StarsWrapper, { "data-component-name": "Feedback/Stars" }, stars);
}
const StarsWrapper = styled_components_1.default.div `
  display: flex;
  gap: var(--spacing-xs);
  flex-direction: row;
  align-items: center;
`;
const Star = styled_components_1.default.button.attrs(() => ({
    type: 'button',
})) `
  cursor: pointer;
  padding: 0;
  border: none;
  background-color: transparent;
  height: 16px;
  gap: var(--spacing-xs);
`;
